<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Martin Garlovsky" />

<meta name="date" content="2021-04-01" />

<title>Statistical analysis</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DmSP3</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MartinGarlovsky/DmSP3">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Statistical analysis</h1>
<h4 class="author">Martin Garlovsky</h4>
<h4 class="date">2021-04-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DmSP3/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220210code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220210code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220210)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMartinGarlovskyDmSP3tree8e0c73045fe75eb50cfa8ef2c27c03f6b7b4d24btargetblank8e0c730a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MartinGarlovsky/DmSP3/tree/8e0c73045fe75eb50cfa8ef2c27c03f6b7b4d24b" target="_blank">8e0c730</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMartinGarlovskyDmSP3tree8e0c73045fe75eb50cfa8ef2c27c03f6b7b4d24btargetblank8e0c730a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MartinGarlovsky/DmSP3/tree/8e0c73045fe75eb50cfa8ef2c27c03f6b7b4d24b" target="_blank">8e0c730</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  README.html
    Untracked:  code/parse_PAML.R
    Untracked:  data/040721_PBS_NaCl_1PeptideLFQ.csv
    Untracked:  data/DmSP3_Xlinked_Ribosomes.xlsx
    Untracked:  data/DmSPii_Supp.Table3.csv
    Untracked:  data/FlyAtlas2/
    Untracked:  data/FlyBase/
    Untracked:  data/Fly_Proteomes_LumosFusion/
    Untracked:  data/HaltNohaltPBST_uniprot.csv
    Untracked:  data/Halt_NoHalt_PBST.xlsx
    Untracked:  data/KB_10MSE_sperm_Edited.xlsx
    Untracked:  data/dm6_ver78_age.csv
    Untracked:  data/dmel_SFPs_wigby_etal2020.csv
    Untracked:  data/oneratio_SWAMP.txt
    Untracked:  figures/
    Untracked:  output/DmSP_ribosomes.csv
    Untracked:  output/DmSP_which_proteome.csv
    Untracked:  output/GO_lists/
    Untracked:  output/OMIM_results/
    Untracked:  output/PAML/
    Untracked:  output/Recent_DmSP3.csv
    Untracked:  output/Top20DmSP3.csv
    Untracked:  output/Ylinked_DmSP3.csv
    Untracked:  output/paralog_switches.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/statistical_analysis.Rmd</code>) and HTML (<code>docs/statistical_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/8e0c73045fe75eb50cfa8ef2c27c03f6b7b4d24b/analysis/statistical_analysis.Rmd" target="_blank">8e0c730</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
<td>
change readme, fix figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MartinGarlovsky/DmSP3/3ce1c47a48de012c7883d568131a50529ecd5316/docs/statistical_analysis.html" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/d098234e6dafefef80e1c03591ec2e4473786386/analysis/statistical_analysis.Rmd" target="_blank">d098234</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
<td>
update index and analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MartinGarlovsky/DmSP3/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/statistical_analysis.html" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3573ca8c7fd79152a63e2367028bbfac21fd62ff/analysis/statistical_analysis.Rmd" target="_blank">3573ca8</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
<td>
add analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(tidyverse)
library(UpSetR)
library(eulerr)
library(readxl)
library(tidybayes)
library(kableExtra)
library(ggpubr)
library(edgeR)
library(pheatmap)
library(ComplexHeatmap)
library(boot)
library(DT)
library(pals)

library(knitrhooks) # install with devtools::install_github(&quot;nathaneastwood/knitrhooks&quot;)

output_max_height() # a knitrhook option

# set colourblind friendly palette
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)

my_data_table &lt;- function(df){
  datatable(
    df, rownames = FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender = TRUE,
      scrollX = TRUE, scrollY = 400,
      scrollCollapse = TRUE,
      buttons = 
        list(&#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Dpseudo_respiration&#39;)),
      pageLength = 50
    )
  )
}</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<ul>
<li><p>Files downloaded from <a href="http://flybase.org/">FlyBase.org</a>:</p>
<ul>
<li>Genes to Transcript to Protein IDs</li>
<li>Gene names and gene symbols, GO terms, chromosomal location (<code>LOCATION_ARM</code>)</li>
<li><a href="https://flybase.org/reports/FBgg0000130.html">All 168 ribosomal proteins, including paralogs</a></li>
</ul></li>
<li><p>List of all putative Sfps identified by <a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2020.0072">Wigby et al. (2020). <em>Phil. trans. B</em></a></p></li>
<li><p>Previous <em>D. melanogaster</em> sperm proteomes:</p>
<ul>
<li><a href="http://www.nature.com/articles/ng1915">DmSP1</a></li>
<li><a href="http://www.sciencedirect.com/science/article/pii/S1874391910002538">DmSP2</a>, comprising 1108 proteins, combined with the DmSP1</li>
</ul></li>
<li><p>New data generated:</p>
<ul>
<li>Experiment 1: All identified proteins and abundance data</li>
<li>Experiment 2: Abundance data for proteins from ‘NoHalt’, ‘Halt’ controls or ‘PBST’ treatment</li>
<li>Experiment 3: Abundance data for proteins from ‘PBS’ control or ‘NaCl’ treatment</li>
</ul></li>
</ul>
<pre class="r"><code># gene conversion table from FlyBase.org
gene2tran2prot &lt;- read.csv(&#39;data/FlyBase/fbgn_fbtr_fbpp_fb_2021_01.csv&#39;)

# gene IDs and GO terms (by importing gene conversion table FBgns to flybase)
flybase_results &lt;- read.delim(&#39;data/FlyBase/flybase_all-genes.csv&#39;, sep = &#39;,&#39;) %&gt;% 
  dplyr::select(-H_SAPIENS_ORTHOLOGS, -NAME) %&gt;% 
  dplyr::rename(FBgn = X.SUBMITTED.ID)

# Dmel ribosomes - all and those found in sperm
Dm_ribosomes &lt;- read_delim(&#39;data/FlyBase/FlyBase_ribosomes_169.txt&#39;) %&gt;% 
  dplyr::select(FBgn = `#SUBMITTED ID`, H_SAPIENS_ORTHOLOGS:SYMBOL, -SPECIES_ABBREVIATION)

# List of SFPs collated by Wigby et al. 2020 Phil. Trans. B.
wigbySFP &lt;- read.csv(&#39;data/dmel_SFPs_wigby_etal2020.csv&#39;)
# only high confidence Sfps
SFPs &lt;- wigbySFP %&gt;% 
  filter(category == &#39;highconf&#39;)

# Dmel Sperm proteome 1/2 from Wasbrough et al. 2010 J. Prot. 
DmSPI &lt;- read.csv(&#39;data/DmSPii_Supp.Table3.csv&#39;) %&gt;% 
  filter(Proteome.Overlap == &#39;DmSPI&#39; | Proteome.Overlap == &#39;Current Study and DmSPI&#39;)
DmSPII &lt;- read.csv(&#39;data/DmSPii_Supp.Table3.csv&#39;) %&gt;% 
  filter(Proteome.Overlap == &#39;Current Study&#39; | Proteome.Overlap == &#39;Current Study and DmSPI&#39;)
DmSP2 &lt;- read.csv(&#39;data/DmSPii_Supp.Table3.csv&#39;) 

### new data ###
DmSPIII &lt;- read_excel(&#39;data/DmSP3_Xlinked_Ribosomes.xlsx&#39;, sheet = 1) %&gt;% 
  dplyr::rename(CG.no = `CG#`)

# Protein abundance data 
DmSPintensity &lt;- read_excel(&#39;data/KB_10MSE_sperm_Edited.xlsx&#39;, sheet = 1) %&gt;% 
  dplyr::rename(FBgn = `Ensembl Gene ID`) 

# Halt/NoHalt/PBST treatment experiment
PBST_dat &lt;- readxl::read_excel(&#39;data/Halt_NoHalt_PBST.xlsx&#39;) %&gt;% 
  left_join(read_csv(&#39;data/HaltNohaltPBST_uniprot.csv&#39;)) %&gt;% 
  left_join(read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;), 
            by = c(&#39;FBgn&#39; = &#39;X.SUBMITTED.ID&#39;)) %&gt;% 
  distinct(Accession, .keep_all = TRUE)

# NaCl wash data
salt_dat &lt;- read.csv(&#39;data/040721_PBS_NaCl_1PeptideLFQ.csv&#39;, 
                     na.strings = c(&quot;NA&quot;, &quot;NaN&quot;, &quot; &quot;, &#39;&#39;)) %&gt;% 
  dplyr::rename(FBgn = Ensembl.Gene.ID)</code></pre>
</div>
<div id="overlap-between-new-datasets" class="section level3">
<h3>Overlap between new datasets</h3>
<p>We performed three LC-MS experiments on purified sperm samples. For experiment 1, we combine the IDs from two algorithms (i) identifying proteins and (ii) from protein quantitation. We then combine all IDs across the three experiments to compile a complete list of all proteins IDd in the current study.</p>
<pre class="r"><code>### Additional abundance data 
# Data processed for identification was processed separately for quantification. Differences in algorithms results in a slight disparity in the number of protein identifications. 

ab_fbgn &lt;- data.frame(FBgn = unlist(str_split(DmSPintensity$FBgn, pattern = &#39;; &#39;))) %&gt;% na.omit()

# upset(fromList(list(DmSPIII.id = DmSPIII$FBgn,
#                     DmSPIII.ab = ab_fbgn$FBgn)))

# proteins IDd in DmSP3 (identification + abundance data)
DmSP_exp1 &lt;- data.frame(FBgn = unique(c(DmSPIII$FBgn, ab_fbgn$FBgn))) %&gt;% 
  left_join(flybase_results %&gt;% dplyr::select(FBgn, SYMBOL))

## compare ids in each dataset
# upset(fromList(list(exp1 = DmSP_exp1$FBgn,
#                     PBST = PBST_dat$FBgn,
#                     Salt = salt_dat$FBgn)))

# euler diagram
#pdf(&#39;figures/current_study_overlap.pdf&#39;, height = 4, width = 4)
plot(euler(c(&#39;exp1&#39; = 392, &quot;exp2&quot; = 247, &quot;exp3&quot; = 416, 
             &#39;exp1&amp;exp2&#39; = 74, 
             &#39;exp1&amp;exp3&#39; = 169, 
             &#39;exp2&amp;exp3&#39; = 240, 
             &#39;exp1&amp;exp2&amp;exp3&#39; = 1009)
           ),
     quantities = TRUE,
     fills = list(fill = viridis::plasma(n = 3), alpha = .5))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-3-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()

# combine all new data
DmSPIII.2 &lt;- data.frame(FBgn = unique(
  c(DmSP_exp1$FBgn, PBST_dat$FBgn, salt_dat$FBgn))) %&gt;%
  separate_rows(FBgn) %&gt;% 
  left_join(flybase_results %&gt;% dplyr::select(FBgn, SYMBOL)) %&gt;% 
  distinct(FBgn, .keep_all = TRUE)</code></pre>
</div>
<div id="overlap-between-dmsp-1--2-and--3" class="section level2 tabset">
<h2 class="tabset">Overlap between DmSP-1, -2, and -3</h2>
<p>Here we look at the overlap between proteins IDd in the current study (n = 2563) with the previous releases of the DmSP (n = 1108. The current study increases the number of identified proteins significantly.</p>
<div id="dmsp1-vs.-dmsp2-vs.-dmsp3" class="section level3">
<h3>DmSP1 vs. DmSP2 vs. DmSP3</h3>
<pre class="r"><code># upset plot to get numbers in each group
listInput &lt;- list(DmSP.1 = DmSPI$FBgn,
                  DmSP.2 = DmSPII$FBgn,
                  DmSP.3 = DmSPIII.2$FBgn)

#upset(fromList(listInput), order.by = &#39;degree&#39;)

# Eulerr diagram
DmSP_overlap &lt;- plot(euler(c(&#39;DmSP1&#39; = 68, &quot;DmSP2&quot; = 538, &quot;Current Study&quot; = 2069, 
                             &#39;DmSP1&amp;DmSP2&#39; = 8, 
                             &#39;DmSP1&amp;Current Study&#39; = 84, 
                             &#39;DmSP2&amp;Current Study&#39; = 229,
                             &#39;DmSP1&amp;DmSP2&amp;Current Study&#39; = 181)),
     quantities = TRUE,
     fills = list(fill = viridis::viridis(n = 3), alpha = .5))

#pdf(&#39;figures/DmSP1-2-3_overlap.pdf&#39;, height = 4, width = 4)
DmSP_overlap</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-4-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()

# extract genes in each category
x &lt;- upset(fromList(listInput))

intersect_dat &lt;- x$New_data %&gt;% rownames_to_column()

x1 &lt;- unlist(listInput, use.names = FALSE)
x1 &lt;- x1[ !duplicated(x1) ]

# in all 3
all_3 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 1, DmSP.2 == 1, DmSP.3 == 1)
# in 1 and 2
in1_2 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 1, DmSP.2 == 1, DmSP.3 == 0)
# in 1 and 3
in1_3 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 1, DmSP.2 == 0, DmSP.3 == 1)
# in 2 and 3
in2_3 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 0, DmSP.2 == 1, DmSP.3 == 1)
# 1 only 
only1 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 1, DmSP.2 == 0, DmSP.3 == 0)
# 1 only 
only2 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 0, DmSP.2 == 1, DmSP.3 == 0)
# 1 only 
only3 &lt;- intersect_dat %&gt;% filter(DmSP.1 == 0, DmSP.2 == 0, DmSP.3 == 1)</code></pre>
</div>
<div id="cumulative-number-idd" class="section level3">
<h3>Cumulative number ID’d</h3>
<p>Here we combine the list of all proteins identified in the current study with the DmSP2 to compile the DmSP3. We calculated average abundances across experiments, excluding the PBST treatment values, which had a significant effect on the abundance of a large number of proteins (<a href="#PBST">see below</a>). We also create our ‘high-confidence’ DmSP3, excluding proteins identified by fewer than 2 unique peptides or identified in fewer than 2 biological replicates across all experiments.</p>
<pre class="r"><code># new column names for abundance data
exp_names &lt;- c(&#39;REP1.1&#39;,    &#39;REP1.2&#39;,   &#39;REP1.3&#39;,   
               &#39;Halt1&#39;, &#39;Halt2&#39;, &#39;Halt3&#39;,   
               &#39;NoHalt1&#39;,   &#39;NoHalt2&#39;, &#39;NoHalt3&#39;,   
               &#39;NaCl1.1&#39;,   &#39;NaCl1.2&#39;, &#39;NaCl2.1&#39;,   &#39;NaCl2.2&#39;, 
               &#39;NaCl3.1&#39;, &#39;NaCl3.2&#39;, &#39;NaCl4.1&#39;, &#39;NaCl4.2&#39;,  
               &#39;PBS1.1&#39;, &#39;PBS1.2&#39;, &#39;PBS2.1&#39;, &#39;PBS2.2&#39;, 
               &#39;PBS3.1&#39;, &#39;PBS3.2&#39;, &#39;PBS4.1&#39;, &#39;PBS4.2&#39;,  
               &#39;NaCl1&#39;, &#39;NaCl2&#39;, &#39;NaCl3&#39;,   &#39;NaCl4&#39;,    
               &#39;PBS1&#39;,  &#39;PBS2&#39;, &#39;PBS3&#39;, &#39;PBS4&#39;)

# Make the combined DmSP table 
DmSP3 &lt;- tibble(
  # combine IDs for DmSP1, DmSP2, and current study
  FBgn = c(DmSPI$FBgn, DmSPII$FBgn, DmSPIII.2$FBgn)) %&gt;% 
  left_join(read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;), 
            by = c(&#39;FBgn&#39; = &#39;X.SUBMITTED.ID&#39;)) %&gt;% 
  separate_rows(FBgn) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  # variable indicating which study each protein is identified in
  mutate(Sperm_Proteome = case_when(FBgn %in% x1[as.numeric(all_3$rowname)] ~ &#39;DmSP1_2_3&#39;,
                                    FBgn %in% x1[as.numeric(in1_2$rowname)] ~ &#39;DmSP1_2&#39;,
                                    FBgn %in% x1[as.numeric(in1_3$rowname)] ~ &#39;DmSP1_3&#39;,
                                    FBgn %in% x1[as.numeric(in2_3$rowname)] ~ &#39;DmSP2_3&#39;,
                                    FBgn %in% x1[as.numeric(only1$rowname)] ~ &#39;DmSP1&#39;,
                                    FBgn %in% x1[as.numeric(only2$rowname)] ~ &#39;DmSP2&#39;,
                                    FBgn %in% x1[as.numeric(only3$rowname)] ~ &#39;DmSP3&#39;)) %&gt;%
  # Add abundance data from experiment 1 and calculate mean abundance
  left_join(DmSPintensity %&gt;% 
              dplyr::select(FBgn, unique.one = `# Unique Peptides`, contains(&#39;ed):&#39;)) %&gt;% 
              mutate(reps.one = rowSums(dplyr::select(., contains(&#39;ed):&#39;)) &gt; 0),
                     mn.one = rowMeans(dplyr::select(., contains(&#39;ed):&#39;)), na.rm = TRUE))
            ) %&gt;% 
  # Add abundance data from experiment 2 excluding PBST treatment and calculate mean abundance
  left_join(PBST_dat %&gt;% dplyr::select(FBgn, unique.two = `# Unique Peptides`, 45:50) %&gt;% 
              mutate(reps.two = rowSums(dplyr::select(., contains(&#39;Ab&#39;)) &gt; 0),
                     mn.two = rowMeans(dplyr::select(., contains(&#39;Ab&#39;)), na.rm = TRUE))
            ) %&gt;% 
  # Add abundance data from experiment 3 and calculate mean abundance
  left_join(salt_dat %&gt;% dplyr::select(FBgn, unique.three = X..Unique.Peptides, 55:70) %&gt;% 
              mutate(repl1 = rowSums(dplyr::select(., 3:4), na.rm = TRUE),
                     repl2 = rowSums(dplyr::select(., 5:6), na.rm = TRUE),
                     repl3 = rowSums(dplyr::select(., 7:8), na.rm = TRUE),
                     repl4 = rowSums(dplyr::select(., 9:10), na.rm = TRUE),
                     
                     repl5 = rowSums(dplyr::select(., 11:12), na.rm = TRUE),
                     repl6 = rowSums(dplyr::select(., 13:14), na.rm = TRUE),
                     repl7 = rowSums(dplyr::select(., 15:16), na.rm = TRUE),
                     repl8 = rowSums(dplyr::select(., 17:18), na.rm = TRUE)) %&gt;% 
              mutate(reps.three = rowSums(dplyr::select(., contains(&#39;repl&#39;)) &gt; 0),
                     mn.three = rowMeans(dplyr::select(., contains(&#39;repl&#39;)), na.rm = TRUE))
            ) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;%
  # Add variables indicating protein confidence
  mutate(
    # number of replicates each protein found in separately for each experiment and combined
    comb.reps = case_when(reps.one &gt;= 2 | reps.two &gt;= 2 | reps.three &gt;= 2 ~ &#39;confident&#39;,
                          rowSums(dplyr::select(., starts_with(&#39;reps&#39;)), 
                                  na.rm = TRUE) &gt;= 2 ~ &#39;found&#39;,
                          TRUE ~ &#39;no.reps&#39;),
    # number of unique peptides each protein identified by separately for each experiment and combined
    comb.peps = case_when(unique.one &gt;= 2 | unique.two &gt;= 2 | unique.three &gt;= 2 ~ &#39;confident&#39;,
                          rowSums(dplyr::select(., starts_with(&#39;un&#39;)), 
                                  na.rm = TRUE) &gt;= 2 ~ &#39;found&#39;,
                          TRUE ~ &#39;no.peps&#39;),
    # ranked abundance separately for each experiment
    perc.one = percent_rank(mn.one) * 100,
    perc.two = percent_rank(mn.two) * 100,
    perc.three = percent_rank(mn.three) * 100) %&gt;% 
  # rename abundance columns
    rename_at(all_of(
      colnames(dplyr::select(., starts_with(&#39;Abun&#39;), starts_with(&#39;repl&#39;)))), ~ exp_names) %&gt;% 
  mutate(
    # calculate mean abundance across all experiments 
    grand.mean = rowMeans(dplyr::select(., REP1.1:REP1.3, Halt1:NoHalt3, NaCl1:PBS4), na.rm = TRUE),
    # ranked abundance across all experiments
    mean.perc = percent_rank(grand.mean) * 100,
    # add variable for presence in list of putative Sfps
    Sfp = case_when(FBgn %in% SFPs$FBgn ~ &#39;SFP.high&#39;,
                    FBgn %in% wigbySFP$FBgn ~ &#39;SFP.low&#39;,
                    TRUE ~ &#39;Sperm.only&#39;)) %&gt;% 
  drop_na(FBgn)

# #write to file
# DmSP3 %&gt;% 
#   mutate(DmSP2 = if_else(FBgn %in% DmSP2$FBgn, TRUE, FALSE)) %&gt;% 
#   #filter(comb.reps == &#39;found&#39; | comb.peps != &#39;no.peps&#39;) %&gt;% dim
#   #write_csv(&#39;output/DmSP_which_proteome.csv&#39;)

# write FBgn to file for GO analysis
#DmSP3 %&gt;% dplyr::select(FBgn) %&gt;% write_csv(&#39;output/GO_lists/DmSP3_3176.csv&#39;)

# number identified by two or more unique peptides in a single experiment
#DmSP3 %&gt;% filter(comb.peps == &#39;confident&#39;) %&gt;% dim

# number identified in two or more replicates across any experiment 
#DmSP3 %&gt;% filter(comb.reps != &#39;no.reps&#39;) %&gt;% dim

# Confident proteins in current study
DmSPnew_conf &lt;- DmSP3 %&gt;% 
  filter(comb.peps != &#39;no.peps&#39; | comb.reps != &#39;no.reps&#39;)

# combined DmSP1+2+DmSP3 (confident)
DmSP_comb &lt;- DmSP3 %&gt;% 
  filter(comb.peps != &#39;no.peps&#39; | comb.reps != &#39;no.reps&#39; | FBgn %in% DmSP2$FBgn)

# plot cumulative total IDs
cum_plot &lt;- DmSP3 %&gt;% 
  mutate(SP = str_sub(Sperm_Proteome, start = 5)) %&gt;% 
  separate(SP, into = c(&#39;one&#39;, &#39;two&#39;, &#39;three&#39;), sep = &#39;_&#39;, extra = &#39;merge&#39;) %&gt;% 
  pivot_longer(cols = c(one, two, three)) %&gt;% 
  drop_na(value) %&gt;% distinct(FBgn, .keep_all = TRUE) %&gt;% 
  group_by(value) %&gt;% 
  dplyr::count() %&gt;% ungroup %&gt;% 
  mutate(cum_sum = cumsum(n)) %&gt;% 
  ggplot(aes(x = value, y = cum_sum)) +
  geom_col() +
  geom_bar(stat = &#39;identity&#39;, aes(y = n, fill = value)) +
  scale_fill_viridis_d() +
  scale_x_discrete(labels = c(&#39;1&#39; = &#39;DmSP1&#39;, 
                              &#39;2&#39; = &#39;DmSP2&#39;,
                              &#39;3&#39; = &#39;DmSP3&#39;)) + 
  labs(y = &#39;No. identified proteins&#39;) +
  theme_bw() +
  theme(legend.position = &#39;none&#39;,
        axis.title.x = element_blank()) +
  #ggsave(filename = &#39;figures/cumulative_IDs.pdf&#39;, width = 3, height = 3) +
  NULL

cum_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-5-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="coefficient-of-variation" class="section level3">
<h3>Coefficient of variation</h3>
<p>For each experiment we calculate the coefficient of variation across replicates (log10 protein abundance).</p>
<pre class="r"><code>cv &lt;- function(x) sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE)

# get all experiments
allexp &lt;- apply(log10(DmSP3 %&gt;% dplyr::select(REP1.1:REP1.3, Halt1:NoHalt3, NaCl1:PBS4)), FUN = cv, 1)

CV_dat &lt;- DmSP3 %&gt;% 
  dplyr::select(FBgn, 
                REP1.1:REP1.3, 
                Halt1:NoHalt3,
                NaCl1:PBS4) %&gt;% 
  pivot_longer(cols = 2:18) %&gt;% 
  mutate(log_val = log10(value),
         experiment = case_when(grepl(&#39;REP&#39;, name) ~ &#39;Experiment 1&#39;,
                                grepl(&#39;Halt&#39;, name) ~ &#39;Experiment 2&#39;,
                                TRUE ~ &#39;Experiment 3&#39;),
         treatment = case_when(grepl(&#39;REP&#39;, name) ~ &#39;exp1&#39;,
                               grepl(&#39;^Halt&#39;, name) ~ &#39;exp2.1&#39;,
                               grepl(&#39;No&#39;, name) ~ &#39;exp2.2&#39;,
                               grepl(&#39;NaCl&#39;, name) ~ &#39;exp3.1&#39;,
                               TRUE ~ &#39;exp3.2&#39;))

# calculate medians
CV_dat %&gt;% 
  group_by(FBgn, experiment) %&gt;% 
  summarise(CV = cv(log_val)) %&gt;% 
  bind_rows(data.frame(FBgn = DmSP3$FBgn,
                       experiment = &#39;All&#39;,
                       CV = allexp)) %&gt;%
  group_by(experiment) %&gt;%
  summarise(N = n(),
            md = median(CV, na.rm = TRUE),
            sd = sd(CV, na.rm = TRUE))</code></pre>
<pre># A tibble: 4 × 4
  experiment       N     md     sd
  <chr>        <int>  <dbl>  <dbl>
1 All           3176 0.0641 0.0335
2 Experiment 1  3176 0.0183 0.0274
3 Experiment 2  3176 0.0541 0.0350
4 Experiment 3  3176 0.0326 0.0286
</pre>
<pre class="r"><code>CV_plot &lt;- CV_dat %&gt;% 
  group_by(FBgn, experiment) %&gt;% 
  summarise(CV = cv(log_val)) %&gt;% 
  ggplot(aes(x = experiment, y = CV, fill = experiment)) + 
  geom_boxplot(notch = TRUE) + 
  scale_fill_viridis_d(option = &#39;plasma&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;,
        axis.title.x = element_blank()) + 
  NULL

CV_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-6-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-20-most-abundant-proteins" class="section level3">
<h3>Top 20 most abundant proteins</h3>
<pre class="r"><code>DmSP3 %&gt;% 
  arrange(desc(grand.mean)) %&gt;% 
  head(20) %&gt;% #write_csv(&#39;output/Top20DmSP3.csv&#39;)
  mutate(NAME = if_else(NAME == &#39;-&#39;, ANNOTATION_SYMBOL, NAME)) %&gt;% 
  dplyr::select(FBgn, Name = NAME, `Ranked abundance (%)` = mean.perc) %&gt;% 
  mutate(across(3, ~round(.x, 1))) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-97c7c9d9f0f86d49b030" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-97c7c9d9f0f86d49b030">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0003884","FBgn0003889","FBgn0259795","FBgn0003885","FBgn0033868","FBgn0035915","FBgn0052064","FBgn0045770","FBgn0039071","FBgn0034132","FBgn0041102","FBgn0031545","FBgn0037862","FBgn0038373","FBgn0002865","FBgn0035240","FBgn0025111","FBgn0069354","FBgn0052351","FBgn0012036"],["alpha-Tubulin at 84B","beta-Tubulin at 85D","loopin-1","alpha-Tubulin at 84D","Sperm-Leucylaminopeptidase 7","Sperm-Leucylaminopeptidase 1","Sperm-Leucylaminopeptidase 4","Sperm-Leucylaminopeptidase 3","big bubble 8","Sperm-Leucylaminopeptidase 8","ocnus","CG3213","Mitochondrial aconitase 2","CG4546","Male-specific RNA 98Ca","CG33791","Adenine nucleotide translocase 2","Porin2","Sperm-Leucylaminopeptidase 2","Aldehyde dehydrogenase"],[100,100,99.9,99.9,99.8,99.8,99.7,99.7,99.6,99.6,99.5,99.5,99.5,99.4,99.4,99.3,99.3,99.2,99.2,99.1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>Ranked abundance (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="chromosomal-distribution" class="section level1 tabset">
<h1 class="tabset">Chromosomal distribution</h1>
<p>We retrieved chromosomal location of all genes in the genome from <a href="http://flybase.org/">FlyBase.org</a> (n = 13957) and summarised the total number of genes on each chromosome. We then counted the observed number of sperm genes (n = 3176) on each chromosome, and calculated the expected number based on the total number of sperm proteins identified. Finally, we calculated <span class="math inline">\(\chi^2\)</span> statistics for each chromosome and the associated p-values. We excluded the Y chromosome due to the small numbers of proteins. We used the Bejamini-Hochberg false discovery rate procedure to correct for multiple testing.</p>
<pre class="r"><code># Total number of genes on each chromosome - to work out &#39;expected&#39;
TotalGeneNumber &lt;- 
  # here I parsed the ~22k proteins from the Dmel uniprot proteome and submitted to FlyBase.org
  # I then remove any duplicate genes (i.e. some proteins have multiple isoforms)
  read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;) %&gt;% 
  dplyr::select(FBgn = X.SUBMITTED.ID, LOCATION_ARM) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  filter(LOCATION_ARM %in% c(&#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;, &#39;X&#39;, &#39;Y&#39;)) %&gt;% 
  group_by(LOCATION_ARM) %&gt;%
  summarise(N = n()) %&gt;% 
  mutate(pr.total.genes = N/sum(N))

# Number of sperm genes on each chromosome - &#39;observed&#39;
gene.no &lt;- DmSP3 %&gt;% 
  # replace DmSP3 with DmSP2 to compare results with previous studies
  # DmSP2 %&gt;% 
  # left_join(read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;), 
  #           by = c(&#39;FBgn&#39; = &#39;X.SUBMITTED.ID&#39;)) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  filter(LOCATION_ARM %in% c(&#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;, &#39;X&#39;, &#39;Y&#39;)) %&gt;%
  group_by(LOCATION_ARM) %&gt;% 
  summarise(obs.genes = n())

# Calculate observed and expected no. genes in each comparison on each chromosome to do X^2 test
chm_dist &lt;- gene.no %&gt;% 
  inner_join(TotalGeneNumber %&gt;% 
               dplyr::rename(all.genes = N)) %&gt;% 
  # remove Chm 4 and Y due to low numbers of genes present
  filter(LOCATION_ARM != &#39;Y&#39;) %&gt;% 
  # Calculate X^2 statistics
  mutate(exp.genes = round(n_distinct(DmSP3$FBgn) * pr.total.genes, 1), # expected no. genes 
         obs.exp = obs.genes/exp.genes, # observed / expected no. genes
         X2 = (obs.genes - exp.genes)^2/exp.genes, # calculate X^2 stat
         pval = 1 - (pchisq(X2, df = 1))) # get pvalue

# FDR corrected pval
chm_dist$FDR &lt;- p.adjust(chm_dist$pval, method = &#39;fdr&#39;) 

# plot chromosomal distribution
chm_plot &lt;- chm_dist %&gt;% 
  mutate(sigLabel = case_when(FDR &lt; 0.001 ~ &quot;***&quot;, 
                              FDR &lt; 0.01 &amp; FDR &gt; 0.001 ~ &quot;**&quot;,
                              FDR &lt; 0.05 &amp; FDR &gt; 0.01 ~ &quot;*&quot;,
                              TRUE ~ &#39;&#39;),
         Chromosome = fct_relevel(LOCATION_ARM, &#39;X&#39;, &#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;)) %&gt;% 
  mutate(chm_n = paste0(Chromosome, &#39;\n(&#39;, all.genes, &#39;)&#39;)) %&gt;% 
  ggplot(aes(x = Chromosome, y = obs.exp, fill = Chromosome)) +
  geom_histogram(stat = &#39;identity&#39;) +
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;, colour = &quot;black&quot;) +
  #scale_fill_viridis_d(direction = -1) +
  scale_fill_brewer(palette = &#39;Spectral&#39;) +
  labs(x = &quot;Chromosome&quot;, y = &quot;observed/expected\n no. genes&quot;) +
  theme_bw() +
  theme(legend.position = &#39;none&#39;,
        legend.text = element_text(size = 10),
        strip.text.y = element_text(face = &quot;italic&quot;)) +
  geom_text(aes(label = sigLabel), 
            size = 10, colour = &quot;black&quot;) +
  geom_text(aes(y = -0.05, label = paste0(obs.genes, &#39;/&#39;, exp.genes)), 
            size = 5, colour = &quot;black&quot;) +
  #ggsave(filename = &#39;figures/chm_dist.pdf&#39;, width = 4, height = 3) +
  NULL</code></pre>
<div id="plot" class="section level2">
<h2>Plot</h2>
<p>The X and 3R chromosomes have significantly fewer sperm genes than expected with a FDR cut-off &lt; 0.05.</p>
<pre class="r"><code>chm_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-9-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="y-linked-genes" class="section level2">
<h2>Y-linked genes</h2>
<p>We identified 9 Y chromosome genes. All above the DmSP3 average abundance:</p>
<pre class="r"><code>DmSP3 %&gt;%
  filter(LOCATION_ARM == &#39;Y&#39;) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  dplyr::select(FBgn, Name = NAME, `Ranked abundance (%)` = mean.perc) %&gt;% 
  arrange(desc(`Ranked abundance (%)`)) %&gt;% #write_csv(&#39;output/Ylinked_DmSP3.csv&#39;)
  kable(digits = 1) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
FBgn
</th>
<th style="text-align:left;">
Name
</th>
<th style="text-align:right;">
Ranked abundance (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FBgn0267433
</td>
<td style="text-align:left;">
male fertility factor kl5
</td>
<td style="text-align:right;">
98.8
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0267432
</td>
<td style="text-align:left;">
male fertility factor kl3
</td>
<td style="text-align:right;">
98.8
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0058064
</td>
<td style="text-align:left;">
Aldehyde reductase Y
</td>
<td style="text-align:right;">
95.5
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0001313
</td>
<td style="text-align:left;">
male fertility factor kl2
</td>
<td style="text-align:right;">
93.6
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0046323
</td>
<td style="text-align:left;">
Occludin-Related Y
</td>
<td style="text-align:right;">
92.9
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0267449
</td>
<td style="text-align:left;">
WD40 Y
</td>
<td style="text-align:right;">
86.4
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0267592
</td>
<td style="text-align:left;">
Coiled-Coils Y
</td>
<td style="text-align:right;">
86.0
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0046697
</td>
<td style="text-align:left;">
Ppr-Y
</td>
<td style="text-align:right;">
78.3
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0046698
</td>
<td style="text-align:left;">
Protein phosphatase 1, Y-linked 2
</td>
<td style="text-align:right;">
65.2
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="gene-age" class="section level1">
<h1>Gene age</h1>
<p>As with chromosomal distribution, we calculated the observed and expected number of genes in each age class (retrieved from <a href="http://gentree.ioz.ac.cn/index.php" class="uri">http://gentree.ioz.ac.cn/index.php</a> and recoded as in <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/evo.14297">Patlar et al. (2021). <em>Evolution</em></a>), calculated <span class="math inline">\(\chi^2\)</span> statistics and the associated p-values, and used the Bejamini-Hochberg false discovery rate procedure to correct for multiple testing.</p>
<pre class="r"><code>gene_age &lt;- read.csv(&#39;data/dm6_ver78_age.csv&#39;) %&gt;% 
  left_join(read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;), 
            by = c(&#39;FBgn&#39; = &#39;X.SUBMITTED.ID&#39;)) %&gt;% 
  distinct(FBtr, .keep_all = TRUE) %&gt;% 
  mutate(sp_prot = if_else(FBgn %in% DmSP3$FBgn, &#39;Sperm&#39;, &#39;Other&#39;),
         # recode age class
         gene_class = case_when(branch == 0 ~ &#39;ancient&#39;,
                                branch &lt;= 2 ~ &#39;sophophora&#39;,
                                branch &lt;= 3 ~ &#39;mel_group&#39;,
                                branch &lt;= 4 ~ &#39;mel_sub&#39;,
                                TRUE ~ &#39;recent&#39;),
         gene_class = fct_relevel(gene_class, 
                                  &#39;ancient&#39;, &#39;sophophora&#39;, &#39;mel_group&#39;, &#39;mel_sub&#39;, 
                                  &#39;recent&#39;))

#xtabs(~ gene_class + sp_prot, data = gene_age)

# Total number of genes in each age class - to work out &#39;expected&#39;
TotalGeneNumber.ageclass &lt;- gene_age %&gt;% 
  #filter(sp_prot == &#39;Sperm&#39;) %&gt;%
  group_by(gene_class) %&gt;%
  dplyr::count(name = &#39;N&#39;) %&gt;% 
  mutate(pr.total.genes = N/nrow(gene_age))

# Number of sperm genes in each age class - &#39;observed&#39;
gene.no.ageclass &lt;- gene_age %&gt;% 
  filter(sp_prot == &#39;Sperm&#39;) %&gt;%
  group_by(gene_class) %&gt;% 
  dplyr::count(name = &#39;obs.genes&#39;)

# Calculate observed and expected no. genes in each comparison on each chromosome to do X^2 test
age_dist &lt;- gene.no.ageclass %&gt;% 
  inner_join(TotalGeneNumber.ageclass %&gt;% 
               dplyr::rename(all.genes = N)) %&gt;% 
  # Calculate X^2 statistics
  mutate(exp.genes = round(nrow(gene_age %&gt;% filter(sp_prot == &#39;Sperm&#39;)) * 
                             pr.total.genes, 1), # expected no. genes 
         obs.exp = obs.genes/exp.genes, # observed / expected no. genes
         X2 = (obs.genes - exp.genes)^2/exp.genes, # calculate X^2 stat
         pval = 1 - (pchisq(X2, df = 1))) # get pvalue

# FDR corrected pval
age_dist$FDR &lt;- p.adjust(age_dist$pval, method = &#39;fdr&#39;) 

# plot chromosomal distribution
age_plot &lt;- age_dist %&gt;% 
  mutate(sigLabel = case_when(FDR &lt; 0.001 ~ &quot;***&quot;, 
                              FDR &lt; 0.01 &amp; FDR &gt; 0.001 ~ &quot;**&quot;,
                              FDR &lt; 0.05 &amp; FDR &gt; 0.01 ~ &quot;*&quot;,
                              TRUE ~ &#39;&#39;)) %&gt;% 
  mutate(age_n = paste0(gene_class, &#39;\n(&#39;, all.genes, &#39;)&#39;)) %&gt;% 
  ggplot(aes(x = gene_class, y = obs.exp, fill = gene_class)) +
  geom_histogram(stat = &#39;identity&#39;) +
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;, colour = &quot;black&quot;) +
  scale_fill_viridis_d() +
  scale_x_discrete(labels = c(&#39;ancient&#39; = &#39;Ancient&#39;, &#39;sophophora&#39; = &#39;Sophophora&#39;,
                              &#39;mel_group&#39; = expression(paste(italic(&#39;D. mel&#39;), &#39; gr.&#39;)),
                              &#39;mel_sub&#39; = expression(paste(italic(&#39;D. mel&#39;), &#39; subgr.&#39;)),
                              &#39;recent&#39; = &#39;Recent&#39;)) +
  labs(x = &quot;Age&quot;, y = &quot;observed/expected\n no. genes&quot;) +
  theme_bw() +
  theme(legend.position = &#39;none&#39;,
        legend.text = element_text(size = 10),
        strip.text.y = element_text(face = &quot;italic&quot;)) +
  geom_text(aes(label = sigLabel), 
            size = 10, colour = &quot;black&quot;) +
  geom_text(aes(y = -0.05, label = paste0(obs.genes, &#39;/&#39;, exp.genes)), 
            size = 5, colour = &quot;black&quot;) +
  #ggsave(filename = &#39;figures/age_dist.pdf&#39;, width = 4, height = 3) +
  NULL</code></pre>
<div id="plot-1" class="section level2">
<h2>Plot</h2>
<pre class="r"><code>age_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-12-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="newrecent-genes-in-the-dmsp3" class="section level2">
<h2>New/recent genes in the DmSP3</h2>
<pre class="r"><code>gene_age %&gt;%
  filter(gene_class == &#39;recent&#39; &amp; sp_prot == &#39;Sperm&#39;) %&gt;% #write_csv(&#39;output/Recent_DmSP3.csv&#39;)
  dplyr::select(FBgn, Name = NAME, Symbol = SYMBOL, Chromosome = LOCATION_ARM) %&gt;% 
  mutate(Name = if_else(Name == &#39;-&#39;, Symbol, Name)) %&gt;% 
  arrange(Symbol) %&gt;% 
  kable() %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
FBgn
</th>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Symbol
</th>
<th style="text-align:left;">
Chromosome
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FBgn0035571
</td>
<td style="text-align:left;">
CG12493
</td>
<td style="text-align:left;">
CG12493
</td>
<td style="text-align:left;">
3L
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0031935
</td>
<td style="text-align:left;">
CG13793
</td>
<td style="text-align:left;">
CG13793
</td>
<td style="text-align:left;">
2L
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0040028
</td>
<td style="text-align:left;">
CG17450
</td>
<td style="text-align:left;">
CG17450
</td>
<td style="text-align:left;">
X
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0032868
</td>
<td style="text-align:left;">
CG17472
</td>
<td style="text-align:left;">
CG17472
</td>
<td style="text-align:left;">
2L
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0030629
</td>
<td style="text-align:left;">
CG9123
</td>
<td style="text-align:left;">
CG9123
</td>
<td style="text-align:left;">
X
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0264077
</td>
<td style="text-align:left;">
Calnexin 14D
</td>
<td style="text-align:left;">
Cnx14D
</td>
<td style="text-align:left;">
X
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0260484
</td>
<td style="text-align:left;">
Hsc/Hsp70-interacting protein
</td>
<td style="text-align:left;">
HIP
</td>
<td style="text-align:left;">
X
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0052580
</td>
<td style="text-align:left;">
Mucin 14A
</td>
<td style="text-align:left;">
Muc14A
</td>
<td style="text-align:left;">
X
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0053105
</td>
<td style="text-align:left;">
p24-related-2
</td>
<td style="text-align:left;">
p24-2
</td>
<td style="text-align:left;">
3R
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0050382
</td>
<td style="text-align:left;">
Proteasome alpha1 subunit-related
</td>
<td style="text-align:left;">
Prosalpha1R
</td>
<td style="text-align:left;">
2R
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0013301
</td>
<td style="text-align:left;">
Protamine B
</td>
<td style="text-align:left;">
ProtB
</td>
<td style="text-align:left;">
2L
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0028986
</td>
<td style="text-align:left;">
Serpin 38F
</td>
<td style="text-align:left;">
Spn38F
</td>
<td style="text-align:left;">
2L
</td>
</tr>
<tr>
<td style="text-align:left;">
FBgn0260463
</td>
<td style="text-align:left;">
Uncoordinated 115b
</td>
<td style="text-align:left;">
Unc-115b
</td>
<td style="text-align:left;">
3R
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="omim" class="section level1 tabset">
<h1 class="tabset">OMIM</h1>
<p>We used the precomputed list of human disease orthologs from <a href="http://flybase.org/">FlyBase.org</a> to retrieve <a href="omim.org">OMIM</a> hits for proteins in the DmSP3.</p>
<pre class="r"><code>OMIM &lt;- read.csv(&#39;data/FlyBase/DmSP3_OMIM.csv&#39;, na.strings = c(&#39;&#39;)) %&gt;%
  dplyr::rename(FBgn = X..Dmel_gene_ID)

Hsap_hom &lt;- read.delim(&#39;data/FlyBase/FlyBase_Hsap_homologs.txt&#39;, na.strings = c(&#39;NA&#39;, &#39;&#39;, &#39;-&#39;)) %&gt;%
  dplyr::rename(FBgn = X.SUBMITTED.ID)

# # number of human homologs
# Hsap_hom %&gt;%
#   drop_na(H_SAPIENS_ORTHOLOGS) %&gt;%
#   distinct(FBgn) %&gt;%
#   count()

# # number of fly genes with more than one human homolog (disease vs not)
# OMIM %&gt;% 
#   mutate(OMIM = if_else(is.na(OMIM_Phenotype_IDs), &#39;No&#39;, &#39;Yes&#39;)) %&gt;% 
#   group_by(FBgn, OMIM) %&gt;% 
#   summarise(N = n_distinct(Human_gene_symbol)) %&gt;% 
#   group_by(OMIM, N) %&gt;% count %&gt;% 
#   mutate(N_genes = if_else(N &gt; 1, &#39;&gt; 1&#39;, &#39;1&#39;)) %&gt;% 
#   ggplot(aes(x = OMIM, y = n, fill = N_genes)) +
#   geom_col(position = &#39;fill&#39;) +
#   scale_fill_manual(values = cbPalette[2:1]) +
#   scale_y_continuous(labels = scales::percent) +
#   theme_bw() + 
#   #theme(legend.title = element_blank()) +
#   NULL

# # total number of homologs per Dmel gene
# Hsap_hom %&gt;% 
#   separate_rows(H_SAPIENS_ORTHOLOGS, sep = &#39; &lt;newline&gt; &#39;) %&gt;% 
#   mutate(homolog = if_else(is.na(H_SAPIENS_ORTHOLOGS) == TRUE, &#39;no&#39;, &#39;yes&#39;)) %&gt;% 
#   group_by(FBgn, homolog) %&gt;% count() %&gt;% 
#   mutate(N_genes = case_when(n &gt; 1 &amp; homolog == &#39;yes&#39; ~ &#39;&gt; 1&#39;, 
#                              n == 1 &amp; homolog == &#39;yes&#39; ~ &#39;1&#39;,
#                              TRUE ~ &#39;No&#39;)) %&gt;% 
#   group_by(N_genes) %&gt;% count()

# # number of genes with disease phenotype
# Hsap_hom %&gt;% 
#   mutate(omim = case_when(FBgn %in% OMIM$FBgn[is.na(OMIM$OMIM_Phenotype_IDs) == FALSE] ~ &#39;omim&#39;,
#                           TRUE ~ &#39;no&#39;)) %&gt;% 
#   group_by(omim) %&gt;% count()

# # number of human diseases per Dmel gene
# OMIM %&gt;% 
#   drop_na(OMIM_Phenotype_IDs) %&gt;% 
#   left_join(Hsap_hom, by = &#39;FBgn&#39;) %&gt;% 
#   group_by(FBgn) %&gt;% 
#   count() %&gt;% 
#   mutate(N_genes = if_else(n &gt; 1, &#39;&gt; 1&#39;, &#39;1&#39;)) %&gt;% 
#   group_by(N_genes) %&gt;% count()

omim_1 &lt;- data.frame(homolog = c(&#39;No&#39;, &#39;1&#39;, &#39;&gt; 1&#39;),
                     n = c(1974, 785, 417),
                     row = &#39;A&#39;) %&gt;% 
  ggplot(aes(x = n, y = row, fill = homolog)) +
  geom_col(position = &#39;fill&#39;) +
  scale_fill_manual(values = cbPalette[3:1],
                    name = &quot;Disease\nhomolog&quot;) +
  scale_x_continuous(labels = scales::percent) +
  theme_bw() + 
  theme(#legend.title = element_blank(),
        legend.position = &#39;bottom&#39;,
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank()) +
  annotate(&quot;text&quot;, x = c(0.3, 0.75, 0.93), y = 1, label = c(1974, 785, 417),
           size = 5) + 
  #ggsave(filename = &#39;figures/OMIM_no.pdf&#39;, width = 2, height = 4) +
  NULL


omim_no &lt;- OMIM %&gt;% 
  mutate(OMIM = if_else(is.na(OMIM_Phenotype_IDs), &#39;zNo&#39;, &#39;Yes&#39;)) %&gt;% 
  group_by(FBgn, OMIM) %&gt;%
  summarise(N = n_distinct(Human_gene_symbol)) %&gt;% 
  arrange(OMIM) %&gt;%
  group_by(FBgn) %&gt;% 
  slice(1) %&gt;%
  ungroup() %&gt;% 
  mutate(N_genes = case_when(N &gt; 1 &amp; OMIM == &#39;Yes&#39; ~ &#39;&gt; 1&#39;, 
                             N == 1 &amp; OMIM == &#39;Yes&#39; ~ &#39;1&#39;,
                             TRUE ~ &#39;No&#39;)) %&gt;% 
  group_by(OMIM, N_genes) %&gt;% 
  dplyr::count() %&gt;% 
  mutate(row = &#39;A&#39;) 

omim_1 &lt;- omim_no %&gt;% 
  ggplot(aes(x = n, y = row, fill = N_genes)) +
  geom_col(position = &#39;fill&#39;) +
  scale_fill_manual(values = cbPalette[3:1],
                    name = &quot;Disease\nhomolog&quot;) +
  scale_x_continuous(labels = scales::percent) +
  theme_bw() + 
  theme(#legend.title = element_blank(),
        legend.position = &#39;bottom&#39;,
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank()) +
  annotate(&quot;text&quot;, x = c(0.25, 0.66, 0.91), y = 1, label = rev(omim_no$n),
           size = 5) + 
  #ggsave(filename = &#39;figures/OMIM_no.pdf&#39;, width = 2, height = 4) +
  NULL

## all FBgns with associated disease phenotype
# OMIM %&gt;%
#   drop_na(OMIM_Phenotype_IDs) %&gt;%
#   distinct(FBgn, .keep_all = TRUE) #%&gt;% write_csv(&#39;output/OMIM_results/all_FBgns.csv&#39;)

# # % genes with no disease ortholog
# OMIM %&gt;%
#   drop_na(OMIM_Phenotype_IDs) %&gt;%
#   distinct(FBgn) %&gt;% dplyr::count() / n_distinct(OMIM$FBgn)

omim_plot &lt;- plot(euler(c(&#39;DmSP3&#39; = length(DmSP3$FBgn),
                          &#39;DmSP3&amp;Human\nhomologs&#39; = 2598,
                          &#39;DmSP3&amp;Human\nhomologs&amp;Disease\nhomologs&#39; = 1202)),
                  quantities = TRUE,
                  fills = list(fill = viridis::viridis(n = 3), alpha = .8))</code></pre>
<div id="number-of-omim-orthologs" class="section level2">
<h2>Number of OMIM orthologs</h2>
<pre class="r"><code>#pdf(&#39;figures/OMIM_orths.pdf&#39;, height = 4, width = 4)
omim_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-15-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-15-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="human-homologs" class="section level2">
<h2>Human homologs</h2>
<pre class="r"><code>#pdf(&#39;figures/OMIM_plot.pdf&#39;, height = 6, width = 4)
gridExtra::grid.arrange(omim_plot, omim_1, 
                        layout_matrix = rbind(c(1, 1),
                                              c(1, 1),
                                              c(2, 2)))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code># chrom location of disease orthologs
omim.no &lt;- OMIM %&gt;%
  drop_na(OMIM_Phenotype_IDs) %&gt;% 
  left_join(read.delim(&#39;data/FlyBase/uniprot2FlyBase_chrm.txt&#39;),
            by = c(&#39;FBgn&#39; = &#39;X.SUBMITTED.ID&#39;)) %&gt;%
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  filter(LOCATION_ARM %in% c(&#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;, &#39;X&#39;, &#39;Y&#39;)) %&gt;%
  group_by(LOCATION_ARM) %&gt;% 
  summarise(obs.genes = n())

# Calculate observed and expected no. genes to do X^2 test
omim_dist &lt;- omim.no %&gt;% 
  inner_join(TotalGeneNumber %&gt;% 
               dplyr::rename(all.genes = N)) %&gt;% 
  # remove Chm 4 and Y due to low numbers of genes present
  filter(LOCATION_ARM != &#39;Y&#39;) %&gt;% 
  # Calculate X^2 statistics
  mutate(exp.genes = round(1202 * pr.total.genes, 1), # expected no. genes 
         obs.exp = obs.genes/exp.genes, # observed / expected no. genes
         X2 = (obs.genes - exp.genes)^2/exp.genes, # calculate X^2 stat
         pval = 1 - (pchisq(X2, df = 1))) # get pvalue

# FDR corrected pval
omim_dist$FDR &lt;- p.adjust(omim_dist$pval, method = &#39;fdr&#39;)</code></pre>
</div>
<div id="chromosomal-distribution-table" class="section level2">
<h2>Chromosomal distribution table</h2>
<pre class="r"><code># table of results
omim_dist %&gt;% 
  dplyr::select(Chromosome = LOCATION_ARM, Observed = obs.genes, Expected = exp.genes,
                Chi2 = X2, FDR) %&gt;% 
  kable(digits = 2) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Chromosome
</th>
<th style="text-align:right;">
Observed
</th>
<th style="text-align:right;">
Expected
</th>
<th style="text-align:right;">
Chi2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
222
</td>
<td style="text-align:right;">
229.1
</td>
<td style="text-align:right;">
0.22
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
<tr>
<td style="text-align:left;">
2R
</td>
<td style="text-align:right;">
248
</td>
<td style="text-align:right;">
244.9
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.84
</td>
</tr>
<tr>
<td style="text-align:left;">
3L
</td>
<td style="text-align:right;">
227
</td>
<td style="text-align:right;">
235.7
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
<tr>
<td style="text-align:left;">
3R
</td>
<td style="text-align:right;">
305
</td>
<td style="text-align:right;">
295.3
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
6.9
</td>
<td style="text-align:right;">
3.77
</td>
<td style="text-align:right;">
0.31
</td>
</tr>
<tr>
<td style="text-align:left;">
X
</td>
<td style="text-align:right;">
181
</td>
<td style="text-align:right;">
188.1
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Parse Phenotype ID numbers and rank order
top_IDs &lt;- data.frame(ID = gsub(&#39;\\[.*&#39;, &#39;&#39;,
                                x = unlist(str_split(OMIM$OMIM_Phenotype_IDs.name.,
                                                     pattern = &#39;],&#39;))),
                      DESCRIPTION = gsub(&#39;.*\\[&#39;, &#39;&#39;,
                                         x = unlist(str_split(OMIM$OMIM_Phenotype_IDs.name.,
                                                              pattern = &#39;],&#39;)))) %&gt;%
  mutate(DESCRIPTION = gsub(&#39;]&#39;, &#39;&#39;, x = DESCRIPTION))

n_ids &lt;- top_IDs %&gt;%
  group_by(ID) %&gt;% dplyr::count() %&gt;%
  arrange(desc(n)) %&gt;%
  filter(ID != &#39;&#39;)

# n_ids %&gt;% group_by(n) %&gt;% dplyr::count() %&gt;%
#   arrange(-n) %&gt;%
#   mutate(n_genes = n * nn)

# top_IDs[grep(paste(unlist(n_ids[1:34, &#39;ID&#39;]), collapse=&quot;|&quot;), x = top_IDs$ID), ] %&gt;%
#   group_by(DESCRIPTION) %&gt;%
#   dplyr::count() %&gt;%
#   arrange(desc(n)) %&gt;% print(n = 34) #%&gt;% write_csv(&#39;output/OMIM_results/OMIM_tophits.csv&#39;)

# # grep top 17 IDs
# OMIM[grep(paste(c(top_IDs$ID[1:34]), collapse=&quot;|&quot;),
#           x = OMIM$OMIM_Phenotype_IDs.name.), ] %&gt;%
#   dplyr::select(-c(3:7)) %&gt;%
#   distinct(OMIM_Phenotype_IDs.name., .keep_all = TRUE)

## write tophits to files - 1 per phenotype
# for(i in 1:34) {
#
#   db = OMIM[grep(n_ids$ID[i], x = OMIM$OMIM_Phenotype_IDs.name.), ]
#
#   write_csv(db,
#             paste0(&#39;output/OMIM_results/&#39;,
#                    gsub(&#39; &#39;, &#39;_&#39;,
#                         str_replace_all(top_IDs$DESCRIPTION[i], &quot;[[:punct:]]&quot;, &quot; &quot;)),
#                    &#39;.csv&#39;))
#
# }

### Ribosomal hits
# OMIM %&gt;% 
#   drop_na(OMIM_Phenotype_IDs) %&gt;%
#   filter(FBgn %in% Dm_ribosomes$FBgn)</code></pre>
</div>
</div>
<div id="ribosomal-proteins" class="section level1">
<h1>Ribosomal proteins</h1>
<p>We downloaded the 169 <em>D. melanogaster</em> ribosomal proteins curated by <a href="https://flybase.org/reports/FBgg0000130">FlyBase.org</a> to compare the number, abundance, and distribution of ribosomal proteins found in the DmSP3. We also searched for other recent proteomics studies from other tissues or cell types in <em>D. melanogaster</em> and downloaded the supplementary materials containing the full lists of proteins to extract the ribosomal proteins identified in each study to compare to the DmSP3.</p>
<pre class="r"><code># rename the ribosomal data set and add data on presence in the sperm proteome and chromosomal location
all_rib &lt;- Dm_ribosomes %&gt;%
  # label genes based on presence in sperm proteome
  mutate(sp_prot = case_when(
    FBgn %in% c(DmSP3 %&gt;% filter(comb.peps == &#39;confident&#39;) %&gt;% pull(FBgn)) ~ &#39;Sperm.conf&#39;,
    FBgn %in% c(DmSP3 %&gt;% pull(FBgn)) ~ &#39;Sperm&#39;,
    TRUE ~ &#39;Other&#39;),
    # define ribosomal class (small/large; cytoplasmic/mitochondrial)
    CLASS = case_when(grepl(&#39;^RpL&#39;, x = SYMBOL) ~ &#39;CYT_LARGE&#39;,
                      grepl(&#39;^RpS&#39;, x = SYMBOL) ~ &#39;CYT_SMALL&#39;,
                      grepl(&#39;^mRpL&#39;, x = SYMBOL) ~ &#39;MIT_LARGE&#39;,
                      SYMBOL == &#39;sta&#39; ~ &#39;CYT_SMALL&#39;,
                      TRUE ~ &#39;MIT_SMALL&#39;),
    loc = str_sub(CLASS, start = 1, 1),
    size = str_sub(CLASS, start = 5, 5),
    paralog = str_remove(SYMBOL, &quot;[^0-9]+$&quot;))

# # write supp table
# DmSP3 %&gt;% filter(FBgn %in% all_rib$FBgn) %&gt;% write_csv(&#39;output/DmSP_ribosomes.csv&#39;)

# # number of ribosomal proteins found in sperm by type
# all_rib %&gt;% 
#   filter(sp_prot != &#39;Other&#39;) %&gt;% 
#   group_by(loc) %&gt;% count

#### Compare DmSP&#39;s
# upset(fromList(list(DmSP1 = intersect(DmSPI$FBgn, all_rib$FBgn),
#                     DmSP2 = intersect(DmSP2$FBgn, all_rib$FBgn),
#                     Current.study = intersect(DmSPIII.2$FBgn, all_rib$FBgn))))

# # Compare experiments
# upset(fromList(list(DmSP3 = all_rib %&gt;% filter(sp_prot != &#39;Other&#39;) %&gt;% pull(FBgn),
#                     PBSTd = all_rib %&gt;% filter(FBgn %in% PBST_dat$FBgn) %&gt;% pull(FBgn),
#                     SALTd = all_rib %&gt;% filter(FBgn %in% salt_dat$FBgn) %&gt;% pull(FBgn))))

## Load external data 
# Li et al. 2020 (Cell) - brain
li_brain &lt;- read_xlsx(&#39;data/Fly_Proteomes_LumosFusion/mmc2_Li_etal_2020.xlsx&#39;) %&gt;%
  dplyr::select(Accession = `UniProt Accession`, Species, UP = `Unique Peptides`) %&gt;%
  left_join(read.csv(&#39;data/Fly_Proteomes_LumosFusion/Li_etal_2020_uniprot2FBgn.csv&#39;)) %&gt;%
  filter(Species == &#39;DROME&#39;)

# Cao et al. 2020 (Cell Reports) - embryo
cao_embryo &lt;- read.csv(&#39;data/Fly_Proteomes_LumosFusion/mmc2_Cao_etal_2020.csv&#39;) %&gt;%
  dplyr::rename(FBgn = &#39;FlyBase.ID&#39;)

# McDonough-Goldstein et al. 2020 (Scientific Reports) - oocyte
mcdonough_oocyte &lt;- read_excel(&#39;data/Fly_Proteomes_LumosFusion/McDonough_etal_2021.xlsx&#39;) %&gt;%
  dplyr::select(FBgn = Description, UP = `Number of Unique Peptides`, starts_with(&#39;Abundance&#39;))

colnames(mcdonough_oocyte)[3:8] &lt;- paste0(rep(c(&#39;V&#39;, &#39;M&#39;), each = 3), 1:3)

# Hopes et al. 2021 (Nucleic Acids Research)
hopes &lt;- read.csv(&#39;data/Fly_Proteomes_LumosFusion/Hopes_etal_2021.csv&#39;) %&gt;% 
  dplyr::rename(Accession = &#39;Inf_Accession.Information_BestAccession&#39;) %&gt;% 
  left_join(read.csv(&#39;data/Fly_Proteomes_LumosFusion/Hopes_etal_2021_Accession2FBgn.csv&#39;), 
            by = &#39;Accession&#39;)

# # overlap between datasets
# upset(fromList(list(
#   DmSP3 = all_rib %&gt;% filter(sp_prot != &#39;Other&#39;) %&gt;% pull(FBgn),
#   Oocyte = all_rib %&gt;% filter(FBgn %in% mcdonough_oocyte$FBgn) %&gt;% pull(FBgn),
#   Brain = all_rib %&gt;% filter(FBgn %in% li_brain$FBgn) %&gt;% pull(FBgn),
#   Embryo = all_rib %&gt;% filter(FBgn %in% cao_embryo$FBgn) %&gt;% pull(FBgn),
#   Hopes = all_rib %&gt;% filter(FBgn %in% hopes$FBgn) %&gt;% pull(FBgn))))

plot(venn(c(#&#39;DmSP3&#39; = 168, &#39;Oocyte&#39; = 168, &#39;Brain&#39; = 168, &#39;Embryo&#39; = 168, 
             &#39;Hopes&#39; = 2,
             &#39;Hopes&amp;Embryo&#39; = 4, &#39;Embryo&amp;Oocyte&#39; = 3, &#39;Hopes&amp;Brain&#39; = 1, &#39;DmSP3&amp;Hopes&#39; = 1,
             &#39;Oocyte&amp;Hopes&#39; = 1, 
             &#39;Embryo&amp;Hopes&amp;Oocyte&#39; = 58, &#39;Embryo&amp;Oocyte&amp;DmSP3&#39; = 2, &#39;Embryo&amp;Hopes&amp;DmSP3&#39; = 2,
             &#39;Oocyte&amp;DmSP3&amp;Brain&#39; = 1, &#39;Embryo&amp;Oocyte&amp;Brain&#39; = 1, 
             &#39;Embryo&amp;Hopes&amp;Oocyte&amp;DmSP3&#39; = 14, &#39;Embryo&amp;Hopes&amp;Oocyte&amp;Brain&#39; = 13,
             &#39;Embryo&amp;Hopes&amp;DmSP3&amp;Brain&#39; = 5, &#39;Embryo&amp;Oocyte&amp;DmSP3&amp;Brain&#39; = 2, 
             &#39;Hopes&amp;Oocyte&amp;DmSP3&amp;Brain&#39; = 1, 
             &#39;Embryo&amp;Hopes&amp;Oocyte&amp;DmSP3&amp;Brain&#39; = 55)),
     quantities = TRUE)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-19-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># combine data
ribo_comb &lt;- list(
  DmSP3 = all_rib %&gt;% filter(sp_prot != &#39;Other&#39;) %&gt;% pull(FBgn),
  Oocyte = all_rib %&gt;% filter(FBgn %in% mcdonough_oocyte$FBgn) %&gt;% pull(FBgn),
  Brain = all_rib %&gt;% filter(FBgn %in% li_brain$FBgn) %&gt;% pull(FBgn),
  Embryo = all_rib %&gt;% filter(FBgn %in% cao_embryo$FBgn) %&gt;% pull(FBgn)) %&gt;%
  reshape2::melt() %&gt;%
  dplyr::select(FBgn = value, Proteome = L1) %&gt;%
  left_join(all_rib %&gt;% dplyr::select(FBgn, CLASS))

# perform Chi2 test
rib.test &lt;- ribo_comb %&gt;% 
  group_by(Proteome, CLASS) %&gt;% 
  count() %&gt;%
  left_join(all_rib %&gt;%
              group_by(CLASS) %&gt;% count(name = &#39;All&#39;)) %&gt;%
  mutate(obs.exp = n/All, # observed / expected no. genes
         X2 = (n - All)^2/All, # calculate X^2 stat
         pval = 1 - (pchisq(X2, df = 1))) # get pvalue

# FDR corrected pval
rib.test$FDR &lt;- p.adjust(rib.test$pval, method = &#39;fdr&#39;)

rib_plot &lt;- rib.test %&gt;%
  mutate(sigLabel = case_when(FDR &lt; 0.001 ~ &quot;***&quot;,
                              FDR &lt; 0.01 &amp; FDR &gt; 0.001 ~ &quot;**&quot;,
                              FDR &lt; 0.05 &amp; FDR &gt; 0.01 ~ &quot;*&quot;,
                              TRUE ~ &#39;&#39;),
         Proteome = fct_relevel(Proteome, &#39;embryo&#39;, &#39;oocyte&#39;, &#39;brain&#39;, &#39;sperm&#39;)) %&gt;%
  ggplot(aes(x = CLASS, y = n, fill = Proteome)) +
  geom_col(position = &#39;dodge&#39;) +
  scale_fill_brewer(palette = &#39;RdBu&#39;, direction = -1) +
  scale_x_discrete(labels = c(&#39;CYT_LARGE&#39; = &#39;Cyt. large&#39;, &#39;CYT_SMALL&#39; = &#39;Cyt. small&#39;,
                              &#39;MIT_LARGE&#39; = &#39;Mt. large&#39;,  &#39;MIT_SMALL&#39; = &#39;Mt. small&#39;)) +
  labs(x = &quot;Class&quot;, y = &quot;obs. no. genes&quot;) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;,
        legend.title = element_blank()) +
  geom_text(aes(label = sigLabel),
            size = 5, colour = &quot;black&quot;, position = position_dodge(width = .9)) +
  geom_segment(aes(x = 0.6, y = 54, xend = 1.4, yend = 54), lty = 2) +
  geom_segment(aes(x = 1.6, y = 38, xend = 2.4, yend = 38), lty = 2) +
  geom_segment(aes(x = 2.6, y = 47, xend = 3.4, yend = 47), lty = 2) +
  geom_segment(aes(x = 3.6, y = 29, xend = 4.4, yend = 29), lty = 2) +
  #ggsave(filename = &#39;figures/ribo_comp.pdf&#39;, width = 4, height = 3) +
  NULL</code></pre>
<div id="compare-tissuecell-types" class="section level2">
<h2>Compare tissue/cell types</h2>
<p>We calculated <span class="math inline">\(\chi^2\)</span> statistics with the null expectation that each tissue or cell type would have complete representation of all 168 ribosomal proteins. We also tested whether the overlap between ribosomal proteins in brain and sperm was greater than expected using Fisher’s exact test.</p>
<pre class="r"><code>## overlap
# upset(fromList(list(Ribosomes = all_rib %&gt;% pull(FBgn),
#                     DmSP3 = all_rib %&gt;% filter(sp_prot != &#39;Other&#39;) %&gt;% pull(FBgn),
#                     Brain = all_rib %&gt;% filter(FBgn %in% li_brain$FBgn) %&gt;% pull(FBgn))))

rib_br_sp &lt;- plot(euler(c(&#39;Ribosomes&#39; = 169,
                          &#39;Ribosomes&amp;Brain&#39; = 15,
                          &#39;Ribosomes&amp;DmSP3&#39; = 19,
                          &#39;Ribosomes&amp;Brain&amp;DmSP3&#39; = 64)),
                  quantities = TRUE,
                  fills = list(fill = c(NA, RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 4)[4:3]), 
                               alpha = 1))

gridExtra::grid.arrange(rib_plot, rib_br_sp, nrow = 1)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-20-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># # test overlap between brain and testes
# broom::tidy(fisher.test(matrix(c(169, 19, 15, 63), nrow = 2), alternative = &quot;greater&quot;))</code></pre>
</div>
<div id="paralog-switching" class="section level2">
<h2>Paralog switching</h2>
<p>There are 93 cytoplasmic ribosomal proteins on FlyBase.org, including 13 paralogs (for 80 per protein).</p>
<pre class="r"><code>paralogs &lt;- read.csv(&#39;data/FlyBase/FlyBase_Ribosome_Paralogs.csv&#39;) %&gt;% 
  dplyr::select(-c(Location, Strand, Paralog_Location:DIOPT_score))

p_genes &lt;- paralogs %&gt;% 
  filter(FBgn %in% all_rib$FBgn, !str_detect(GeneSymbol, &#39;^m&#39;)) %&gt;% 
  filter(Paralog_FBgn_ID %in% all_rib$FBgn, !str_detect(Paralog_GeneSymbol, &#39;^m&#39;)) %&gt;% 
  mutate(base_gene = str_remove(GeneSymbol, &quot;[^0-9]+$&quot;),
         para_gene = str_remove(Paralog_GeneSymbol, &quot;[^0-9]+$&quot;)) %&gt;% 
  filter(base_gene == para_gene) %&gt;% 
  #distinct(GeneSymbol, .keep_all = TRUE) %&gt;% 
  mutate(ps = paste(pmin(GeneSymbol, Paralog_GeneSymbol), pmax(GeneSymbol, Paralog_GeneSymbol))) %&gt;% 
  distinct(ps, .keep_all = TRUE) %&gt;% 
  distinct(base_gene, .keep_all = TRUE)

# # number of ribosomal proteins (cytoplasmic) in sperm proteome
# all_rib %&gt;% 
#   filter(loc == &#39;C&#39;) %&gt;% 
#   group_by(sp_prot) %&gt;% count()

paralog_switches &lt;- data.frame(FBgn = c(p_genes$FBgn, p_genes$Paralog_FBgn_ID),
                               SYMBOL = c(p_genes$GeneSymbol, p_genes$Paralog_GeneSymbol))
# abundance of each paralog
rb_paralogs &lt;- data.frame(FBgn = paralog_switches$FBgn,
                          SYMBOL = c(p_genes$GeneSymbol, p_genes$Paralog_GeneSymbol)) %&gt;% 
  mutate(base_gene = str_remove(SYMBOL, &quot;[^0-9]+$&quot;)) %&gt;% 
  left_join(DmSP3 %&gt;% dplyr::select(FBgn, REP1.1:REP1.3, Halt1:NoHalt3, NaCl1:PBS4), 
            by = &#39;FBgn&#39;) %&gt;% 
  pivot_longer(cols = REP1.1:PBS4) %&gt;% 
  mutate(log_val = log10(value + 1),
         experiment = case_when(grepl(&#39;REP&#39;, name) ~ &#39;Experiment 1&#39;,
                                grepl(&#39;Halt&#39;, name) ~ &#39;Experiment 2&#39;,
                                TRUE ~ &#39;Experiment 3&#39;)) %&gt;% 
  group_by(FBgn, SYMBOL, base_gene) %&gt;% 
  summarise(N = n(),
            mn = mean(log_val, na.rm = TRUE),
            se = sd(log_val, na.rm = TRUE)/sqrt(N)) %&gt;% 
  drop_na(mn)

# Hopes data
hopes2 &lt;- hopes %&gt;% 
  select(Accession, FBgn, NAME = TMT1_Primary.Gene.names, contains(&#39;Log2&#39;), -contains(&#39;Scaled&#39;))

colnames(hopes2) &lt;- gsub(&#39;Quantification_Log2.&#39;, &#39;&#39;, colnames(hopes2))
colnames(hopes2) &lt;- gsub(&#39;Normalised.Abundances...F...&#39;, &#39;&#39;, colnames(hopes2))
colnames(hopes2) &lt;- gsub(&#39;..Sample..&#39;, &#39;_&#39;, colnames(hopes2))

hopes_ribosomes &lt;- hopes2 %&gt;% 
  filter(FBgn %in% all_rib$FBgn) %&gt;% 
  pivot_longer(cols = 4:19) %&gt;% 
  mutate(name = gsub(&#39;.fly.tissue&#39;, &#39;&#39;, name),
         name = gsub(&#39;\\.&#39;, &#39;_&#39;, name),
         base_gene = str_remove(NAME, &quot;[^0-9]+$&quot;)) %&gt;% 
  separate(name, into = c(&#39;experiment&#39;, &#39;reporter&#39;, &#39;tissue&#39;, &#39;rib&#39;), sep = &#39;_&#39;)

# # Hopes abundances
# hopes_ribosomes %&gt;% 
#   filter(FBgn %in% paralog_switches,
#          tissue == &#39;Testes&#39; | tissue == &#39;Head&#39; | tissue == &#39;Ovaries&#39; | tissue == &#39;Embryo&#39;) %&gt;% 
#   ggplot(aes(x = NAME, y = value, fill = tissue)) +
#   geom_col(position = &#39;dodge&#39;) +
#   facet_wrap(~base_gene, scales = &#39;free_x&#39;, nrow = 1) + 
#   NULL

# # Ovary vs. Testes abundance (see Fig. 2b in hopes et al.)
# hopes_ribosomes %&gt;% 
#   filter(FBgn %in% paralog_switches,
#          tissue == &#39;Testes&#39; | tissue == &#39;Ovaries&#39;) %&gt;% 
#   group_by(NAME, tissue, base_gene) %&gt;% 
#   summarise(mn = mean(value, na.rm = TRUE)) %&gt;% 
#   group_by(NAME, base_gene) %&gt;% 
#   summarise(diff = mn[1] - mn[2]) %&gt;% 
#   ggplot(aes(x = NAME, y = diff)) +
#   geom_col(position = &#39;dodge&#39;) +
#   facet_wrap(~base_gene, scales = &#39;free_x&#39;, nrow = 1) + 
#   NULL


# # same paralog as Hopes?
# hopes_ribosomes %&gt;% 
#   filter(FBgn %in% paralog_switches, tissue == &#39;Testes&#39;) %&gt;%
#   bind_rows(rb_paralogs %&gt;% mutate(tissue = &#39;sperm&#39;) %&gt;% 
#               select(NAME = SYMBOL, base_gene, tissue, value = mn)) %&gt;% 
#   ggplot(aes(x = NAME, y = value, fill = tissue)) +
#   geom_col(position = &#39;dodge&#39;) +
#   facet_wrap(~base_gene, scales = &#39;free_x&#39;, nrow = 1) + 
#   NULL

# # RpL22 means per tissue/DmSP3
# hopes_ribosomes %&gt;% 
#   mutate(tissue = str_sub(tissue, 1, 4)) %&gt;% 
#   filter(base_gene == &#39;RpL22&#39;) %&gt;% 
#   group_by(NAME, tissue) %&gt;% 
#   summarise(N = n(),
#             mn = mean(value, na.rm = TRUE),
#             se = sd(value, na.rm = TRUE)/sqrt(N)) %&gt;% 
#   bind_rows(rb_paralogs %&gt;% mutate(tissue = &#39;Sperm&#39;) %&gt;% 
#               select(NAME = SYMBOL, base_gene, tissue, mn, se) %&gt;% 
#               filter(base_gene == &#39;RpL22&#39;)) %&gt;% 
#   ggplot(aes(x = NAME, y = mn, fill = tissue)) +
#   geom_col(position = &#39;dodge&#39;) +
#   geom_errorbar(aes(ymin = mn - se, ymax = mn + se), width = .2) +
#   facet_wrap(~tissue, scales = &#39;free_x&#39;, nrow = 1,
#              labeller = as_labeller(
#                c(Embr = &#39;Embryo&#39;, Head = &#39;Head&#39;, 
#                  Ovar = &#39;Ovaries&#39;, Sperm = &#39;Sperm&#39;, Test = &#39;Testes&#39;))) + 
#   theme_bw() + 
#   theme(legend.position = &#39;&#39;) +
#   NULL

# Table
paralog_switches %&gt;% 
  left_join(
    hopes_ribosomes %&gt;% 
      filter(FBgn %in% paralog_switches$FBgn) %&gt;% 
      mutate(tissue = str_sub(tissue, 1, 4)) %&gt;% 
      group_by(FBgn, NAME, tissue) %&gt;% 
      summarise(N = n(),
                mn = mean(value, na.rm = TRUE),
                se = sd(value, na.rm = TRUE)/sqrt(N)) %&gt;% ungroup() %&gt;% 
      bind_rows(rb_paralogs %&gt;% mutate(tissue = &#39;Sperm&#39;) %&gt;% 
                  select(NAME = SYMBOL, tissue, mn, se)) %&gt;% 
      dplyr::select(FBgn, NAME, tissue, mn) %&gt;% 
      pivot_wider(names_from = tissue, values_from = mn)) %&gt;% 
  mutate(base_gene = str_remove(SYMBOL, &quot;[^0-9]+$&quot;)) %&gt;% 
  arrange(base_gene) %&gt;% 
  dplyr::select(-base_gene, -NAME) %&gt;% 
  #write_csv(&#39;output/paralog_switches.csv&#39;) %&gt;% 
  kable(digits = 1, 
        caption = &#39;Abundance of each paralog&#39;) %&gt;% 
  kable_styling(full_width = FALSE) %&gt;%
  group_rows(&quot;RpL10&quot;, 1, 2) %&gt;%
  group_rows(&quot;RpL22&quot;, 3, 4) %&gt;%
  group_rows(&quot;RpL24&quot;, 5, 6) %&gt;%
  group_rows(&quot;RpL34&quot;, 7, 8) %&gt;% 
  group_rows(&quot;RpL37&quot;, 9, 10) %&gt;% 
  group_rows(&quot;RpL7&quot;, 11, 12) %&gt;% 
  group_rows(&quot;RpLP0&quot;, 13, 14) %&gt;% 
  group_rows(&quot;RpS10&quot;, 15, 16) %&gt;% 
  group_rows(&quot;RpS14&quot;, 17, 18) %&gt;% 
  group_rows(&quot;RpS15&quot;, 19, 20) %&gt;% 
  group_rows(&quot;RpS19&quot;, 21, 22) %&gt;% 
  group_rows(&quot;RpS28&quot;, 23, 24) %&gt;% 
  group_rows(&quot;RpS5&quot;, 25, 26)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Abundance of each paralog
</caption>
<thead>
<tr>
<th style="text-align:left;">
FBgn
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:right;">
Embr
</th>
<th style="text-align:right;">
Head
</th>
<th style="text-align:right;">
Ovar
</th>
<th style="text-align:right;">
Test
</th>
<th style="text-align:right;">
Sperm
</th>
</tr>
</thead>
<tbody>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0036213
</td>
<td style="text-align:left;">
RpL10Ab
</td>
<td style="text-align:right;">
14.1
</td>
<td style="text-align:right;">
13.8
</td>
<td style="text-align:right;">
14.4
</td>
<td style="text-align:right;">
13.7
</td>
<td style="text-align:right;">
6.5
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0038281
</td>
<td style="text-align:left;">
RpL10Aa
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL22</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0015288
</td>
<td style="text-align:left;">
RpL22
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
13.7
</td>
<td style="text-align:right;">
14.2
</td>
<td style="text-align:right;">
13.2
</td>
<td style="text-align:right;">
6.8
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0034837
</td>
<td style="text-align:left;">
RpL22-like
</td>
<td style="text-align:right;">
10.1
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
10.1
</td>
<td style="text-align:right;">
14.5
</td>
<td style="text-align:right;">
0.8
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL24</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0032518
</td>
<td style="text-align:left;">
RpL24
</td>
<td style="text-align:right;">
13.3
</td>
<td style="text-align:right;">
13.1
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
6.2
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0037899
</td>
<td style="text-align:left;">
RpL24-like
</td>
<td style="text-align:right;">
7.9
</td>
<td style="text-align:right;">
8.0
</td>
<td style="text-align:right;">
10.3
</td>
<td style="text-align:right;">
8.3
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL34</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0037686
</td>
<td style="text-align:left;">
RpL34b
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
12.6
</td>
<td style="text-align:right;">
13.1
</td>
<td style="text-align:right;">
12.7
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0039406
</td>
<td style="text-align:left;">
RpL34a
</td>
<td style="text-align:right;">
9.7
</td>
<td style="text-align:right;">
8.9
</td>
<td style="text-align:right;">
9.6
</td>
<td style="text-align:right;">
9.3
</td>
<td style="text-align:right;">
6.5
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL37</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0030616
</td>
<td style="text-align:left;">
RpL37a
</td>
<td style="text-align:right;">
12.4
</td>
<td style="text-align:right;">
11.8
</td>
<td style="text-align:right;">
12.4
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
5.5
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0034822
</td>
<td style="text-align:left;">
RpL37b
</td>
<td style="text-align:right;">
5.2
</td>
<td style="text-align:right;">
5.3
</td>
<td style="text-align:right;">
5.0
</td>
<td style="text-align:right;">
10.6
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpL7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0005593
</td>
<td style="text-align:left;">
RpL7
</td>
<td style="text-align:right;">
13.8
</td>
<td style="text-align:right;">
13.3
</td>
<td style="text-align:right;">
13.9
</td>
<td style="text-align:right;">
13.4
</td>
<td style="text-align:right;">
6.8
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0032404
</td>
<td style="text-align:left;">
RpL7-like
</td>
<td style="text-align:right;">
9.3
</td>
<td style="text-align:right;">
9.1
</td>
<td style="text-align:right;">
11.1
</td>
<td style="text-align:right;">
9.1
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpLP0</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0000100
</td>
<td style="text-align:left;">
RpLP0
</td>
<td style="text-align:right;">
14.8
</td>
<td style="text-align:right;">
14.1
</td>
<td style="text-align:right;">
14.9
</td>
<td style="text-align:right;">
14.3
</td>
<td style="text-align:right;">
7.1
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0033485
</td>
<td style="text-align:left;">
RpLP0-like
</td>
<td style="text-align:right;">
10.8
</td>
<td style="text-align:right;">
9.4
</td>
<td style="text-align:right;">
11.2
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0027494
</td>
<td style="text-align:left;">
RpS10a
</td>
<td style="text-align:right;">
8.4
</td>
<td style="text-align:right;">
7.3
</td>
<td style="text-align:right;">
8.4
</td>
<td style="text-align:right;">
11.7
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0285947
</td>
<td style="text-align:left;">
RpS10b
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
13.0
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
13.2
</td>
<td style="text-align:right;">
6.3
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS14</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0004403
</td>
<td style="text-align:left;">
RpS14a
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0004404
</td>
<td style="text-align:left;">
RpS14b
</td>
<td style="text-align:right;">
12.6
</td>
<td style="text-align:right;">
12.4
</td>
<td style="text-align:right;">
12.7
</td>
<td style="text-align:right;">
12.2
</td>
<td style="text-align:right;">
6.2
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS15</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0010198
</td>
<td style="text-align:left;">
RpS15Aa
</td>
<td style="text-align:right;">
13.8
</td>
<td style="text-align:right;">
13.4
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
13.4
</td>
<td style="text-align:right;">
6.6
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0033555
</td>
<td style="text-align:left;">
RpS15Ab
</td>
<td style="text-align:right;">
8.3
</td>
<td style="text-align:right;">
8.7
</td>
<td style="text-align:right;">
8.6
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS19</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0010412
</td>
<td style="text-align:left;">
RpS19a
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
14.1
</td>
<td style="text-align:right;">
14.4
</td>
<td style="text-align:right;">
13.5
</td>
<td style="text-align:right;">
7.1
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0039129
</td>
<td style="text-align:left;">
RpS19b
</td>
<td style="text-align:right;">
9.7
</td>
<td style="text-align:right;">
9.2
</td>
<td style="text-align:right;">
9.6
</td>
<td style="text-align:right;">
14.3
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS28</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0030136
</td>
<td style="text-align:left;">
RpS28b
</td>
<td style="text-align:right;">
11.6
</td>
<td style="text-align:right;">
10.9
</td>
<td style="text-align:right;">
11.3
</td>
<td style="text-align:right;">
10.8
</td>
<td style="text-align:right;">
3.7
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0039739
</td>
<td style="text-align:left;">
RpS28a
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
4.0
</td>
<td style="text-align:right;">
3.8
</td>
<td style="text-align:right;">
6.5
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr grouplength="2">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>RpS5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0002590
</td>
<td style="text-align:left;">
RpS5a
</td>
<td style="text-align:right;">
10.6
</td>
<td style="text-align:right;">
12.1
</td>
<td style="text-align:right;">
11.2
</td>
<td style="text-align:right;">
11.5
</td>
<td style="text-align:right;">
6.1
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
FBgn0038277
</td>
<td style="text-align:left;">
RpS5b
</td>
<td style="text-align:right;">
13.8
</td>
<td style="text-align:right;">
11.9
</td>
<td style="text-align:right;">
13.6
</td>
<td style="text-align:right;">
12.7
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<div id="abundance-of-proteins-in-the-dmsp3" class="section level2 tabset">
<h2 class="tabset">Abundance of proteins in the DmSP3</h2>
<p>We tested for differences in abundance between proteins on the sex vs. autosomes, and those identified as ribosomal proteins, or putative Sfps, compared the the remaining sperm proteome using the Kruskal-Wallace rank sum test followed by pairwise Wilcoxon rank sum tests corrected for multiple testing using the Benjamini-Hochberg procedure. We used the grand mean abundance and used the filtered dataset where a protein must be identified by 2 or more unique peptides or in 2 or more biological replicates</p>
<pre class="r"><code>comb_abun &lt;- DmSP_comb %&gt;% 
  drop_na(grand.mean) %&gt;% 
  dplyr::select(FBgn, NAME, SYMBOL, LOCATION_ARM, Sfp, grand.mean) %&gt;% 
  mutate(ribosome = if_else(FBgn %in% all_rib$FBgn, &#39;Ribosome&#39;, &#39;Other&#39;),
        # relabel chromosomes
         Chromosome = case_when(
           LOCATION_ARM %in% c(&#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;, &#39;X&#39;, &#39;Y&#39;) ~ LOCATION_ARM,
           LOCATION_ARM == &#39;MT; dmel_mitochondrion_genome&#39; ~ &#39;mt&#39;,
           TRUE ~ &#39;other&#39;),
         # label X/Y or autosomal
         Chm = case_when(Chromosome %in% c(&#39;X&#39;, &#39;Y&#39;, &#39;mt&#39;) ~ Chromosome,
                         Chromosome %in% c(&#39;2L&#39;, &#39;2R&#39;, &#39;3L&#39;, &#39;3R&#39;, &#39;4&#39;) ~ &#39;A&#39;,
                         TRUE ~ &#39;other&#39;))</code></pre>
<div id="ribosomes" class="section level3">
<h3>Ribosomes</h3>
<pre class="r"><code># KW test
rib_kw &lt;- broom::tidy(kruskal.test(grand.mean ~ as.factor(ribosome), data = comb_abun))</code></pre>
<p>We compared the abundance of 70 ribosomal proteins to the remaining 1918 other sperm proteins. There was no significant difference in the abundance of ribosomal proteins compared to the remaining sperm proteins (Kruskal-Wallis rank sum test, <span class="math inline">\(\chi^2\)</span> = 0.063, p = 0.803).</p>
<pre class="r"><code>ribo_abun &lt;- comb_abun %&gt;% 
  ggplot(aes(x = ribosome, y = log2(grand.mean), fill = ribosome)) +
  stat_halfeye() +
  gghalves::geom_half_point(aes(colour = ribosome), size = .5, alpha = .35, side = &quot;l&quot;, range_scale = 0.5) +
  scale_fill_manual(values = c(cbPalette[2], 
                                 RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 4)[4])) + 
  scale_colour_manual(values = c(cbPalette[2], 
                                 RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 4)[4])) +
  labs(y = &#39;log2(abundance)&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;,
        axis.title.x = element_blank()) + 
  geom_signif(y_position = 37, xmin = 1, xmax = 2, annotation = &#39;n.s.&#39;, tip_length = 0.01) +
  # ggrepel::geom_label_repel(
  #   data = comb_abun %&gt;% filter(ribosome == &#39;Ribosome&#39;),
  #   aes(label = SYMBOL),
  #   size = 5, colour = &#39;black&#39;, fill = &#39;white&#39;,
  #   box.padding = unit(0.35, &quot;lines&quot;),
  #   point.padding = unit(0.3, &quot;lines&quot;),
  #   max.overlaps = 100
  # ) +
  #ggsave(&#39;figures/ribo_abun.pdf&#39;, height = 4, width = 4) +
  NULL

ribo_abun</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-24-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sex-vs.-autosomes" class="section level3">
<h3>Sex vs. autosomes</h3>
<pre class="r"><code># subset only proteins coded on sex chromosomes or autosomes
axy &lt;- comb_abun %&gt;% filter(Chm %in% c(&#39;X&#39;, &#39;Y&#39;, &#39;A&#39;))

# KW test
sex_kw &lt;- kruskal.test(grand.mean ~ as.factor(Chm), data = axy)
sex_wilcox &lt;- broom::tidy(pairwise.wilcox.test(axy$grand.mean, axy$Chm, p.adjust.method = &quot;BH&quot;))</code></pre>
<p>There was a significant difference in the abundance of proteins encoded on the X- or Y- chromosome and autosomes (Kruskal-Wallis rank sum test, <span class="math inline">\(\chi^2\)</span> = 19, p &lt; 0.001).</p>
<pre class="r"><code>axy_plot &lt;- axy %&gt;% 
  mutate(Chm = fct_relevel(Chm, &#39;X&#39;, &#39;A&#39;, &#39;Y&#39;)) %&gt;% 
  ggplot(aes(x = Chm, y = log2(grand.mean), fill = Chm)) +
  geom_violin(alpha = .5) +
  geom_jitter(data = axy %&gt;% filter(Chm == &#39;X&#39;), 
              colour = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[1], 
              fill = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[1], width = .3) + 
  geom_jitter(data = axy %&gt;% filter(Chm == &#39;A&#39;), 
              colour = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[1], 
              fill = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[1], width = .3, alpha = 0) + 
  geom_jitter(data = axy %&gt;% filter(Chm == &#39;Y&#39;), 
              colour = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[3], 
              fill = RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[3], width = .3) + 
  geom_boxplot(alpha = .8, notch = TRUE, width = .3, outlier.shape = NA) +
  scale_colour_manual(values = c(RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[1],
                                 NA,
                                 RColorBrewer::brewer.pal(name = &#39;RdBu&#39;, n = 3)[3])) +
  scale_fill_brewer(palette = &#39;RdBu&#39;) +
  labs(x = &#39;Chromosome&#39;, y = &#39;log2(abundance)&#39;) +
  theme_bw() +
  theme(legend.position = &#39;none&#39;) +
  geom_signif(y_position = 37, xmin = 1, xmax = 3, annotation = &#39;***&#39;, tip_length = 0.01) +
  geom_signif(y_position = 36, xmin = c(1, 2), xmax = c(2, 3), annotation = c(&#39;*&#39;, &#39;***&#39;), tip_length = 0.01) +
  #ggsave(&#39;figures/chm_abundance.pdf&#39;, height = 4, width = 4) +
  NULL

axy_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-26-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="seminal-fluid-proteins-in-the-dmsp3" class="section level1">
<h1>Seminal fluid proteins in the DmSP3</h1>
<p>We identified a considerable number of putative seminal fluid proteins (Sfps) in the DmSP3.</p>
<div id="overlap-between-the-dmsp3-n-3176-and-putative-sfps-described-in-wigby-et-al.-2020-as-high-confidence-sfps-or-low-confidencetransferred-sfps." class="section level4">
<h4>Overlap between the DmSP3 (n = 3176) and putative Sfps described in Wigby et al. (2020) as ‘high confidence’ Sfps, or ‘low confidence/transferred Sfps’.</h4>
<pre class="r"><code># upset(fromList(list(Sfps = wigbySFP$FBgn,
#                     Sfps.conf = wigbySFP$FBgn[wigbySFP$category == &#39;highconf&#39;],
#                     DmSP3 = DmSP3$FBgn)))

# Eulerr diagram
sfp_euler &lt;- plot(euler(c(&#39;Sfps&#39; = 161, &quot;DmSP3&quot; = 2899, 
                          &#39;Sfps&amp;Sfps.conf&#39; = 170, 
                          &#39;Sfps&amp;DmSP3&#39; = 156, 
                          &#39;Sfps&amp;Sfps.conf&amp;DmSP3&#39; = 122)),
                  quantities = TRUE,
                  fills = list(fill = viridis::viridis(n = 3)[c(2, 1, 3)], alpha = .5))

#pdf(&#39;figures/Sfp_sperm_overlap.pdf&#39;, height = 4, width = 4)
sfp_euler</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-27-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-27-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code># # Kruskal-Wallis test
sfp_kw &lt;- broom::tidy(kruskal.test(grand.mean ~ as.factor(Sfp), data = comb_abun))</code></pre>
<p>We found no significant difference in the abundance between ‘high confidence’ Sfps, ‘low confidence/transferred’ Sfps, or other proteins found in the DmSP3 (Kruskal-Wallis rank sum test, <span class="math inline">\(\chi^2\)</span> = 4.277, p = 0.118).</p>
<pre class="r"><code># sfp vs. other
sfp_abun &lt;- comb_abun %&gt;% 
  ggplot(aes(x = Sfp, y = log2(grand.mean), fill = Sfp)) +
  geom_violin(alpha = .5) +
  geom_jitter(data = comb_abun %&gt;% filter(Sfp == &#39;SFP.high&#39;), 
              colour = viridis::viridis(n = 3)[3], pch = 16, width = .3) + 
  geom_jitter(data = comb_abun %&gt;% filter(Sfp == &#39;SFP.low&#39;),
              colour = viridis::viridis(n = 3)[2], pch = 16, width = .3) + 
  geom_jitter(data = comb_abun %&gt;% filter(Sfp == &#39;Sperm.only&#39;), pch = 16, width = .3, alpha = 0) + 
  geom_boxplot(notch = TRUE, outlier.shape = NA, alpha = .6, width = .3) +
  scale_fill_viridis_d(direction = -1) +
  #scale_colour_manual(values = c(viridis::viridis(n = 3, direction = -1)[-3], NA)) +
  scale_x_discrete(labels = c(&quot;SFP.high&quot; = &quot;\&quot;High conf.\&quot;\nSfps&quot;, 
                              &quot;SFP.low&quot; = &quot;\&quot;Low conf.\&quot;\nSfps&quot;,
                              &quot;Sperm.only&quot; = &quot;Other\nsperm&quot;)) + 
  labs(y = &#39;log2(abundance)&#39;) +
  theme_bw() +
  theme(legend.position = &#39;none&#39;,
        axis.title.x = element_blank()) +
  # geom_signif(y_position = 37, xmin = 1, xmax = 3, annotation = &#39;**&#39;, tip_length = 0.01) +
  # geom_signif(y_position = 36, xmin = c(1, 2), xmax = c(2, 3), annotation = c(&#39;**&#39;, &#39;n.s.&#39;), tip_length = 0.01) +
  geom_text(data = comb_abun %&gt;% group_by(Sfp) %&gt;% dplyr::count(), 
            aes(y = 13, label = n), size = 5, colour = &quot;black&quot;) +
  #ggsave(&#39;figures/sperm_sfp_abundance.pdf&#39;, height = 4, width = 4) +
  NULL

sfp_abun</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-29-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-29-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="high-abundance-sfps-in-the-dmsp3" class="section level3">
<h3>High abundance Sfps in the DmSP3</h3>
<pre class="r"><code>comb_abun %&gt;% 
  mutate(p.rank = percent_rank(grand.mean) * 100,
         p.rank = round(p.rank, 1),
         NAME = if_else(NAME == &#39;-&#39;, SYMBOL, NAME)) %&gt;% 
  filter(Sfp == &#39;SFP.high&#39; &amp; grand.mean &gt; median(comb_abun$grand.mean[comb_abun$Sfp == &#39;Sperm.only&#39;])) %&gt;% 
  dplyr::select(FBgn, Name = NAME, Chromosome, `Ranked abundance (%)` = p.rank) %&gt;% 
  arrange(desc(`Ranked abundance (%)`)) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-d1fd0327dbd6172359b8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d1fd0327dbd6172359b8">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0003885","FBgn0011559","FBgn0265264","FBgn0051872","FBgn0030362","FBgn0022355","FBgn0037879","FBgn0002855","FBgn0004181","FBgn0250841","FBgn0034229","FBgn0032868","FBgn0050104","FBgn0030828","FBgn0286818","FBgn0054051","FBgn0260745","FBgn0053126","FBgn0040097","FBgn0037888","FBgn0026415","FBgn0001218","FBgn0264815","FBgn0043825","FBgn0028986","FBgn0037038","FBgn0011694","FBgn0028987","FBgn0261055","FBgn0034152","FBgn0003748","FBgn0259964","FBgn0020414","FBgn0034417","FBgn0036072","FBgn0250847","FBgn0016123","FBgn0259952","FBgn0259975","FBgn0036969","FBgn0040098","FBgn0039562","FBgn0034753","FBgn0015586"],["alpha-Tubulin at 84D","Accessory gland protein 36DE","CG17097","CG31872","regucalcin","Transferrin 1","SCP-containing protein C","Accessory gland protein 26Aa","Ejaculatory bulb protein","CG17242","CG4847","CG17472","Ecto-5'-nucleotidase 2","CG5162","Protein disulfide isomerase","CG34051","midline fasciclin","Neural Lazarillo","lectin-30A","SCP-containing protein B","Imaginal disc growth factor 4","Heat shock 70-kDa protein cognate 3","Phosphodiesterase 1c","CG18284","Serpin 38F","CG11037","Ejaculatory bulb protein II","Serpin 28F","Seminal fluid protein 26Ad","Accessory gland protein 53C14a","Trehalase","Seminal fluid protein 33A3","Imaginal disc growth factor 3","CG15117","CG6628","CG14034","Alkaline phosphatase 4","Seminal fluid protein 24Bb","Seminal fluid protein 87B","Serpin 77Bb","lectin-29Ca","Glycoprotein 93","CG2852","Accessory gland protein 76A"],["3R","2L","2L","2L","X","X","3R","2L","2R","2L","2R","2L","2R","X","3L","2L","3R","2L","2L","3R","X","X","2L","2L","2L","3L","2R","2L","2L","2R","2R","2L","2L","2R","3L","2L","3R","2L","3R","3L","2L","3R","2R","3L"],[99.8,97.8,94.7,90.1,89.7,87.3,84.6,84,83,79.3,78.9,73.9,72.5,70.6,69.2,69.1,68.3,67.3,66.9,66.8,66.6,66.5,65.5,64.6,64.6,64.4,64,63.9,63.6,63.6,63.4,62.8,62.3,61.6,61.1,60,58.9,58.9,57.7,55.3,54.4,52.2,51.9,50.4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>Chromosome<\/th>\n      <th>Ranked abundance (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="experiment-2---pbst-treatment-pbst" class="section level2 tabset">
<h2 class="tabset">Experiment 2 - PBST treatment {#PBST}</h2>
<p>In experiment 2 we washed sperm samples with PBS and Halt protease inhibitor (‘Halt’ treatment), PBS only (‘NoHalt’ treatment), or PBS containing Triton X detergent (‘PBST’ treatment) to denature the cell plasma membrane. We then performed pairswise differential abundance analysis between each treatment using <code>edgeR</code>. Few proteins were differentially abundant between the ‘Halt’ and ‘NoHalt’ controls. Subsequently, we performed differential abundance analysis between both controls compared to the ‘PBST’ treatment.</p>
<pre class="r"><code>pbst_dat &lt;- PBST_dat %&gt;% 
  mutate(Sfp = case_when(FBgn %in% wigbySFP$FBgn[wigbySFP$category == &#39;highconf&#39;] ~ &#39;SFP.high&#39;,
                         FBgn %in% wigbySFP$FBgn ~ &#39;SFP.low&#39;,
                         TRUE ~ &#39;Sperm.only&#39;)) %&gt;% 
  dplyr::select(Accession, FBgn, Sfp, SYMBOL, 73:77, UP = `# Unique Peptides`, starts_with(&#39;Abundance:&#39;)) %&gt;% 
  mutate(NAME = if_else(NAME == &#39;-&#39;, ANNOTATION_SYMBOL, NAME)) %&gt;% 
  filter(UP &gt;= 2)

colnames(pbst_dat) &lt;- gsub(&#39;.*, &#39;, &#39;&#39;, x = colnames(pbst_dat))

colnames(pbst_dat)[9:17] &lt;- paste0(colnames(pbst_dat)[9:17], 1:3)

# remove proteins which are 0&#39;s across all reps of all treatments
pbst_dat2 &lt;- pbst_dat %&gt;% 
  mutate(across(9:17, ~replace_na(.x, 0)),
         mn_Halt = rowMeans(dplyr::select(., starts_with(&quot;Halt&quot;)), na.rm = TRUE),
         mn_NoHalt = rowMeans(dplyr::select(., starts_with(&quot;NoHalt&quot;)), na.rm = TRUE),
         mn_PBST = rowMeans(dplyr::select(., starts_with(&quot;PBST&quot;)), na.rm = TRUE)) %&gt;% 
  filter(mn_Halt != 0 | mn_NoHalt != 0 | mn_PBST != 0) %&gt;% 
  mutate(across(18:20, ~replace_na(.x, 0)))

# # Sfps detected in dataset?
# pbst_dat2 %&gt;% group_by(Sfp) %&gt;% dplyr::count()

# # how many ID&#39;d in each
# upset(fromList(list(Halt = pbst_dat2$Accession[pbst_dat2$mn_Halt != 0],
#                     NoHalt = pbst_dat2$Accession[pbst_dat2$mn_NoHalt != 0],
#                     PBST = pbst_dat2$Accession[pbst_dat2$mn_PBST != 0])))</code></pre>
<div id="pairwise-analysis" class="section level3 tabset">
<h3 class="tabset">Pairwise analysis</h3>
<p>Here, we performed pairwise analyses between each treatment. We filtered proteins present in fewer than 7 out of 9 replicates.</p>
<pre class="r"><code># make object for protein abundance data
expr_pbst &lt;- pbst_dat2[, 9:17]

# filter data to exclude 0&#39;s
thresh.pbst = expr_pbst &gt; 0
keep.pbst = rowSums(thresh.pbst) &gt;= 7
pbst.Filtered = expr_pbst[keep.pbst, ]

# get sample info
sampInfo_pbst = data.frame(condition = str_sub(colnames(pbst.Filtered), 1, 1),
                           Replicate = str_sub(colnames(pbst.Filtered), -1))

# make design matrix to test diffs between groups
design_pbst &lt;- model.matrix(~0 + sampInfo_pbst$condition)
colnames(design_pbst) &lt;- unique(sampInfo_pbst$condition)
rownames(design_pbst) &lt;- sampInfo_pbst$Replicate

# create DGElist and fit model 
dgeList_pbst &lt;- DGEList(counts = pbst.Filtered, genes = pbst_dat2$Accession[keep.pbst], 
                        group = sampInfo_pbst$condition)
dgeList_pbst &lt;- calcNormFactors(dgeList_pbst, method = &#39;TMM&#39;)
dgeList_pbst &lt;- estimateCommonDisp(dgeList_pbst)
dgeList_pbst &lt;- estimateTagwiseDisp(dgeList_pbst)

# make contrasts - higher values = higher in mated
cont.pbst &lt;- makeContrasts(H.v.N = H - N,
                           H.v.P = H - P,
                           N.v.P = N - P,
                           levels = design_pbst)

# fit GLM
glm_pbst &lt;- glmFit(dgeList_pbst, design_pbst, robust = TRUE)

# pairwise
thresh.hn &lt;- expr_pbst[, 1:6] &gt; 0
thresh.hp &lt;- expr_pbst[, c(1:3, 7:9)] &gt; 0
thresh.np &lt;- expr_pbst[, 4:9] &gt; 0

keep.hn &lt;- rowSums(thresh.hn) &gt;= 4
keep.hp &lt;- rowSums(thresh.hp) &gt;= 4
keep.np &lt;- rowSums(thresh.np) &gt;= 4

hn.Filtered &lt;- expr_pbst[keep.hn, ]
hp.Filtered &lt;- expr_pbst[keep.hp, ]
np.Filtered &lt;- expr_pbst[keep.np, ]

dgeList_hn &lt;- DGEList(counts = hn.Filtered, genes = pbst_dat2$Accession[keep.hn], group = sampInfo_pbst$condition)
dgeList_hn &lt;- calcNormFactors(dgeList_hn, method = &#39;TMM&#39;)
dgeList_hn &lt;- estimateCommonDisp(dgeList_hn)
dgeList_hn &lt;- estimateTagwiseDisp(dgeList_hn)

dgeList_hp &lt;- DGEList(counts = hp.Filtered, genes = pbst_dat2$Accession[keep.hp], group = sampInfo_pbst$condition)
dgeList_hp &lt;- calcNormFactors(dgeList_hp, method = &#39;TMM&#39;)
dgeList_hp &lt;- estimateCommonDisp(dgeList_hp)
dgeList_hp &lt;- estimateTagwiseDisp(dgeList_hp)

dgeList_np &lt;- DGEList(counts = np.Filtered, genes = pbst_dat2$Accession[keep.np], group = sampInfo_pbst$condition)
dgeList_np &lt;- calcNormFactors(dgeList_np, method = &#39;TMM&#39;)
dgeList_np &lt;- estimateCommonDisp(dgeList_np)
dgeList_np &lt;- estimateTagwiseDisp(dgeList_np)

cont.hn &lt;- makeContrasts(H.v.N = H - N, levels = design_pbst)
cont.hp &lt;- makeContrasts(H.v.P = H - P, levels = design_pbst)
cont.np &lt;- makeContrasts(N.v.P = N - P, levels = design_pbst)

glm_hn &lt;- glmFit(dgeList_hn, design_pbst, robust = TRUE)
glm_hp &lt;- glmFit(dgeList_hp, design_pbst, robust = TRUE)
glm_np &lt;- glmFit(dgeList_np, design_pbst, robust = TRUE)</code></pre>
<div id="diagnostic-plots" class="section level4 tabset">
<h4 class="tabset">Diagnostic plots</h4>
<pre class="r"><code>diag_plot &lt;- function(dgelist = NA, design = NA) {
  
  par(mfrow = c(2,2))
  # Biological coefficient of variation
  plotBCV(dgelist)
  # mean-variance trend
  voomed = voom(dgelist, design, plot=TRUE)
  # QQ-plot
  g &lt;- gof(glmFit(dgelist, design_pbst))
  z &lt;- zscoreGamma(g$gof.statistics,shape=g$df/2,scale=2)
  qqnorm(z); qqline(z, col = 4,lwd=1,lty=1)
  # log2 transformed and normalize boxplot of counts across samples
  boxplot(voomed$E, xlab=&quot;&quot;, ylab=&quot;log2(Abundance)&quot;,las=2,main=&quot;Voom transformed logCPM&quot;,
        col = c(rep(2:4, each = 3)))
  abline(h=median(voomed$E),col=&quot;blue&quot;)
  par(mfrow=c(1,1))
  
}</code></pre>
<div id="h-vs.-n" class="section level5">
<h5>H vs. N</h5>
<pre class="r"><code>diag_plot(dgeList_hn, design_pbst)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-34-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-34-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h-vs.-p" class="section level5">
<h5>H vs. P</h5>
<pre class="r"><code>diag_plot(dgeList_hp, design_pbst)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-35-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-35-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="n-vs.-p" class="section level5">
<h5>N vs. P</h5>
<pre class="r"><code>diag_plot(dgeList_np, design_pbst)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-36-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-36-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="correlation-plot" class="section level4">
<h4>Correlation plot</h4>
<pre class="r"><code>## Plot sample correlation
data = glm_pbst$fitted.values %&gt;% as_tibble()
data = as.matrix(data)
sample_cor = cor(data, method = &#39;pearson&#39;, use = &#39;pairwise.complete.obs&#39;)

pheatmap(
  mat               = sample_cor,
  border_color      = NA,
  annotation_legend = TRUE,
  annotation_names_col = FALSE,
  annotation_names_row = FALSE,
  fontsize          = 12#, file = &quot;plots/sample.cor.pdf&quot;, height = 5.5, width = 6.5    
)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-37-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-37-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mds-plot" class="section level4 tabset">
<h4 class="tabset">MDS plot</h4>
<pre class="r"><code>mdsObj &lt;- plotMDS(dgeList_pbst, plot = F, dim.plot = c(1,3))$cmdscale.out

mdsObj &lt;- as.data.frame(as.matrix(mdsObj)) %&gt;% 
  rownames_to_column() %&gt;% 
  mutate(condition = str_sub(rowname, 1, 2)) %&gt;% 
  dplyr::rename(dim1 = V1, dim2 = V2, dim3 = V3)</code></pre>
<div id="dim.1-vs.-dim.-2" class="section level5">
<h5>Dim.1 vs. Dim. 2</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim2, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs(x = &quot;Dimension 1&quot;, y = &quot;Dimension 2&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot12_db.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-39-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-39-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.1-vs.-dim.-3" class="section level5">
<h5>Dim.1 vs. Dim. 3</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 1&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot13.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-40-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-40-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.2-vs.-dim.-3" class="section level5">
<h5>Dim.2 vs. Dim. 3</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim2, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 2&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot23.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-41-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-41-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="perform-differential-abundance-analysis" class="section level4">
<h4>Perform differential abundance analysis</h4>
<pre class="r"><code># perform likelihood ratio tests
lrt.hn &lt;- glmLRT(glm_hn, contrast = cont.hn[,&quot;H.v.N&quot;])
lrt.hp &lt;- glmLRT(glm_hp, contrast = cont.hp[,&quot;H.v.P&quot;])
lrt.np &lt;- glmLRT(glm_np, contrast = cont.np[,&quot;N.v.P&quot;])

# halt vs. no halt
lm_hn.tTags.table &lt;- topTags(lrt.hn, n = NULL) %&gt;% as.data.frame() %&gt;% 
  mutate(condition = &quot;hn&quot;,
         threshold = if_else(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, &quot;SD&quot;, &quot;NS&quot;))

# halt vs. pbst
lm_hp.tTags.table &lt;- topTags(lrt.hp, n = NULL) %&gt;% as.data.frame() %&gt;% 
  mutate(condition = &quot;hp&quot;,
         threshold = if_else(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, &quot;SD&quot;, &quot;NS&quot;))

# no halt vs. pbst
lm_np.tTags.table &lt;- topTags(lrt.np, n = NULL) %&gt;% as.data.frame() %&gt;% 
  mutate(condition = &quot;np&quot;,
         threshold = if_else(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, &quot;SD&quot;, &quot;NS&quot;))

# combine results
comb_TABLES &lt;- rbind(lm_hn.tTags.table,
                     lm_hp.tTags.table, 
                     lm_np.tTags.table) %&gt;% 
  left_join(pbst_dat2,
            by = c(&#39;genes&#39; = &#39;Accession&#39;))


# New facet label names
new.labs &lt;- c(hn = &quot;Halt vs. No Halt&quot;,                    
              hp = &quot;Halt vs. PBST&quot;,
              np = &quot;No Halt vs. PBST&quot;)</code></pre>
<div id="volcano-plot" class="section level5">
<h5>volcano plot</h5>
<pre class="r"><code>comb_TABLES %&gt;% filter(threshold != &#39;SD&#39;) %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(alpha = .3, colour = &#39;grey&#39;) +
  geom_point(data = comb_TABLES %&gt;% 
               filter(threshold != &#39;NS&#39;),
             aes(colour = Sfp), alpha = .8) +
  scale_colour_viridis_d(direction = -1) +
  labs(x = &#39;logFC&#39;, y = &#39;-log10(p-value)&#39;) +
  facet_wrap(~condition, labeller = as_labeller(new.labs)) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;,#c(0.8, 0.8),
        legend.title = element_blank(),
        legend.background = element_rect(fill = NA)) +
  ggrepel::geom_label_repel(
    data = comb_TABLES %&gt;% filter(threshold != &#39;NS&#39; &amp; Sfp == &#39;SFP.high&#39;),
    aes(label = SYMBOL),
    size = 2,
    colour = &#39;black&#39;,
    box.padding = unit(0.35, &quot;lines&quot;),
    point.padding = unit(0.3, &quot;lines&quot;),
    max.overlaps = 20
  ) +
  #ggsave(&#39;figures/volcano_PBSTexp.pdf&#39;, height = 4, width = 10) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-43-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># # DA proteins between &#39;controls&#39;
# comb_TABLES %&gt;% filter(threshold == &#39;SD&#39; &amp; condition == &#39;hn&#39;)

# # number DA in each comparison
# comb_TABLES %&gt;% 
#   filter(threshold == &#39;SD&#39;) %&gt;% 
#   group_by(condition) %&gt;% dplyr::count()

# # Sfps remaining in PBST (i.e. not sig different compared to control)
# comb_TABLES %&gt;% 
#   filter(condition != &#39;hn&#39; &amp; threshold != &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;) %&gt;% 
#   distinct(FBgn, .keep_all = TRUE)

pbst_keep &lt;- data.frame(genes = intersect(
  lm_hp.tTags.table$genes[lm_hp.tTags.table$FDR &gt; 0.05],
  lm_np.tTags.table$genes[lm_np.tTags.table$FDR &gt; 0.05])) %&gt;% 
  left_join(pbst_dat2, by = c(&#39;genes&#39; = &#39;Accession&#39;))

# pbst_keep %&gt;% group_by(Sfp) %&gt;% dplyr::count()
# 
# pbst_keep %&gt;% filter(Sfp == &#39;SFP.high&#39;)
# pbst_keep %&gt;% filter(Sfp == &#39;SFP.low&#39;)

# which are these proteins?
pbst_high &lt;- data.frame(genes = intersect(
  lm_hp.tTags.table$genes[lm_hp.tTags.table$logFC &lt; -1 &amp; lm_hp.tTags.table$FDR &lt; 0.05],
  lm_np.tTags.table$genes[lm_np.tTags.table$logFC &lt; -1 &amp; lm_np.tTags.table$FDR &lt; 0.05])) %&gt;% 
  left_join(pbst_dat2, by = c(&#39;genes&#39; = &#39;Accession&#39;))

# absent from PBST
pbst_absent &lt;- data.frame(genes = intersect(pbst_dat2$Accession[!keep.hp], 
                                            pbst_dat2$Accession[!keep.np])) %&gt;% 
  left_join(pbst_dat2, by = c(&#39;genes&#39; = &#39;Accession&#39;))

# pbst_absent %&gt;% group_by(Sfp) %&gt;% dplyr::count()
# 
# pbst_absent %&gt;% filter(Sfp == &#39;SFP.high&#39;)
# pbst_absent %&gt;% filter(Sfp == &#39;SFP.low&#39;)</code></pre>
</div>
<div id="proteins-lower-in-abundance-in-pbst-vs.-nohalt" class="section level5">
<h5>proteins lower in abundance in PBST vs. no/halt</h5>
<pre class="r"><code>pbst_low &lt;- data.frame(genes = intersect(
  lm_hp.tTags.table$genes[lm_hp.tTags.table$logFC &gt; 0 &amp; lm_hp.tTags.table$FDR &lt; 0.05],
  lm_np.tTags.table$genes[lm_np.tTags.table$logFC &gt; 0 &amp; lm_np.tTags.table$FDR &lt; 0.05])) %&gt;% 
  left_join(pbst_dat2, by = c(&#39;genes&#39; = &#39;Accession&#39;))

# pbst_low %&gt;% group_by(Sfp) %&gt;% dplyr::count()
# 
# pbst_low %&gt;% filter(Sfp == &#39;SFP.high&#39;)
# pbst_low %&gt;% filter(Sfp == &#39;SFP.low&#39;)

# plot abundances of SFPs for each treatment
pbst_low %&gt;% 
  filter(Sfp == &#39;SFP.high&#39;) %&gt;% 
  pivot_longer(cols = 9:17) %&gt;% 
  mutate(condition = str_sub(name, end = -2),
         repl = str_sub(name, end = -1, -1)) %&gt;% 
  ggplot(aes(x = condition, y = log2(value))) +
  geom_jitter(aes(colour = condition, shape = repl), width = .2, size = 3) +
  facet_wrap(~NAME, ncol = round(nrow(pbst_low %&gt;% filter(Sfp == &#39;SFP.high&#39;))/3,0), scales = &#39;free_y&#39;) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  stat_summary(geom = &#39;point&#39;, fun = &#39;median&#39;, pch = 23, size = 3) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-44-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="single-model" class="section level3 tabset">
<h3 class="tabset">Single model</h3>
<p>Here we test for differences in abundance between the PBST treatment vs. the average of both controls. We first exclude the 16 proteins which showed significant differences in abundance between ‘Halt’ and ‘NoHalt’.</p>
<pre class="r"><code># remove proteins differentially abundant between Halt/NoHalt
expr_pbst2 &lt;- pbst_dat2 %&gt;% 
  filter(!Accession %in% lm_hn.tTags.table$genes[lm_hn.tTags.table$threshold == &#39;SD&#39;]) %&gt;% 
  dplyr::select(Accession, 9:17)

# filter data to exclude 0&#39;s
thresh.pbst = expr_pbst2[, -1] &gt; 0
keep.pbst = rowSums(thresh.pbst) &gt;= 7
pbst.Filtered = expr_pbst2[keep.pbst, -1]

# create DGElist and fit model 
dgeList_pbst &lt;- DGEList(counts = pbst.Filtered, genes = expr_pbst2$Accession[keep.pbst], 
                        group = sampInfo_pbst$condition)
dgeList_pbst &lt;- calcNormFactors(dgeList_pbst, method = &#39;TMM&#39;)
dgeList_pbst &lt;- estimateCommonDisp(dgeList_pbst)
dgeList_pbst &lt;- estimateTagwiseDisp(dgeList_pbst)</code></pre>
<div id="diagnostic-plots-1" class="section level4">
<h4>Diagnostic plots</h4>
<pre class="r"><code>par(mfrow = c(2,2))
# Biological coefficient of variation
plotBCV(dgeList_pbst)
# mean-variance trend
voomed = voom(dgeList_pbst, design_pbst, plot = TRUE)
# QQ-plot
g &lt;- gof(glmFit(dgeList_pbst, design_pbst))
z &lt;- zscoreGamma(g$gof.statistics,shape=g$df/2,scale=2)
qqnorm(z); qqline(z, col = 4,lwd=1,lty=1)
# log2 transformed and normalize boxplot of counts across samples
boxplot(voomed$E, xlab=&quot;&quot;, ylab=&quot;log2(Abundance)&quot;,las=2,main=&quot;Voom transformed logCPM&quot;,
        col = c(rep(2:4, each = 3)))
abline(h=median(voomed$E),col=&quot;blue&quot;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow=c(1,1))

# fit model
dgeList2_fit &lt;- glmFit(dgeList_pbst, design_pbst)</code></pre>
</div>
<div id="correlation-plot-1" class="section level4">
<h4>Correlation plot</h4>
<pre class="r"><code>## Plot sample correlation
data = glm_pbst$fitted.values %&gt;% as_tibble()
data = as.matrix(data)
sample_cor = cor(data, method = &#39;pearson&#39;, use = &#39;pairwise.complete.obs&#39;)

pheatmap(
  mat               = sample_cor,
  border_color      = NA,
  annotation_legend = TRUE,
  annotation_names_col = FALSE,
  annotation_names_row = FALSE,
  fontsize          = 12#, file = &quot;plots/sample.cor.pdf&quot;, height = 5.5, width = 6.5    
)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-47-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mds-plot-1" class="section level4 tabset">
<h4 class="tabset">MDS plot</h4>
<pre class="r"><code>mdsObj &lt;- plotMDS(dgeList_pbst, plot = F, dim.plot = c(1,3))$cmdscale.out

mdsObj &lt;- as.data.frame(as.matrix(mdsObj)) %&gt;% 
  rownames_to_column() %&gt;% 
  mutate(condition = str_sub(rowname, 1, 2)) %&gt;% 
  dplyr::rename(dim1 = V1, dim2 = V2, dim3 = V3)</code></pre>
<div id="dim.1-vs.-dim.-2-1" class="section level5">
<h5>Dim.1 vs. Dim. 2</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim2, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs(x = &quot;Dimension 1&quot;, y = &quot;Dimension 2&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot12_db.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-49-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-49-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.1-vs.-dim.-3-1" class="section level5">
<h5>Dim.1 vs. Dim. 3</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 1&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot13.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-50-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-50-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.2-vs.-dim.-3-1" class="section level5">
<h5>Dim.2 vs. Dim. 3</h5>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim2, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 2&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot23.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-51-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-51-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># make contrast
pbst.v.rest &lt;- makeContrasts(((H + N)/2) - P, levels = design_pbst)

# perform LRT
lrt.pbst_v_rest &lt;- glmLRT(dgeList2_fit, contrast = pbst.v.rest)

# make results table
lrt.pbst_v_rest.tTags &lt;- topTags(lrt.pbst_v_rest, n = NULL) %&gt;% as.data.frame() %&gt;% 
  mutate(threshold = if_else(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, &quot;SD&quot;, &quot;NS&quot;)) %&gt;% 
  left_join(pbst_dat2,
            by = c(&#39;genes&#39; = &#39;Accession&#39;))

pbst_volcano &lt;- lrt.pbst_v_rest.tTags %&gt;% filter(threshold != &#39;SD&#39;) %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(alpha = .3, colour = &#39;grey&#39;) +
  geom_point(data = lrt.pbst_v_rest.tTags %&gt;% filter(threshold != &#39;NS&#39;),
             aes(colour = Sfp), alpha = .6) +
  scale_colour_viridis_d(direction = -1) +
  labs(x = &#39;logFC&#39;, y = &#39;-log10(p-value)&#39;) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;,#c(0.8, 0.8),
        legend.title = element_blank(),
        legend.background = element_rect(fill = NA)) +
  ggrepel::geom_label_repel(
    data = lrt.pbst_v_rest.tTags %&gt;% filter(threshold != &#39;NS&#39; &amp; Sfp == &#39;SFP.high&#39;),
    aes(label = SYMBOL),
    size = 2,
    colour = &#39;black&#39;,
    box.padding = unit(0.35, &quot;lines&quot;),
    point.padding = unit(0.3, &quot;lines&quot;),
    max.overlaps = 20
  ) +
  #ggsave(&#39;figures/volcano_PBSTvCOMBINED.pdf&#39;, height = 4.5, width = 4) +
  NULL</code></pre>
</div>
</div>
<div id="volcano-plot-1" class="section level4">
<h4>Volcano plot</h4>
<pre class="r"><code>pbst_volcano</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-53-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-53-1">
Past versions of unnamed-chunk-53-1.png
</button>
</p>
<div id="fig-unnamed-chunk-53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-53-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#table(lrt.pbst_v_rest.tTags$threshold)

# lrt.pbst_v_rest.tTags %&gt;% filter(threshold == &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;)
# lrt.pbst_v_rest.tTags %&gt;% filter(threshold != &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;)

# # number DA in each comparison
# lrt.pbst_v_rest.tTags %&gt;% filter(threshold == &#39;SD&#39; &amp; logFC &lt; 0)
# lrt.pbst_v_rest.tTags %&gt;% filter(threshold == &#39;SD&#39; &amp; logFC &gt; 0)

# compare to pairwise analysis
# upset(
#   fromList(
#     list(pairwise = pbst_low$genes,
#          one_mod = lrt.pbst_v_rest.tTags$genes[lm_np.tTags.table$logFC &gt; 1 &amp; lm_np.tTags.table$FDR &lt; 0.05])))

# # what proteins are not DA or filtered?
# upset(
#   fromList(
#     list(total = pbst_dat2 %&gt;% filter(Sfp == &#39;SFP.high&#39;) %&gt;% pull(Accession),
#          filt = lrt.pbst_v_rest.tTags %&gt;% filter(Sfp == &#39;SFP.high&#39;) %&gt;% pull(genes))))

# data.frame(Accession = setdiff(pbst_dat2 %&gt;% filter(Sfp == &#39;SFP.high&#39;) %&gt;% pull(Accession),
#                                lrt.pbst_v_rest.tTags %&gt;% filter(Sfp == &#39;SFP.high&#39;) %&gt;% pull(genes))) %&gt;% 
#   left_join(pbst_dat2[, -c(2:8)], by = &#39;Accession&#39;)


# # write to file for GO
# lrt.pbst_v_rest.tTags[lrt.pbst_v_rest.tTags$logFC &gt; 1 &amp; lrt.pbst_v_rest.tTags$FDR &lt; 0.05, ] #%&gt;% write_csv(&#39;output/GO_lists/pbst_low.csv&#39;)</code></pre>
</div>
<div id="proteins-depleted-after-pbst-treatment" class="section level4">
<h4>Proteins depleted after PBST treatment</h4>
<pre class="r"><code>lrt.pbst_v_rest.tTags[lrt.pbst_v_rest.tTags$logFC &gt; 1 &amp; lrt.pbst_v_rest.tTags$FDR &lt; 0.05, ] %&gt;% 
  mutate(NAME = if_else(is.na(NAME) == TRUE, ANNOTATION_SYMBOL, NAME),
         Sfp = case_when(Sfp == &#39;SFP.high&#39; ~ &#39;High conf. Sfp&#39;,
                         Sfp == &#39;SFP.low&#39; ~ &#39;Low conf. Sfp&#39;,
                         TRUE ~ &#39;Sperm&#39;)) %&gt;% 
  dplyr::select(FBgn, Name = NAME, genes, Sfp, Chromosome = LOCATION_ARM) %&gt;% 
  arrange(Name) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-7c5cc93b617cad5fb939" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7c5cc93b617cad5fb939">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0002855","FBgn0011559","FBgn0034153","FBgn0015586","FBgn0000044","FBgn0000042","FBgn0000047","FBgn0042138","FBgn0025111","FBgn0010548","FBgn0087002","FBgn0028646","FBgn0010217","FBgn0286508","FBgn0250788","FBgn0005585","FBgn0038395","FBgn0027552","FBgn0250847","FBgn0038685","FBgn0038656","FBgn0265271","FBgn0037942","FBgn0030283","FBgn0031108","FBgn0033446","FBgn0265264","FBgn0260453","FBgn0250841","FBgn0032868","FBgn0023537","FBgn0039674","FBgn0034753","FBgn0050103","FBgn0050491","FBgn0051195","FBgn0051482","FBgn0051659","FBgn0051680","FBgn0051693","FBgn0051872","FBgn0052081","FBgn0052832","FBgn0053303","FBgn0064119","FBgn0054051","FBgn0260766","FBgn0031407","FBgn0266448","FBgn0034229","FBgn0039371","FBgn0039358","FBgn0038326","FBgn0034914","FBgn0036186","FBgn0032314","FBgn0034432","FBgn0035567","FBgn0033734","FBgn0034479","FBgn0033679","FBgn0031784","FBgn0028513","FBgn0014141","FBgn0000299","FBgn0019830","FBgn0010434","FBgn0038194","FBgn0032833","FBgn0284248","FBgn0015623","FBgn0010019","FBgn0038797","FBgn0033248","FBgn0000454","FBgn0033663","FBgn0035909","FBgn0284245","FBgn0029176","FBgn0000559","FBgn0001942","FBgn0283427","FBgn0266451","FBgn0263199","FBgn0004868","FBgn0001114","FBgn0037970","FBgn0001098","FBgn0001125","FBgn0001142","FBgn0001149","FBgn0001128","FBgn0004507","FBgn0266064","FBgn0039562","FBgn0001218","FBgn0015245","FBgn0031728","FBgn0001233","FBgn0266599","FBgn0001220","FBgn0001196","FBgn0001200","FBgn0038233","FBgn0260466","FBgn0026415","FBgn0027108","FBgn0027291","FBgn0034405","FBgn0266449","FBgn0001258","FBgn0010397","FBgn0002526","FBgn0261800","FBgn0040098","FBgn0010482","FBgn0263594","FBgn0262559","FBgn0267366","FBgn0026409","FBgn0038662","FBgn0028479","FBgn0025352","FBgn0026061","FBgn0014863","FBgn0002789","FBgn0038294","FBgn0002921","FBgn0036316","FBgn0010352","FBgn0053126","FBgn0264975","FBgn0030724","FBgn0033268","FBgn0034277","FBgn0014868","FBgn0020513","FBgn0003149","FBgn0035438","FBgn0003076","FBgn0014869","FBgn0265297","FBgn0004363","FBgn0069354","FBgn0286818","FBgn0267385","FBgn0033814","FBgn0285937","FBgn0015790","FBgn0020255","FBgn0033906","FBgn0030362","FBgn0260442","FBgn0028697","FBgn0285950","FBgn0010078","FBgn0026372","FBgn0035422","FBgn0285949","FBgn0064225","FBgn0005593","FBgn0000100","FBgn0010412","FBgn0011284","FBgn0039757","FBgn0010408","FBgn0021906","FBgn0028569","FBgn0037298","FBgn0086906","FBgn0037888","FBgn0259952","FBgn0028987","FBgn0028986","FBgn0037239","FBgn0030026","FBgn0003360","FBgn0028560","FBgn0284408","FBgn0033644","FBgn0027560","FBgn0004117","FBgn0011726","FBgn0031757","FBgn0040259","FBgn0053519","FBgn0004169","FBgn0022097","FBgn0016075","FBgn0004397","FBgn0283471","FBgn0040064","FBgn0005391","FBgn0265991","FBgn0052311",null,null,null],["Accessory gland protein 26Aa","Accessory gland protein 36DE","Accessory gland protein 53C14b","Accessory gland protein 76A","Actin 57B","Actin 5C","Actin 88F","Acyl-protein thioesterase 1","Adenine nucleotide translocase 2","Aldehyde dehydrogenase type III","apolipophorin","aralar1","ATP synthase, beta subunit","beta Hydroxy acid dehydrogenase 1","beta Spectrin","Calreticulin","CG10407","CG10863","CG14034","CG14282","CG14294","CG14717","CG14721","CG15198","CG15459","CG1648","CG17097","CG17140","CG17242","CG17472","CG17896","CG1907","CG2852","CG30103","CG30491","CG31195","CG31482","CG31659","CG31680","CG31693","CG31872","CG32081","CG32832","CG33303","CG33934","CG34051","CG42564","CG4270","CG45078","CG4847","CG4960","CG5028","CG5044","CG5554","CG6071","CG7309","CG7461","CG7514","CG8520","CG8654","CG8888","CG9222","CG9254","cheerio","Collagen type IV alpha 1","congested-like trachea","coracle","Cyp6d5","Cytochrome c oxidase subunit 4","Cytochrome c proximal","Cytochrome P450 reductase","Cytochrome P450-4g1","Dicarboxylate carrier 2","Dicarboxylate carrier 3","Dipeptidase B","Endoplasmic reticulum p60","ergic53","eukaryotic translation elongation factor 1 alpha 1","eukaryotic translation elongation factor 1 gamma","eukaryotic translation elongation factor 2","eukaryotic translation initiation factor 4A","Fatty acid synthase 1","fau","Galactokinase","GDP dissociation inhibitor","Glutactin","Glutamate Carrier 2","Glutamate dehydrogenase","Glutamate oxaloacetate transaminase 2","Glutamine synthetase 1","Glutathione S transferase D1","Glycerol-3-phosphate dehydrogenase 1","Glycogen phosphorylase","Glycogen synthase","Glycoprotein 93","Heat shock 70-kDa protein cognate 3","Heat shock protein 60A","Heat shock protein 60C","Heat shock protein 83","Heat shock protein cognate 4","Heat shock protein cognate 5","Histone H2A","Histone H4","HTRA2-related serine protease","I'm not dead yet 2","Imaginal disc growth factor 4","Innexin 2","Isocitrate dehydrogenase 3a","Juvenile hormone epoxide hydrolase 2","Kruppel homolog 2","Lactate dehydrogenase","Lamin C","Laminin A","LanB1","lectin-29Ca","lethal (2) 01289","lost","Malate dehydrogenase 2","milkah","Mitochondrial phosphate carrier protein 2","Mitochondrial pyruvate carrier","Mitochondrial trifunctional protein alpha subunit","Mitochondrial trifunctional protein beta subunit","Multiple inositol polyphosphate phosphatase 1","Muscle LIM protein at 84B","Muscle protein 20","Myofilin","Na pump alpha subunit","nebulosa","Neural conserved at 73EF","Neural Lazarillo","Neuroglian","Nipsnap","Odorant-binding protein 44a","Oligosaccharide transferase Delta subunit","Oligosaccharyltransferase 48kD subunit","PAICS bifunctional enzyme","Paramyosin","PHGPx","Phosphoglucose mutase 1","Phosphoglyceromutase 78","poly(A) binding protein","porin","Porin2","Protein disulfide isomerase","Pyruvate kinase","Quiescin sulfhydryl oxidase 1","Rab1","Rab11","Ran","Receptor expression enhancing protein B","regucalcin","rhea","Ribosomal protein L15","Ribosomal protein L19","Ribosomal protein L23","Ribosomal protein L23A","Ribosomal protein L28","Ribosomal protein L31","Ribosomal protein L5","Ribosomal protein L7","Ribosomal protein LP0","Ribosomal protein S19a","Ribosomal protein S4","Ribosomal protein S7","Ribosomal protein S9","Rieske iron-sulfur protein","robl37BC","Saccheropin dehydrogenase 1","sallimus","SCP-containing protein B","Seminal fluid protein 24Bb","Serpin 28F","Serpin 38F","Sideroflexin 1/3","sniffer","stress-sensitive B","sugar transporter 4","terribly reduced optic lobes","Trehalose transporter 1-2","Trehalose-6-phosphate synthase 1","Tropomyosin 2","twinstar","Ucp4C","UDP-glycosyltransferase family 302 member C1","Unc-89","upheld","Vacuolar H[+] ATPase 36kD subunit 1","viking","Vinculin","wings up A","yippee interacting protein 2","Yolk protein 2","Z band alternatively spliced PDZ-motif protein 52","zormin",null,null,null],["P10333","X2J8Y6","Q6GUR9","Q9VVW1","P53501","P10987","P83967","C0PDF4","O62526","A0A0B4KEL0","Q9V496","Q9VA73","Q05825","Q9VXI1","M9PF16","Q53YH3","Q9VEZ6","Q9Y112","Q0E8T1","Q9VDZ2","Q8T975","Q9VGJ4","A0A0B4KFS3","Q9VZ12","Q9VR92","A1Z7Z4","E2QCS7","Q9VKP4","Q9VQA3","Q9VIL5","Q7KW39","Q9VAJ9","Q9W227","A1ZAU0","Q7JUS1","Q8IN49","Q9VI35","Q1ECA7","Q8INU9","Q7KRS1","Q9VKT1","Q8IQF3","Q8INZ7","Q76NQ0","E1JIQ7","Q6IH48","Q9VI17","Q7KU14","C0HK94","Q7K5N8","Q9VBM3","A8JRC1","Q86BP1","A0A0B4LHC9","Q9VTL4","Q7KTF1","A1ZBJ2","Q9VZ94","Q7K4T8","Q7K3M6","Q7K3N4","Q8T4D4","Q9VJW8","A0A0B4KHN1","P08120","Q9VQG4","Q9V8R9","Q9VFP1","Q9VIQ8","P84029","M9PCQ1","Q9V3S0","Q9VDM3","A1Z787","Q9VFQ9","Q3YMU0","Q9V3A8","P08736","Q9NJH0","P13060","Q02748","Q9VQL7","C0HK95","Q95U34","Q9VLB7","P33438","Q8INJ3","P54385","Q9VQ61","E1JHQ1","P20432","P13706","A4UZZ4","Q9VFC8","Q9VAY2","P29844","O02649","Q9VMN5","M9PBL3","P11147","P29845","P84051","P84040","A0A0B4KG52","Q9VDQ0","X2JEB6","E1JJF3","Q9VWH4","Q7KB18","X2J4W8","B5RJQ0","Q03427","Q00174","P11046","Q9VLM0","Q0E9N2","Q9VN21","Q9VEB1","Q9VAZ1","Q0E8E8","Q7KSC4","Q9V397","Q9W1H8","Q9VV72","Q24400","P14318","Q9VFC7","P13607","Q9VU17","A8JNU6","Q8SXR1","M9NEZ5","Q9VXK0","Q7K084","Q7K110","Q24319","Q9I7S8","A4V1N8","Q9VZQ8","Q9VUY9","Q9VAN7","P21187","Q94920","Q9VKP2","P54399","O62619","Q7JQR3","O18332","O18335","A4V4A5","A1Z9M2","Q9VYR1","Q9VSL8","O17445","P36241","P48159","Q9W0A8","Q9VZS5","Q9V597","Q9W5R8","X2J5G6","P19889","P39018","P41042","Q9VA91","C6SUW3","Q9VQ29","Q9VJ18","Q9VN86","M9PBJ0","Q8T3W8","B4ZJ91","Q9VLQ7","Q9VII7","Q9VN13","Q9W3H4","Q26365","Q7K2U8","M9NDL5","Q8MKK4","Q9Y119","A0A0B4KHJ9","P45594","Q9VMK1","Q9VGT3","A8DYP0","M9NH07","Q9V7D2","Q9VMV5","O46037","M9PJQ5","Q9VL70","X2JB25","A0A0B4LGL0","M9PBJ1","P18091-3","Q99323-1","P19351-11"],["High conf. Sfp","High conf. Sfp","High conf. Sfp","High conf. Sfp","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","High conf. Sfp","Low conf. Sfp","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","High conf. Sfp","High conf. Sfp","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","High conf. Sfp","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","High conf. Sfp","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Low conf. Sfp","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Low conf. Sfp","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","High conf. Sfp","Sperm","Sperm","Low conf. Sfp","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Low conf. Sfp","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Low conf. Sfp","Low conf. Sfp","Low conf. Sfp","Low conf. Sfp","High conf. Sfp","Low conf. Sfp","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Low conf. Sfp","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","High conf. Sfp","High conf. Sfp","High conf. Sfp","High conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Sperm","Low conf. Sfp","Sperm","Sperm","Sperm","Sperm"],["2L","2L","2R","3L","2R","X","3R","3L","X","2R","4","3R","4","X","X","3R","3R","3L","2L","3R","3R","3R","3R","X","X","2R","2L","2L","2L","2L","X","3R","2R","2R","2R","3R","3R","2L","2L","2L","2L","3L","2L","2L","3R","2L","3R","2L","3R","2R","3R","3R","3R","2R","3L","2L","2R","3L","2R","2R","2R","2L","2L","3R","2L","2L","2R","3R","2L","2L","2L","X","3R","2R","3R","2R","3L","2R","3R","2L","2L","2L","3R","3L","2L","2L","3R","3R","2L","2L","3R","2L","2L","3R","3R","X","X","2L","3L","3R","2R","-","-","3R","3R","X","X","X","2R","2L","3L","2R","3L","2L","2L","2R","3R","3R","3R","3L","3R","2L","2R","3L","3R","2R","3R","3R","3L","3L","2L","X","X","2R","2R","X","X","3L","3L","3L","3R","2R","2L","2L","3L","3R","2R","3R","3R","X","2R","X","3L","3L","2R","2R","3L","3L","2R","2L","2L","3L","X","3L","3R","3L","2L","2L","3R","3L","3R","2L","2L","2L","3R","X","X","2R","X","2R","2L","3R","2R","2L","3R","2R","X","2R","2L","X","X","2L","X","2R","3L",null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>genes<\/th>\n      <th>Sfp<\/th>\n      <th>Chromosome<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="sfps-depleted-after-pbst-treatment" class="section level4">
<h4>Sfps depleted after PBST treatment</h4>
<pre class="r"><code>lrt.pbst_v_rest.tTags[lrt.pbst_v_rest.tTags$Sfp == &#39;SFP.high&#39; &amp; lrt.pbst_v_rest.tTags$logFC &gt; 1 &amp; lrt.pbst_v_rest.tTags$FDR &lt; 0.05, ] %&gt;% #write_csv(&#39;output/GO_lists/PBST_Sfp_depleted.csv&#39;)
  dplyr::select(FBgn, Name = NAME, Symbol = SYMBOL, Chromosome = LOCATION_ARM) %&gt;% 
  arrange(Symbol) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-879ac2b5147c82bbbbbd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-879ac2b5147c82bbbbbd">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0002855","FBgn0011559","FBgn0034153","FBgn0015586","FBgn0005585","FBgn0250847","FBgn0265264","FBgn0250841","FBgn0032868","FBgn0034753","FBgn0051659","FBgn0051680","FBgn0051872","FBgn0054051","FBgn0260766","FBgn0034229","FBgn0036186","FBgn0039562","FBgn0001218","FBgn0026415","FBgn0040098","FBgn0053126","FBgn0034277","FBgn0286818","FBgn0030362","FBgn0037888","FBgn0259952","FBgn0028987","FBgn0028986"],["Accessory gland protein 26Aa","Accessory gland protein 36DE","Accessory gland protein 53C14b","Accessory gland protein 76A","Calreticulin","CG14034","CG17097","CG17242","CG17472","CG2852","CG31659","CG31680","CG31872","CG34051","CG42564","CG4847","CG6071","Glycoprotein 93","Heat shock 70-kDa protein cognate 3","Imaginal disc growth factor 4","lectin-29Ca","Neural Lazarillo","Oligosaccharide transferase Delta subunit","Protein disulfide isomerase","regucalcin","SCP-containing protein B","Seminal fluid protein 24Bb","Serpin 28F","Serpin 38F"],["Acp26Aa","Acp36DE","Acp53C14b","Acp76A","Calr","CG14034","CG17097","CG17242","CG17472","CG2852","CG31659","CG31680","CG31872","CG34051","CG42564","CG4847","CG6071","Gp93","Hsc70-3","Idgf4","lectin-29Ca","NLaz","OstDelta","Pdi","regucalcin","scpr-B","Sfp24Bb","Spn28F","Spn38F"],["2L","2L","2R","3L","3R","2L","2L","2L","2L","2R","2L","2L","2L","2L","3R","2R","3L","3R","X","X","2L","2L","2R","3L","X","3R","2L","2L","2L"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>Symbol<\/th>\n      <th>Chromosome<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="sfps-remaining-after-pbst-treatment" class="section level4">
<h4>Sfps remaining after PBST treatment</h4>
<pre class="r"><code>lrt.pbst_v_rest.tTags[lrt.pbst_v_rest.tTags$Sfp == &#39;SFP.high&#39; &amp; lrt.pbst_v_rest.tTags$FDR &gt; 0.05, ] %&gt;% #write_csv(&#39;output/GO_lists/PBST_Sfp_remaining.csv&#39;)
  dplyr::select(FBgn, Name = NAME, Symbol = SYMBOL, Chromosome = LOCATION_ARM) %&gt;% 
  arrange(Symbol) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-ee329062ce30c37f180d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ee329062ce30c37f180d">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0003885","FBgn0004181","FBgn0011694","FBgn0020414","FBgn0260745","FBgn0050104","FBgn0264815","FBgn0261055","FBgn0259975","FBgn0052203","FBgn0036969","FBgn0036970","FBgn0034709","FBgn0003748"],["alpha-Tubulin at 84D","Ejaculatory bulb protein","Ejaculatory bulb protein II","Imaginal disc growth factor 3","midline fasciclin","Ecto-5'-nucleotidase 2","Phosphodiesterase 1c","Seminal fluid protein 26Ad","Seminal fluid protein 87B","Serpin 75F","Serpin 77Bb","Serpin 77Bc","Secreted Wg-interacting molecule","Trehalase"],["alphaTub84D","Ebp","EbpII","Idgf3","mfas","NT5E-2","Pde1c","Sfp26Ad","Sfp87B","Spn75F","Spn77Bb","Spn77Bc","Swim","Treh"],["3R","2R","2R","2L","3R","2R","2L","2L","3R","3L","3L","3L","2R","2R"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>Symbol<\/th>\n      <th>Chromosome<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="experiment-3---salt-wash" class="section level2 tabset">
<h2 class="tabset">Experiment 3 - salt wash</h2>
<p>Here we washed sperm samples with high molar NaCl or PBS (control). We performed LC-MS analysis for 4 biological replicates of each treatment, with 2 technical replicates each. We first summed technical replicates. We then filtered proteins present in fewer than 5 out of 8 replicates total.</p>
<pre class="r"><code>salt_dat2 &lt;- salt_dat %&gt;% 
  dplyr::select(FBgn, Accession, Gene.Symbol, UniqueP = X..Unique.Peptides,
                contains(&#39;Abundance..F&#39;)) %&gt;% 
  mutate(across(5:20, ~replace_na(.x, 0)),
         Sfp = case_when(FBgn %in% wigbySFP$FBgn[wigbySFP$category == &#39;highconf&#39;] ~ &#39;SFP.high&#39;,
                         FBgn %in% wigbySFP$FBgn ~ &#39;SFP.low&#39;,
                         TRUE ~ &#39;Sperm.only&#39;),
         mn = rowMeans(.[, 5:20], na.rm = TRUE)) %&gt;% 
  # filter two unique peptides
  filter(mn &gt; 0, UniqueP &gt;= 2) %&gt;% dplyr::select(-mn)

colnames(salt_dat2) &lt;- gsub(&#39;.*Sample..&#39;, &#39;&#39;, x = colnames(salt_dat2))
colnames(salt_dat2)[5:20] &lt;- paste(colnames(salt_dat2)[5:20], 1:8, sep = &#39;_&#39;)

# make object for protein abundance data and replace NA&#39;s with 0&#39;s
expr_salt &lt;- salt_dat2[, 5:20] %&gt;% 
  mutate(NaCl1 = rowSums(dplyr::select(., 1:2), na.rm = TRUE),
         NaCl2 = rowSums(dplyr::select(., 3:4), na.rm = TRUE),
         NaCl3 = rowSums(dplyr::select(., 5:6), na.rm = TRUE),
         NaCl4 = rowSums(dplyr::select(., 7:8), na.rm = TRUE),
         PBS1 = rowSums(dplyr::select(., 9:10), na.rm = TRUE),
         PBS2 = rowSums(dplyr::select(., 11:12), na.rm = TRUE),
         PBS3 = rowSums(dplyr::select(., 13:14), na.rm = TRUE),
         PBS4 = rowSums(dplyr::select(., 15:16), na.rm = TRUE)) %&gt;% 
  dplyr::select(-c(1:16))

# filter data to exclude 0&#39;s
thresh.salt = expr_salt[, ] &gt; 0
keep.salt = rowSums(thresh.salt) &gt;= 5
salt.Filtered = expr_salt[keep.salt, ]

# get sample info
sampInfo_salt = data.frame(condition = str_sub(colnames(salt.Filtered), 1, 1),
                           Replicate = str_sub(colnames(salt.Filtered), -1))

# make design matrix to test diffs between groups
design_salt &lt;- model.matrix(~0 + sampInfo_salt$condition)
colnames(design_salt) &lt;- unique(sampInfo_salt$condition)
rownames(design_salt) &lt;- sampInfo_salt$Replicate

# create DGElist and fit model 
dgeList_salt &lt;- DGEList(counts = salt.Filtered, genes = salt_dat2$Accession[keep.salt], 
                        group = sampInfo_salt$condition)
dgeList_salt &lt;- calcNormFactors(dgeList_salt, method = &#39;TMM&#39;)
dgeList_salt &lt;- estimateCommonDisp(dgeList_salt)
dgeList_salt &lt;- estimateTagwiseDisp(dgeList_salt)

# make contrasts - higher values = higher in mated
cont.salt &lt;- makeContrasts(N.v.P = N - P,
                           levels = design_salt)

glm_salt &lt;- glmFit(dgeList_salt, design_salt, robust = TRUE)</code></pre>
<div id="diagnostic-plots-2" class="section level3">
<h3>Diagnostic plots</h3>
<pre class="r"><code>par(mfrow=c(2,2))
# Biological coefficient of variation
plotBCV(dgeList_salt)
# mean-variance trend
voomed = voom(dgeList_salt, design_salt, plot=TRUE)
# QQ-plot
g &lt;- gof(glmFit(dgeList_salt, design_salt))
z &lt;- zscoreGamma(g$gof.statistics,shape=g$df/2,scale=2)
qqnorm(z); qqline(z, col = 4,lwd=1,lty=1)
# log2 transformed and normalize boxplot of counts across samples
boxplot(voomed$E, xlab=&quot;&quot;, ylab=&quot;log2(Abundance)&quot;,las=2,main=&quot;Voom transformed logCPM&quot;,
        col = c(rep(2:3, each = 4)))
abline(h=median(voomed$E),col=&quot;blue&quot;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-58-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-58-1">
Past versions of unnamed-chunk-58-1.png
</button>
</p>
<div id="fig-unnamed-chunk-58-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-58-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow=c(1,1))</code></pre>
</div>
<div id="mds-plot-2" class="section level3 tabset">
<h3 class="tabset">MDS plot</h3>
<pre class="r"><code>mdsObj &lt;- plotMDS(dgeList_salt, plot = F, dim.plot = c(1,3))$cmdscale.out

mdsObj &lt;- as.data.frame(as.matrix(mdsObj)) %&gt;% 
  rownames_to_column() %&gt;% 
  mutate(condition = str_sub(rowname, 1, 2)) %&gt;% 
  dplyr::rename(dim1 = V1, dim2 = V2, dim3 = V3)</code></pre>
<div id="dim.1-vs.-dim.-2-2" class="section level4">
<h4>Dim.1 vs. Dim. 2</h4>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim2, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs(x = &quot;Dimension 1&quot;, y = &quot;Dimension 2&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot12_db.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-60-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-60-1">
Past versions of unnamed-chunk-60-1.png
</button>
</p>
<div id="fig-unnamed-chunk-60-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-60-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.1-vs.-dim.-3-2" class="section level4">
<h4>Dim.1 vs. Dim. 3</h4>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim1, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 1&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot13.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-61-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-61-1">
Past versions of unnamed-chunk-61-1.png
</button>
</p>
<div id="fig-unnamed-chunk-61-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-61-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dim.2-vs.-dim.-3-2" class="section level4">
<h4>Dim.2 vs. Dim. 3</h4>
<pre class="r"><code>mdsObj %&gt;% 
  ggplot(aes(x = dim2, y = dim3, colour = condition)) +
  geom_point(size = 8, alpha = .7) +
  labs( x = &quot;Dimension 2&quot;, y = &quot;Dimension 3&quot;) +
  theme_bw() +
  theme(legend.title = element_blank(), 
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 12)) +
  #ggsave(&#39;plots/MDSplot23.pdf&#39;, height = 4, width = 5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-62-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-62-1">
Past versions of unnamed-chunk-62-1.png
</button>
</p>
<div id="fig-unnamed-chunk-62-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-62-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-plot-2" class="section level3">
<h3>volcano plot</h3>
<pre class="r"><code># perform likelihood ratio tests
lrt.salt &lt;- glmLRT(glm_salt, contrast = cont.salt[,&quot;N.v.P&quot;])

lm_salt.tTags.table &lt;- topTags(lrt.salt, n = NULL) %&gt;% as.data.frame() %&gt;% 
  mutate(threshold = if_else(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, &quot;SD&quot;, &quot;NS&quot;)) %&gt;% 
  left_join(salt_dat2,
            by = c(&#39;genes&#39; = &#39;Accession&#39;)) %&gt;% 
  mutate(bin = ntile(logCPM, 10))
lm_salt.tTags.table %&gt;% filter(threshold != &#39;SD&#39;) %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(alpha = .1) +
  geom_point(data = lm_salt.tTags.table %&gt;% 
               filter(threshold != &#39;NS&#39;),
             aes(colour = Sfp), alpha = .8) +
  scale_colour_viridis_d(direction = -1) +
  labs(x = &#39;logFC&#39;, y = &#39;-log10(p-value)&#39;) +
  #coord_cartesian(xlim = c(-6, 6)) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;,#c(0.8, 0.8),
        legend.title = element_blank(),
        legend.background = element_rect(fill = NA)) +
  ggrepel::geom_label_repel(
    data = lm_salt.tTags.table %&gt;% filter(threshold != &#39;NS&#39; &amp; Sfp == &#39;SFP.high&#39;),
    aes(label = Gene.Symbol),
    size = 2,
    colour = &#39;black&#39;,
    box.padding = unit(0.35, &quot;lines&quot;),
    point.padding = unit(0.3, &quot;lines&quot;),
    max.overlaps = 20
  ) +
  #ggsave(&#39;figures/MAplot_PBSTvNoHalt_DEseq2.pdf&#39;, height = 4, width = 4) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-63-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-63-1">
Past versions of unnamed-chunk-63-1.png
</button>
</p>
<div id="fig-unnamed-chunk-63-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-63-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># # number DA in each comparison
# lm_salt.tTags.table %&gt;% 
#   group_by(threshold) %&gt;% dplyr::count()

# # Sfps remaining in Salt (i.e. NS)
# lm_salt.tTags.table %&gt;% 
#   filter(threshold != &#39;SD&#39; &amp; logFC &lt; 1 &amp; Sfp == &#39;SFP.high&#39;) %&gt;% 
#   distinct(FBgn, .keep_all = TRUE)</code></pre>
</div>
<div id="ma-plot" class="section level3">
<h3>MA plot</h3>
<pre class="r"><code>salt_ma &lt;- lm_salt.tTags.table %&gt;% filter(threshold != &#39;SD&#39;) %&gt;% 
  ggplot(aes(x = logCPM, y = logFC)) +
  geom_point(alpha = .1) +
  geom_point(data = lm_salt.tTags.table %&gt;% 
               filter(threshold != &#39;NS&#39;),
             aes(colour = Sfp), alpha = .8) +
  scale_colour_viridis_d(direction = -1) +
  labs(x = &#39;log2(normalised abundance)&#39;, y = &#39;logFC&#39;) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;,#c(0.8, 0.8),
        legend.title = element_blank(),
        legend.background = element_rect(fill = NA)) +
  ggrepel::geom_label_repel(
    data = lm_salt.tTags.table %&gt;% filter(bin &gt;= 9 &amp; Sfp == &#39;SFP.high&#39;),
    aes(label = Gene.Symbol),
    size = 2, colour = &#39;black&#39;,
    box.padding = unit(0.35, &quot;lines&quot;),  point.padding = unit(0.3, &quot;lines&quot;), max.overlaps = 20
  ) +
  ggrepel::geom_label_repel(
    data = lm_salt.tTags.table %&gt;% filter(threshold == &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;),
    aes(label = Gene.Symbol),
    size = 2, colour = &#39;black&#39;,
    box.padding = unit(0.35, &quot;lines&quot;),  point.padding = unit(0.3, &quot;lines&quot;), max.overlaps = 20
  ) +
  #ggsave(&#39;figures/MAplot_Salt.pdf&#39;, height = 4.5, width = 4) +
  NULL

salt_ma</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-64-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-64-1">
Past versions of unnamed-chunk-64-1.png
</button>
</p>
<div id="fig-unnamed-chunk-64-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-64-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="refined-sfp-list" class="section level1 tabset">
<h1 class="tabset">Refined Sfp list</h1>
<p>Proteins that show no difference in abundance between treatments - i.e. proteins that putatively should be defined as sperm proteins rather than Sfps.</p>
<div id="high-confidence-sfps-remaining-after-pbst-wash" class="section level2">
<h2>High confidence Sfps remaining after PBST wash</h2>
<pre class="r"><code>lrt.pbst_v_rest.tTags %&gt;% filter(threshold != &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;) %&gt;% 
  dplyr::select(FBgn, Name = NAME, Chromosome = LOCATION_ARM, Abundance = logCPM) %&gt;% 
  mutate(across(4, ~round(.x, 2))) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-e6b3d5c3cda8cdc28426" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e6b3d5c3cda8cdc28426">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0011694","FBgn0261055","FBgn0004181","FBgn0003885","FBgn0260745","FBgn0036970","FBgn0036969","FBgn0259975","FBgn0034709","FBgn0264815","FBgn0020414","FBgn0050104","FBgn0052203","FBgn0003748"],["Ejaculatory bulb protein II","Seminal fluid protein 26Ad","Ejaculatory bulb protein","alpha-Tubulin at 84D","midline fasciclin","Serpin 77Bc","Serpin 77Bb","Seminal fluid protein 87B","Secreted Wg-interacting molecule","Phosphodiesterase 1c","Imaginal disc growth factor 3","Ecto-5'-nucleotidase 2","Serpin 75F","Trehalase"],["2R","2L","2R","3R","3R","3L","3L","3R","2R","2L","2L","2R","3L","2R"],[5.5,3.72,8.25,16.24,5.66,5.24,2.47,4.05,3.78,5.31,6.77,7.02,1.74,3.61]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Name<\/th>\n      <th>Chromosome<\/th>\n      <th>Abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="high-confidence-sfps-remaining-after-salt-wash" class="section level2">
<h2>High confidence Sfps remaining after salt wash</h2>
<pre class="r"><code>lm_salt.tTags.table %&gt;% filter(threshold != &#39;SD&#39; &amp; Sfp == &#39;SFP.high&#39;) %&gt;% 
  dplyr::select(FBgn, Symbol = Gene.Symbol, Abundance = logCPM) %&gt;% 
  mutate(across(3, ~round(.x, 2))) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-e295b598a035716f6e44" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e295b598a035716f6e44">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0034152","FBgn0260745","FBgn0038067","FBgn0054051","FBgn0030362","FBgn0001218","FBgn0032868","FBgn0011559","FBgn0026415","FBgn0036186","FBgn0037879","FBgn0051680","FBgn0034753","FBgn0005585","FBgn0023529","FBgn0000592","FBgn0039562","FBgn0051872","FBgn0034277","FBgn0053126","FBgn0022355","FBgn0050104","FBgn0260766","FBgn0038069","FBgn0265264","FBgn0052203","FBgn0054033","FBgn0259975","FBgn0264815","FBgn0003748","FBgn0040093","FBgn0040092","FBgn0032774","FBgn0040097","FBgn0002855","FBgn0032122","FBgn0259952","FBgn0040098","FBgn0261058","FBgn0036969","FBgn0034417","FBgn0028986","FBgn0265267","FBgn0028987","FBgn0003885","FBgn0034229","FBgn0083141","FBgn0250847","FBgn0030828","FBgn0033875","FBgn0051659","FBgn0083938","FBgn0051198","FBgn0043530","FBgn0250841","FBgn0004181","FBgn0037038","FBgn0250835","FBgn0015584","FBgn0011694","FBgn0083966","FBgn0050488","FBgn0034153","FBgn0043825","FBgn0261055"],["Acp53C14a","mfas","CG11598","CG34051","regucalcin","Hsc70-3","CG17472","Acp36DE","Idgf4","CG6071","scpr-C","CG31680","CG2852","Crc; Calr","CG2918","Est-6","Gp93","CG31872","CG6370; OstDelta","NLaz","Tsf1","CG30104; NT5E-2","CG42564","CG11608","CG17097","Spn75F","CG34033","Sfp87B","Pde1c","Treh","lectin-46Ca","lectin-46Cb","CG17549","lectin-30A","Acp26Aa","CG31883","Sfp24Bb","lectin-29Ca","Sfp38D","Spn77Bb","CG15117","Spn3; Spn38F","CG18258","Spn2; Spn28F","alphaTub84D","CG4847","Spn28B; Spn7","CG14034","CG5162","CG6357","CG31659","BG642163","CG31198","Obp51a","CG17242","Peb; Ebp","CG11037","CG15394","Acp53Ea","PebII; EbpII","CG34130","CG30488; antr","Acp53C14b","CG18284","Sfp26Ad"],[4.97,5.63,1.89,5.46,8.74,4.86,6.27,12,5.16,2.1,7.69,1.13,2.95,1.99,-0.13,2.58,2.05,9.33,1.29,5.31,8.65,5.45,2.44,0.98,10.81,1.78,-0.12,4.13,4.99,4.97,1.59,1.55,1.19,5.22,8.1,-1.38,3.47,3.33,1.63,4.18,4.7,4.97,2.97,4.84,16.21,7.11,-0.47,4.6,5.89,2.52,3.14,2.1,0.97,3.01,7.24,7.68,4.32,-0.88,2.85,5,2.33,1.87,2.91,5.17,5.08]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Symbol<\/th>\n      <th>Abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="overlap" class="section level2">
<h2>Overlap</h2>
<pre class="r"><code>upset(
  fromList(
    list(PBST = lrt.pbst_v_rest.tTags$genes[lrt.pbst_v_rest.tTags$threshold != &#39;SD&#39; &amp; 
                                              lrt.pbst_v_rest.tTags$Sfp == &#39;SFP.high&#39;],
         Salt = lm_salt.tTags.table$genes[lm_salt.tTags.table$threshold != &#39;SD&#39; &amp; 
                                              lm_salt.tTags.table$Sfp == &#39;SFP.high&#39;])))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-67-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>remaining_sfp &lt;- data.frame(FBgn = unique(c(
  lrt.pbst_v_rest.tTags$FBgn[lrt.pbst_v_rest.tTags$threshold != &#39;SD&#39; &amp; lrt.pbst_v_rest.tTags$Sfp == &#39;SFP.high&#39;],
  lm_salt.tTags.table$FBgn[lm_salt.tTags.table$threshold != &#39;SD&#39; &amp; lm_salt.tTags.table$Sfp == &#39;SFP.high&#39;]))) %&gt;%
  left_join(DmSP3, by = c(&#39;FBgn&#39;)) %&gt;% 
  mutate(NAME = if_else(NAME == &#39;-&#39;, ANNOTATION_SYMBOL, NAME),
         Abundance = log2(grand.mean))</code></pre>
</div>
<div id="sfps-remaining-after-salt-or-detergent-treatment" class="section level2">
<h2>Sfps remaining after salt or detergent treatment</h2>
<pre class="r"><code>remaining_sfp %&gt;% 
  dplyr::select(FBgn, Symbol = NAME, Chromosome = LOCATION_ARM, Abundance) %&gt;% 
  mutate(across(4, ~round(.x, 2))) %&gt;% 
  arrange(desc(Abundance)) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-fa553e9aa2ed17c2a950" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fa553e9aa2ed17c2a950">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0003885","FBgn0011559","FBgn0265264","FBgn0051872","FBgn0030362","FBgn0022355","FBgn0037879","FBgn0002855","FBgn0004181","FBgn0250841","FBgn0034229","FBgn0032868","FBgn0050104","FBgn0030828","FBgn0054051","FBgn0260745","FBgn0053126","FBgn0040097","FBgn0001218","FBgn0026415","FBgn0264815","FBgn0028986","FBgn0043825","FBgn0037038","FBgn0011694","FBgn0028987","FBgn0034152","FBgn0261055","FBgn0003748","FBgn0020414","FBgn0034417","FBgn0250847","FBgn0259952","FBgn0259975","FBgn0036969","FBgn0040098","FBgn0039562","FBgn0034753","FBgn0000592","FBgn0034153","FBgn0051659","FBgn0033875","FBgn0265267","FBgn0260766","FBgn0043530","FBgn0015584","FBgn0005585","FBgn0036186","FBgn0038067","FBgn0034277","FBgn0034709","FBgn0083966","FBgn0036970","FBgn0052203","FBgn0083938","FBgn0040092","FBgn0050488","FBgn0040093","FBgn0083141","FBgn0261058","FBgn0051198","FBgn0032774","FBgn0051680","FBgn0038069","FBgn0023529","FBgn0054033","FBgn0250835","FBgn0032122"],["alpha-Tubulin at 84D","Accessory gland protein 36DE","CG17097","CG31872","regucalcin","Transferrin 1","SCP-containing protein C","Accessory gland protein 26Aa","Ejaculatory bulb protein","CG17242","CG4847","CG17472","Ecto-5'-nucleotidase 2","CG5162","CG34051","midline fasciclin","Neural Lazarillo","lectin-30A","Heat shock 70-kDa protein cognate 3","Imaginal disc growth factor 4","Phosphodiesterase 1c","Serpin 38F","CG18284","CG11037","Ejaculatory bulb protein II","Serpin 28F","Accessory gland protein 53C14a","Seminal fluid protein 26Ad","Trehalase","Imaginal disc growth factor 3","CG15117","CG14034","Seminal fluid protein 24Bb","Seminal fluid protein 87B","Serpin 77Bb","lectin-29Ca","Glycoprotein 93","CG2852","Esterase 6","Accessory gland protein 53C14b","CG31659","CG6357","CG18258","CG42564","Odorant-binding protein 51a","Accessory gland protein 53Ea","Calreticulin","CG6071","CG11598","Oligosaccharide transferase Delta subunit","Secreted Wg-interacting molecule","CG34130","Serpin 77Bc","Serpin 75F","BG642163","lectin-46Cb","antares","lectin-46Ca","Serpin 28B","Seminal fluid protein 38D","CG31198","CG17549","CG31680","CG11608","CG2918","CG34033","CG15394","CG31883"],["3R","2L","2L","2L","X","X","3R","2L","2R","2L","2R","2L","2R","X","2L","3R","2L","2L","X","X","2L","2L","2L","3L","2R","2L","2R","2L","2R","2L","2R","2L","2L","3R","3L","2L","3R","2R","3L","2R","2L","2R","X","3R","2R","2R","3R","3L","3R","2R","2R","3R","3L","3L","2L","2R","2R","2R","2L","2L","3R","2L","2L","3R","X","2R","2L","2L"],[35.19,30.87,29.62,28.17,28.06,27.54,27.01,26.9,26.67,26.03,25.92,25.16,24.93,24.7,24.54,24.46,24.3,24.25,24.22,24.22,24.09,24,23.97,23.95,23.91,23.9,23.86,23.85,23.82,23.65,23.55,23.34,23.24,23.09,22.84,22.76,22.55,22.51,22.33,22.15,22.11,22.01,21.98,21.94,21.93,21.89,21.78,21.37,21.31,21.29,21.27,21.24,21.2,21.17,21.03,21.02,21.02,20.97,20.94,20.93,20.75,20.66,20.37,20.32,19.45,19.26,18.82,17.97]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>Symbol<\/th>\n      <th>Chromosome<\/th>\n      <th>Abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="flyatlas2-comparisons" class="section level1">
<h1>FlyAtlas2 comparisons</h1>
<p>We downloaded gene expression data from <a href="https://flyatlas.gla.ac.uk/FlyAtlas2/index.html">FlyAtlas2</a> (FPKM; fragments per kilobase of transcript per million mapped reads) for accessory glands, testis, and carcass. We also included female expression data for the ovary. We filtered genes with log2(FPKM) less than 2. We classified Sfps remaining in the DmSP3 after salt or detergent treatment as ‘sperm associated Sfps’ (SAS), and the remaining Sfps as ‘Sfp’.</p>
<pre class="r"><code>flyatlas2_names &lt;- c(&#39;Tissue&#39;,  
                     &#39;FPKM.m&#39;,  &#39;SD.m&#39;, &#39;Enrichment.m&#39;, 
                     &#39;FPKM.f&#39;,  &#39;SD.f&#39;, &#39;Enrichment.f&#39;, 
                     &#39;M.F&#39;, &#39;p.value&#39;,  &#39;FPKM.mf&#39;,  &#39;SD.mf&#39;,    &#39;Enrichment.mf&#39;)

flyatlas2 &lt;- bind_rows(
  read.delim(&#39;data/FlyAtlas2/FlyAtlas2_Acgs.txt&#39;, header = FALSE, na.strings = &#39;-&#39;),
  read.delim(&#39;data/FlyAtlas2/FlyAtlas2_testis.txt&#39;, header = FALSE, na.strings = &#39;-&#39;),
  read.delim(&#39;data/FlyAtlas2/FlyAtlas2_ovary.txt&#39;, header = FALSE, na.strings = &#39;-&#39;)) %&gt;% 
  filter(!str_detect(V1, &#39;Name&#39;)) %&gt;% 
  rename_at(all_of(names(.)), ~ flyatlas2_names) %&gt;% 
  mutate(across(c(2:8, 10:12), ~as.numeric(.x)),
         p.value = as.character(p.value)) %&gt;% 
  bind_rows(read.delim(&#39;data/FlyAtlas2/FlyAtlas2_carcass.txt&#39;, header = TRUE, na.strings = &#39;-&#39;) %&gt;% 
              mutate(across(c(2:8, 10:12), ~as.numeric(.x)))) %&gt;% 
  mutate(FBgn = str_match(Tissue, &quot;fbgn=(.*?)&amp;t&quot;)[,2],
         Tissue = gsub(&#39;.*:&#39;, &#39;&#39;, x = Tissue),
         across(2:12, ~as.numeric(.x)),
         fpkm_mixed = if_else(Tissue == &#39;Ovary&#39;, FPKM.f, FPKM.m),
         lfpkm = log(fpkm_mixed + 1, base = 2),
         Sfp = case_when(FBgn %in% remaining_sfp$FBgn ~ &#39;SAS&#39;,
                         FBgn %in% intersect(SFPs$FBgn, DmSP3$FBgn) ~ &#39;Sfp&#39;, 
                         #FBgn %in% SFPs$FBgn ~ &#39;Sfp&#39;, 
                         TRUE ~ &#39;other&#39;)) %&gt;% 
  left_join(DmSP3 %&gt;% dplyr::select(FBgn, NAME, SYMBOL, grand.mean), by = &#39;FBgn&#39;) %&gt;% 
  # group_by(Tissue) %&gt;%
  # mutate(bin = ntile(lfpkm_mixed, 15)) %&gt;% 
  # ungroup() %&gt;%
  # filter(bin &gt; 2) %&gt;%
  filter(FBgn %in% DmSP3$FBgn) </code></pre>
<div id="gene-expression-in-each-tissue" class="section level2">
<h2>Gene expression in each tissue</h2>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  ggplot(aes(x = Sfp, y = lfpkm, fill = Tissue)) +
  geom_boxplot(notch = TRUE) +
  scale_fill_viridis_d(direction = -1) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-70-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-70-1">
Past versions of unnamed-chunk-70-1.png
</button>
</p>
<div id="fig-unnamed-chunk-70-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-70-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  ggplot(aes(x = lfpkm, y = Tissue, fill = Tissue)) +
  ggridges::geom_density_ridges() +
  scale_fill_viridis_d(direction = -1) +
  facet_wrap(~ Sfp) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-70-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-70-2">
Past versions of unnamed-chunk-70-2.png
</button>
</p>
<div id="fig-unnamed-chunk-70-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-70-2.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  group_by(Tissue) %&gt;%
  do(fit = broom::tidy(kruskal.test(lfpkm ~ Sfp, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  #dplyr::select(`Gene set`, Statistic = statistic, p.value) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Difference in expression between Sfp classes - per tissue&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Difference in expression between Sfp classes - per tissue
</caption>
<thead>
<tr>
<th style="text-align:left;">
Tissue
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:left;">
p.value
</th>
<th style="text-align:right;">
parameter
</th>
<th style="text-align:left;">
method
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:right;">
202.283
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Kruskal-Wallis rank sum test
</td>
</tr>
<tr>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:right;">
8.021
</td>
<td style="text-align:left;">
0.018
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Kruskal-Wallis rank sum test
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:right;">
4.154
</td>
<td style="text-align:left;">
0.125
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Kruskal-Wallis rank sum test
</td>
</tr>
<tr>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:right;">
6.908
</td>
<td style="text-align:left;">
0.032
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Kruskal-Wallis rank sum test
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  group_by(Tissue) %&gt;%
  do(fit = broom::tidy(pairwise.wilcox.test(.$lfpkm, .$Sfp))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Model fits&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Model fits
</caption>
<thead>
<tr>
<th style="text-align:left;">
Tissue
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
0.015
</td>
</tr>
<tr>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.077
</td>
</tr>
<tr>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.077
</td>
</tr>
<tr>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.192
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.777
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
0.777
</td>
</tr>
<tr>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.922
</td>
</tr>
<tr>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
0.028
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  group_by(Sfp) %&gt;%
  do(fit = broom::tidy(kruskal.test(lfpkm ~ Tissue, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  dplyr::select(-method) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Kruskal-Wallace rank sum tests for difference in expression between tissues&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Kruskal-Wallace rank sum tests for difference in expression between tissues
</caption>
<thead>
<tr>
<th style="text-align:left;">
Sfp
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:left;">
p.value
</th>
<th style="text-align:right;">
parameter
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:right;">
238.286
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:right;">
83.414
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:right;">
69.785
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>flyatlas2 %&gt;% 
  filter(fpkm_mixed &gt; 2) %&gt;% 
  group_by(Sfp) %&gt;%
  do(fit = broom::tidy(pairwise.wilcox.test(.$lfpkm, .$Tissue))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Pairwise Wilcoxen rank-sum tests&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Pairwise Wilcoxen rank-sum tests
</caption>
<thead>
<tr>
<th style="text-align:left;">
Sfp
</th>
<th style="text-align:left;">
group1
</th>
<th style="text-align:left;">
group2
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
0.053
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
0.35
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
0.005
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
0.999
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:left;">
0.21
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:left;">
0.999
</td>
</tr>
</tbody>
</table>
</div>
<div id="compare-gene-expression-between-accessory-glands-and-testis" class="section level2">
<h2>Compare gene expression between accessory glands and testis</h2>
<pre class="r"><code>flyatlas2 %&gt;%
  filter(Tissue == &#39;Accessory glands&#39; | Tissue == &#39;Testis&#39;) %&gt;%
  dplyr::select(Tissue, FBgn, lfpkm, Sfp) %&gt;%
  ggplot(aes(x = Tissue, y = lfpkm)) +
  geom_line(alpha = .5, aes(group = FBgn)) +
  stat_halfeye(alpha = .5, aes(fill = Tissue)) +
  stat_pointinterval() +
  facet_wrap(~Sfp, nrow = 1) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-72-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-72-1">
Past versions of unnamed-chunk-72-1.png
</button>
</p>
<div id="fig-unnamed-chunk-72-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-72-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sperm-associated-sfps-with-higher-expression-in-testis-than-accessory-glands." class="section level2">
<h2>‘Sperm associated Sfps’ with higher expression in testis than accessory glands.</h2>
<pre class="r"><code>flyatlas2 %&gt;%
  filter(Tissue == &#39;Accessory glands&#39; | Tissue == &#39;Testis&#39;) %&gt;%
  dplyr::select(Tissue, FBgn, lfpkm, Sfp, NAME) %&gt;%
  pivot_wider(names_from = Tissue, values_from = lfpkm) %&gt;%
  filter(Testis &gt; `Accessory glands`, Sfp == &#39;SAS&#39;) %&gt;% 
  dplyr::select(-Sfp) %&gt;% 
  mutate(across(3:4, ~round(.x, 2))) %&gt;% 
  my_data_table()</code></pre>
<div id="htmlwidget-8ab8e8f1da190735b877" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8ab8e8f1da190735b877">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0020414","FBgn0022355","FBgn0026415","FBgn0033875","FBgn0037879","FBgn0050104","FBgn0264815"],["Imaginal disc growth factor 3","Transferrin 1","Imaginal disc growth factor 4","-","SCP-containing protein C","Ecto-5'-nucleotidase 2","Phosphodiesterase 1c"],[5.12,3.28,4.14,5.82,1.31,5.25,2.34],[6.58,6.08,5.53,6.14,9.56,6.16,2.64]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBgn<\/th>\n      <th>NAME<\/th>\n      <th>Accessory glands<\/th>\n      <th>Testis<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Dpseudo_respiration"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":[2,3]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="abundance-of-proteins-with-higher-expression-in-the-testis" class="section level2">
<h2>Abundance of proteins with higher expression in the testis</h2>
<pre class="r"><code>testis_expr &lt;- c(flyatlas2 %&gt;%
  filter(Tissue == &#39;Accessory glands&#39; | Tissue == &#39;Testis&#39;) %&gt;%
  dplyr::select(Tissue, FBgn, lfpkm, Sfp, NAME) %&gt;%
  pivot_wider(names_from = Tissue, values_from = lfpkm) %&gt;%
  filter(Testis &gt; `Accessory glands`, Sfp == &#39;SAS&#39;) %&gt;% pull(FBgn))

DmSP3 %&gt;% 
  mutate(testis_expr = if_else(FBgn %in% testis_expr, &#39;testis&#39;, &#39;other&#39;)) %&gt;% 
  filter(grand.mean &gt; 0) %&gt;% 
  ggplot(aes(x = testis_expr, y = log2(grand.mean))) +
  geom_boxplot() +
  #geom_jitter(alpha = .5, aes(colour = testis_expr)) +
  labs(x = &#39;XXX&#39;, y = &#39;log2(Protein abundance)&#39;) +
  #scale_colour_viridis_d(direction = -1) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        axis.title.x = element_blank()) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-74-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-74-1">
Past versions of unnamed-chunk-74-1.png
</button>
</p>
<div id="fig-unnamed-chunk-74-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-74-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmaps" class="section level2">
<h2>Heatmaps</h2>
<pre class="r"><code># make DF
fa2wide &lt;- flyatlas2 %&gt;% 
  dplyr::select(FBgn, Tissue, NAME, SYMBOL, Sfp, lfpkm) %&gt;% 
  pivot_wider(names_from = Tissue, values_from = lfpkm) %&gt;% 
  left_join(DmSP3 %&gt;% dplyr::select(FBgn, grand.mean)) %&gt;% 
  mutate(prot = log2(grand.mean + 1))

# remove rows with all 0&#39;s
keep.wide &lt;- rowSums(fa2wide[, 5:8] &gt; 0) &gt;= 1
wide.Filtered &lt;- fa2wide[keep.wide, ]

mat_scaled &lt;- t(apply(wide.Filtered[, 5:8], 1, scale))

colnames(mat_scaled) &lt;- colnames(fa2wide[, 5:8])

labs1 &lt;- rowAnnotation(Sfp_class = fa2wide$Sfp[keep.wide],
                       col = list(Sfp_class = c(other = NA,
                                                SAS = &#39;red&#39;, 
                                                Sfp = &#39;blue&#39;)),
                       Sfp_labs = anno_mark(at = c(grep(paste(testis_expr, collapse = &#39;|&#39;), 
                                                        x = fa2wide$FBgn[keep.wide])), 
                                            labels = data.frame(
                                              FBgn = testis_expr) %&gt;% 
                                              left_join(flybase_results) %&gt;% 
                                              pull(SYMBOL),
                                            labels_gp = gpar(fontsize = 10)),
                       title = NULL,
                       show_annotation_name = FALSE)

# cat_lab1 &lt;- rowAnnotation(clust_lab = sub_grp,
#                           col = list(clust_lab = c(`1` = cbPalette[2],
#                                                    `2` = cbPalette[3])),
#                           title = NULL)

ha &lt;- HeatmapAnnotation(Tissue = anno_block(gp = gpar(fill = viridis::viridis(n = 4)),
                                            labels = c(&#39;Testis&#39;, &#39;Carcass&#39;,
                                                       &#39;Accessory\nglands&#39;, 
                                                       &#39;Ovary&#39;), 
                                            labels_gp = gpar(col = c(&#39;white&#39;, &#39;white&#39;, 
                                                                     &#39;black&#39;, &#39;black&#39;), 
                                                                fontsize = 10)))

fa2_heatmap &lt;- Heatmap(mat_scaled, 
        col = viridis::inferno(15),
        heatmap_legend_param = list(title = &quot;log2(FPKM)&quot;,
                                    title_position = &quot;leftcenter-rot&quot;),
        right_annotation = labs1,
        #left_annotation = cat_lab1,
        top_annotation = ha, 
        show_row_names = FALSE, 
        show_column_names = FALSE, 
        #row_split = 2,
        column_split = 4,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL)

#pdf(&#39;figures/heatmap_flyatlas2.pdf&#39;, height = 7, width = 5)
fa2_heatmap</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-75-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-75-1">
Past versions of unnamed-chunk-75-1.png
</button>
</p>
<div id="fig-unnamed-chunk-75-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-75-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()


# make DF
sfps_wide &lt;- flyatlas2 %&gt;% 
  filter(Sfp != &#39;other&#39;) %&gt;% 
  dplyr::select(FBgn, Tissue, NAME, SYMBOL, Sfp, lfpkm) %&gt;% 
  pivot_wider(names_from = Tissue, values_from = lfpkm)

sfp_scaled &lt;- t(apply(sfps_wide[, 5:8], 1, scale))

colnames(sfp_scaled) &lt;- colnames(sfps_wide[, 5:8])

labs2 &lt;- rowAnnotation(Sfp_class = sfps_wide$Sfp,
                       col = list(Sfp_class = c(other = NA,
                                        SAS = &#39;red&#39;, 
                                        Sfp = &#39;blue&#39;)),
                       Sfp_labs = anno_mark(at = c(grep(&#39;SAS&#39;, x = sfps_wide$Sfp)), 
                                            labels = sfps_wide$SYMBOL[sfps_wide$Sfp == &#39;SAS&#39;],
                                            labels_gp = gpar(fontsize = 5)),
                       title = NULL,
                       show_annotation_name = FALSE)


ha2 &lt;- HeatmapAnnotation(Tissue = anno_block(gp = gpar(fill = viridis::viridis(n = 4)),
                                            labels = c(&#39;Accessory\nglands&#39;,
                                                       &#39;Ovary&#39;, &#39;Testis&#39;, &#39;Carcass&#39;), 
                                            labels_gp = gpar(col = c(&#39;white&#39;, &#39;white&#39;, 
                                                                     &#39;black&#39;, &#39;black&#39;), 
                                                                fontsize = 10)))

#pdf(&#39;figures/heatmap_sfp_expr.pdf&#39;, height = 8, width = 5)
Heatmap(sfp_scaled, 
        col = viridis::inferno(100),
        heatmap_legend_param = list(title = &quot;log2(FPKM)&quot;,
                                    title_position = &quot;leftcenter-rot&quot;),
        right_annotation = labs2,
        #left_annotation = cat_lab1,
        top_annotation = ha2, 
        show_row_names = FALSE, 
        show_column_names = FALSE, 
        row_split = 2,
        column_split = 4,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-75-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-75-2">
Past versions of unnamed-chunk-75-2.png
</button>
</p>
<div id="fig-unnamed-chunk-75-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-75-2.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="protein-abundance---gene-expression-correlations" class="section level2 tabset">
<h2 class="tabset">Protein abundance - gene expression correlations</h2>
<p>We tested whether protein abundance from out datasets correlated with gene expression for (i) all proteins/genes and (ii) only sperm associated Sfps (SAS) or remaining Sfps. We scaled expression and abundance data to have mean 0 and unit variance and fit linear regressions separately for each tissue type.</p>
<div id="all-proteins" class="section level3 tabset">
<h3 class="tabset">All proteins</h3>
<pre class="r"><code>scaled_dat &lt;- flyatlas2 %&gt;% 
  filter(grand.mean &gt; 0, fpkm_mixed &gt; 2) %&gt;% 
  group_by(Tissue) %&gt;%
  mutate(bin = ntile(lfpkm, 15)) %&gt;%
  ungroup() %&gt;%
  filter(bin &gt; 2) %&gt;%
  mutate(scaled_prot = scale(log2(grand.mean + 1)),
         scaled_expr = scale(lfpkm))

# plot
scaled_dat %&gt;% 
  ggplot(aes(x = scaled_expr, y = scaled_prot)) +
  geom_abline(slope = 1, intercept = 0, lty = 2) +
  geom_point(aes(colour = Tissue)) + 
  geom_smooth(method = &#39;lm&#39;, se = FALSE) +
  ggpubr::stat_cor(method = &#39;p&#39;, p.accuracy = 0.001) +
  labs(x = &#39;log2(FPKM)&#39;, y = &#39;log2(Protein abundance)&#39;) +
  scale_colour_manual(values = viridis(n = 4)[c(3, 2, 4, 1)]) +
  #scale_colour_viridis_d(direction = -1) +
  facet_wrap(~factor(Tissue, levels = c(&#39;Testis&#39;, &#39;Carcass&#39;, &#39;Accessory glands&#39;, &#39;Ovary&#39;)), 
             nrow = 1) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  #ggsave(&#39;figures/cor_prot_expr.pdf&#39;, height = 3.5, width = 12) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-76-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-76-1">
Past versions of unnamed-chunk-76-1.png
</button>
</p>
<div id="fig-unnamed-chunk-76-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-76-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="diagnostic-plots-3" class="section level4 tabset">
<h4 class="tabset">Diagnostic plots</h4>
<div id="residual-vs.-fitted" class="section level5">
<h5>residual vs. fitted</h5>
<pre class="r"><code>scaled_dat %&gt;% group_by(Tissue) %&gt;%
  do(fit = broom::augment(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  ggplot(aes(x = .fitted, y = .resid)) +
  geom_point() +
  geom_smooth(colour = &#39;red&#39;, se = FALSE) +
  facet_wrap(~Tissue, scales = &#39;free&#39;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-77-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-77-1">
Past versions of unnamed-chunk-77-1.png
</button>
</p>
<div id="fig-unnamed-chunk-77-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-77-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="scale-location" class="section level5">
<h5>Scale location</h5>
<pre class="r"><code>scaled_dat %&gt;% group_by(Tissue) %&gt;%
  do(fit = broom::augment(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  ggplot(aes(x = .fitted, y = sqrt(abs(.std.resid)))) +
  geom_point() +
  geom_smooth(colour = &#39;red&#39;, se = FALSE) +
  facet_wrap(~Tissue, scales = &#39;free&#39;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-78-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-78-1">
Past versions of unnamed-chunk-78-1.png
</button>
</p>
<div id="fig-unnamed-chunk-78-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-78-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="qq-plot" class="section level5">
<h5>QQ plot</h5>
<pre class="r"><code>scaled_dat %&gt;% group_by(Tissue) %&gt;%
  do(fit = broom::augment(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  ggplot(aes(sample = .std.resid)) + 
  stat_qq() + geom_abline(slope = 1) + 
  facet_wrap(~Tissue, scales = &#39;free&#39;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-79-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-79-1">
Past versions of unnamed-chunk-79-1.png
</button>
</p>
<div id="fig-unnamed-chunk-79-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-79-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="residuals-vs-leverage" class="section level5">
<h5>Residuals vs leverage</h5>
<pre class="r"><code>scaled_dat %&gt;% group_by(Tissue) %&gt;%
  do(fit = broom::augment(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  ggplot(aes(x = .hat, y = .std.resid)) +
  geom_point() +
  geom_smooth(colour = &#39;red&#39;, se = FALSE) +
  facet_wrap(~Tissue, scales = &#39;free&#39;)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-80-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-80-1">
Past versions of unnamed-chunk-80-1.png
</button>
</p>
<div id="fig-unnamed-chunk-80-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-80-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="model-fits" class="section level4">
<h4>Model fits</h4>
<pre class="r"><code>all_cor &lt;- scaled_dat %&gt;% group_by(Tissue) %&gt;%
  do(fit = broom::glance(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, paste0(&#39;= &#39;, round(p.value, 3)))) %&gt;% 
  dplyr::select(-adj.r.squared, -sigma, -c(df:df.residual)) %&gt;% 
  left_join(scaled_dat %&gt;% group_by(Tissue) %&gt;%
              do(fit = broom::tidy(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
              unnest(fit) %&gt;% 
              filter(term == &#39;scaled_expr&#39;) %&gt;% 
              dplyr::select(-statistic, -p.value),
            by = &#39;Tissue&#39;)

all_cor %&gt;% 
  dplyr::select(-term) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Model estimates&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Model estimates
</caption>
<thead>
<tr>
<th style="text-align:left;">
Tissue
</th>
<th style="text-align:right;">
r.squared
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:left;">
p.value
</th>
<th style="text-align:right;">
nobs
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
8.872
</td>
<td style="text-align:left;">
= 0.003
</td>
<td style="text-align:right;">
1001
</td>
<td style="text-align:right;">
0.073
</td>
<td style="text-align:right;">
0.025
</td>
</tr>
<tr>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:right;">
0.023
</td>
<td style="text-align:right;">
27.305
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
1165
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.027
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
10.486
</td>
<td style="text-align:left;">
= 0.001
</td>
<td style="text-align:right;">
825
</td>
<td style="text-align:right;">
0.116
</td>
<td style="text-align:right;">
0.036
</td>
</tr>
<tr>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:right;">
0.133
</td>
<td style="text-align:right;">
229.117
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
<td style="text-align:right;">
1498
</td>
<td style="text-align:right;">
0.460
</td>
<td style="text-align:right;">
0.030
</td>
</tr>
</tbody>
</table>
<pre class="r"><code># plot
cor_prot_expr &lt;- scaled_dat %&gt;% 
  ggplot(aes(x = scaled_expr, y = scaled_prot)) +
  geom_abline(slope = 1, intercept = 0, lty = 2) +
  geom_point(aes(colour = Tissue)) + 
  geom_smooth(method = &#39;lm&#39;, se = FALSE) +
  labs(x = &#39;log2(FPKM)&#39;, y = &#39;log2(Protein abundance)&#39;) +
  scale_colour_manual(values = viridis(n = 4)[c(3, 2, 4, 1)]) +
  facet_wrap(~factor(Tissue, levels = c(&#39;Testis&#39;, &#39;Carcass&#39;, &#39;Accessory glands&#39;, &#39;Ovary&#39;)), 
             nrow = 1) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        axis.title.x = element_blank()) +
  geom_text(data = all_cor,
             aes(x = 2, y = -1, 
                 label = paste0(&#39;rho = &#39;, round(estimate, 3), 
                                &#39;\nR^2 = &#39;, round(r.squared, 3), 
                                &#39;\np &#39;, p.value)),
             hjust = &#39;left&#39;, vjust = 1, size = 3) + 
  #ggsave(&#39;figures/cor_prot_expr.pdf&#39;, height = 3.5, width = 12) +
  NULL

cor_prot_expr</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-81-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-81-1">
Past versions of unnamed-chunk-81-1.png
</button>
</p>
<div id="fig-unnamed-chunk-81-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-81-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="sfps-only" class="section level3">
<h3>Sfps only</h3>
<pre class="r"><code>scaled_dat %&gt;% 
  filter(Sfp != &#39;other&#39;) %&gt;% 
  ggplot(aes(x = scaled_expr, y = scaled_prot)) +
  geom_abline(slope = 1, intercept = 0, lty = 2) +
  geom_point(aes(colour = Tissue)) + 
  geom_smooth(method = &#39;lm&#39;, se = FALSE) +
  labs(x = &#39;log2(FPKM)&#39;, y = &#39;log2(Protein abundance)&#39;) +
  scale_colour_manual(values = viridis(n = 4)[c(3, 2, 4, 1)]) +
  facet_grid(Sfp~factor(Tissue, levels = c(&#39;Testis&#39;, &#39;Carcass&#39;, &#39;Accessory glands&#39;, &#39;Ovary&#39;))) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-82-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-82-1">
Past versions of unnamed-chunk-82-1.png
</button>
</p>
<div id="fig-unnamed-chunk-82-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-82-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sfp_cor &lt;- scaled_dat %&gt;% 
  filter(Sfp != &#39;other&#39;) %&gt;% 
  group_by(Sfp, Tissue) %&gt;%
  do(fit = broom::glance(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
  unnest(fit) %&gt;% 
  mutate(p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  dplyr::select(-adj.r.squared, -sigma, -c(df:df.residual)) %&gt;% 
  left_join(scaled_dat %&gt;% 
              filter(Sfp != &#39;other&#39;) %&gt;% 
              group_by(Sfp, Tissue) %&gt;%
              do(fit = broom::tidy(lm(scaled_prot ~ scaled_expr, data = .))) %&gt;% 
              unnest(fit) %&gt;% 
              dplyr::select(-statistic, -p.value) %&gt;% 
              filter(term == &#39;scaled_expr&#39;),
            by = c(&#39;Sfp&#39;, &#39;Tissue&#39;)) %&gt;% 
  dplyr::select(-term)

sfp_cor %&gt;% 
  kable(digits = 3, 
        caption = &#39;Model summaries&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Model summaries
</caption>
<thead>
<tr>
<th style="text-align:left;">
Sfp
</th>
<th style="text-align:left;">
Tissue
</th>
<th style="text-align:right;">
r.squared
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
nobs
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:right;">
0.024
</td>
<td style="text-align:right;">
1.372
</td>
<td style="text-align:right;">
0.246
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
0.071
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
3.968
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:right;">
0.087
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
1.469
</td>
<td style="text-align:right;">
0.271
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.420
</td>
<td style="text-align:right;">
0.347
</td>
</tr>
<tr>
<td style="text-align:left;">
SAS
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:right;">
0.152
</td>
<td style="text-align:right;">
8.410
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.399
</td>
<td style="text-align:right;">
0.137
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Accessory glands
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
9.571
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
0.274
</td>
<td style="text-align:right;">
0.089
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Carcass
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
2.757
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
0.182
</td>
<td style="text-align:right;">
0.110
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Ovary
</td>
<td style="text-align:right;">
0.048
</td>
<td style="text-align:right;">
0.202
</td>
<td style="text-align:right;">
0.677
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.302
</td>
<td style="text-align:right;">
0.673
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:left;">
Testis
</td>
<td style="text-align:right;">
0.147
</td>
<td style="text-align:right;">
4.653
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.281
</td>
<td style="text-align:right;">
0.130
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>Sfp_prot_expr &lt;- scaled_dat %&gt;% 
  filter(Sfp != &#39;other&#39;) %&gt;% 
  ggplot(aes(x = scaled_expr, y = scaled_prot)) +
  geom_abline(slope = 1, intercept = 0, lty = 2) +
  geom_point(aes(colour = Tissue)) + 
  geom_smooth(method = &#39;lm&#39;, se = FALSE) +
  labs(x = &#39;log2(FPKM)&#39;, y = &#39;log2(Protein abundance)&#39;) +
  scale_colour_manual(values = viridis(n = 4)[c(3, 2, 4, 1)]) +
  facet_grid(Sfp~factor(Tissue, levels = c(&#39;Testis&#39;, &#39;Carcass&#39;, &#39;Accessory glands&#39;, &#39;Ovary&#39;))) +
  theme_bw() +
  theme(legend.position = &#39;&#39;) +
  geom_text(data = sfp_cor,
             aes(x = 2, y = -1, 
                 label = paste0(&#39;rho = &#39;, round(estimate, 3), 
                                &#39;\nR^2 = &#39;, round(r.squared, 3), 
                                &#39;\np &#39;, p.value)),
             hjust = &#39;left&#39;, vjust = 1, size = 3) + 
  #ggsave(&#39;figures/Sfp_prot_expr.pdf&#39;, height = 4, width = 8) +
  NULL

Sfp_prot_expr</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-82-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-82-2">
Past versions of unnamed-chunk-82-2.png
</button>
</p>
<div id="fig-unnamed-chunk-82-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-82-2.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="accessory-glands-vs.-testis-only" class="section level3">
<h3>Accessory glands vs. Testis only</h3>
<pre class="r"><code>scaled_dat %&gt;% 
  filter(Sfp != &#39;other&#39;, Tissue == &#39;Accessory glands&#39; | Tissue == &#39;Testis&#39;) %&gt;% 
  ggplot(aes(x = scaled_expr, y = scaled_prot, colour = Tissue)) +
  geom_abline(slope = 1, intercept = 0, lty = 2) +
  geom_point() + 
  geom_smooth(method = &#39;lm&#39;, se = FALSE) +
  labs(x = &#39;Gene expression (FPKM)&#39;, y = &#39;Protein abundance&#39;) +
  scale_colour_manual(values = viridis(n = 4)[c(3, 1)]) +
  facet_wrap(~ Sfp) +
  theme_bw() +
  theme(legend.position = &#39;bottom&#39;) +
  geom_text(data = sfp_cor %&gt;% filter(Tissue == &#39;Accessory glands&#39; | Tissue == &#39;Testis&#39;),
             aes(x = c(1.5, 3, -1, 0.5), y = c(-1.25, -1.25, 3, 3), 
                 label = paste0(&#39;rho = &#39;, round(estimate, 3), 
                                &#39;\nR^2 = &#39;, round(r.squared, 3), 
                                &#39;\np = &#39;, p.value)),
             hjust = &#39;left&#39;, vjust = 1, size = 4, show.legend = FALSE) + 
  #ggsave(&#39;figures/Sfp_prot_expr_ag-te.pdf&#39;, height = 5, width = 8) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-83-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-83-1">
Past versions of unnamed-chunk-83-1.png
</button>
</p>
<div id="fig-unnamed-chunk-83-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-83-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="combined-plot" class="section level3">
<h3>Combined plot</h3>
<pre class="r"><code>#pdf(&#39;figures/fly_atlas_cors.pdf&#39;, height = 7, width = 8)
gridExtra::grid.arrange(#fa2_heatmap, 
                        cor_prot_expr,
                        Sfp_prot_expr, 
                        layout_matrix = rbind(c(1, 1, 1),
                                              c(2, 2, 2),
                                              c(2, 2, 2))
                        )</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-84-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-84-1">
Past versions of unnamed-chunk-84-1.png
</button>
</p>
<div id="fig-unnamed-chunk-84-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-84-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
</div>
</div>
<div id="evolutionary-rates" class="section level1 tabset">
<h1 class="tabset">Evolutionary rates</h1>
<p>Here we import the results of an evolutionary rates analysis from <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a> generated using a custom pipeline <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.13113">Wright et al. (2015) <em>Molecular Ecology</em></a>. We downloaded coding sequence (CDS) files for <em>D. melanogaster</em>, <em>D. sechellia</em>, <em>D. simulans</em>, and <em>D. yakuba</em>, and the protein coding sequence for <em>D. melanogaster</em> from <a href="https://www.ensembl.org/index.html">Ensembl</a>. We then identified the longest isoform of each gene and found reciprocal 1:1 orthologs. We aligned orthologs using <a href="https://www.ebi.ac.uk/research/goldman/software/prank">PRANK</a> and masked poorly aligned reads using <a href="https://journals.sagepub.com/doi/10.4137/EBO.S18193">SWAMP</a>. We then calculated the ratio of nonsynoymous nucleotide substitutions (dN) to synonymous nucleotide substitutions (dS) for each gene using the one ratio test in CODEML (model 0). Finally, we filtered genes with dS &lt; 2 and S*dS &lt;= 1.</p>
<pre class="r"><code>one_ratio &lt;- read.csv(&#39;output/PAML/oneRatio_SWAMP_nofilter.csv&#39;) %&gt;% 
  # filter dS &lt; 2...
  filter(dS &lt; 2) %&gt;% 
  # filter S x dS &lt;= 1
  filter(S.dS &gt;= 1) %&gt;% 
  # get FBgn from FBtr
  left_join(gene2tran2prot, by = &#39;FBtr&#39;) %&gt;%
  # get chromosome
  left_join(flybase_results %&gt;% dplyr::select(FBgn, LOCATION_ARM, SYMBOL), 
            by = c(&#39;FBgn&#39;)) %&gt;% 
  # get the FBgn to convert to gene names
  #write_csv(&#39;data/one_ratio_wFBgn.csv&#39;) %&gt;% 
  distinct(FBgn, .keep_all = TRUE) %&gt;% 
  mutate(sperm_protein = case_when(FBgn %in% DmSP3$FBgn ~ &#39;Sperm&#39;, TRUE ~ &#39;Non-sperm&#39;),
         x_linked = case_when(LOCATION_ARM == &#39;X&#39; ~ &#39;X-linked&#39;, TRUE ~ &#39;other&#39;),
         y_linked = case_when(LOCATION_ARM == &#39;Y&#39; ~ &#39;Y-linked&#39;, TRUE ~ &#39;other&#39;), # non
         Sfp = case_when(FBgn %in% SFPs$FBgn ~ &#39;Sfp&#39;, TRUE ~ &#39;other&#39;)) %&gt;% 
  # bin evo rates - each bin contains same number of genes - higher bin = higher rate
  mutate(bin = ntile(dN.dS, 10))</code></pre>
<p>We compared <span class="math inline">\(\omega\)</span> between groups of genes found in the sperm proteome compared to ‘all’ genes in the genome (n = 11457). Each group represents an independent set of genes such that ‘Sfps’ found in the sperm proteome are only in SpermP. Likewise, X_link are not included in SpermP, etc. The dashed line represents the genome average dN/dS.</p>
<pre class="r"><code># Mann-Whitney U tests - sperm proteins vs. other in genome
# all proteins
wilcox.test(one_ratio$dN.dS[one_ratio$sperm_protein == &#39;Sperm&#39;], 
            one_ratio$dN.dS[one_ratio$sperm_protein == &#39;Non-sperm&#39;])</code></pre>
<pre>
    Wilcoxon rank sum test with continuity correction

data:  one_ratio$dN.dS[one_ratio$sperm_protein == "Sperm"] and one_ratio$dN.dS[one_ratio$sperm_protein == "Non-sperm"]
W = 9676362, p-value < 2.2e-16
alternative hypothesis: true location shift is not equal to 0
</pre>
<pre class="r"><code># sfps
wilcox.test(one_ratio$dN.dS[one_ratio$Sfp == &#39;Sfp&#39; &amp; one_ratio$sperm_protein == &#39;Non-sperm&#39;], 
            one_ratio$dN.dS[one_ratio$Sfp == &#39;Sfp&#39; &amp; one_ratio$sperm_protein == &#39;Sperm&#39;])</code></pre>
<pre>
    Wilcoxon rank sum test with continuity correction

data:  one_ratio$dN.dS[one_ratio$Sfp == "Sfp" & one_ratio$sperm_protein == "Non-sperm"] and one_ratio$dN.dS[one_ratio$Sfp == "Sfp" & one_ratio$sperm_protein == "Sperm"]
W = 4294.5, p-value = 0.2319
alternative hypothesis: true location shift is not equal to 0
</pre>
<pre class="r"><code># x-linked
wilcox.test(one_ratio$dN.dS[one_ratio$x_linked == &#39;X-linked&#39; &amp; one_ratio$sperm_protein == &#39;Non-sperm&#39;], 
            one_ratio$dN.dS[one_ratio$x_linked == &#39;X-linked&#39; &amp; one_ratio$sperm_protein == &#39;Sperm&#39;])</code></pre>
<pre>
    Wilcoxon rank sum test with continuity correction

data:  one_ratio$dN.dS[one_ratio$x_linked == "X-linked" & one_ratio$sperm_protein == "Non-sperm"] and one_ratio$dN.dS[one_ratio$x_linked == "X-linked" & one_ratio$sperm_protein == "Sperm"]
W = 230625, p-value = 2.424e-06
alternative hypothesis: true location shift is not equal to 0
</pre>
<div id="sperm-proteins-vs.-other" class="section level2">
<h2>Sperm proteins vs. other</h2>
<pre class="r"><code># boxplots
bind_rows(one_ratio %&gt;% mutate(gene_set = &#39;Sfp&#39;) %&gt;% filter(Sfp == &#39;Sfp&#39;),
          one_ratio %&gt;% mutate(gene_set = &#39;X-linked&#39;) %&gt;% filter(x_linked == &#39;X-linked&#39;),
          one_ratio %&gt;% mutate(gene_set = &#39;All&#39;)) %&gt;% 
  ggplot(aes(x = gene_set, y = dN.dS, fill = sperm_protein)) +
  geom_boxplot(notch = TRUE, outlier.shape = 1) +
  theme_bw() +
  scale_fill_manual(values = cbPalette[-1]) +
  labs(x = &#39;Gene set&#39;, y = &#39;dN/dS&#39;) +
  theme(legend.position = &#39;bottom&#39;,#c(0.8, 0.9),
        legend.title = element_blank(),
        legend.background = element_blank()) +
    guides(fill = guide_legend(label.position = &quot;left&quot;, label.hjust = 1)) +
  geom_signif(
    y_position = rep(1.6, 3),
    xmin = c(0.8, 1.8, 2.8),
    xmax = c(1.2, 2.2, 3.2),
    annotation = c(&quot;***&quot;, &quot;n.s.&quot;, &quot;***&quot;), tip_length = 0.01) +
  #ggsave(&#39;figures/sperm_rates_all.vs.all.pdf&#39;, height = 4.5, width = 4) +
  NULL</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-87-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-87-1">
Past versions of unnamed-chunk-87-1.png
</button>
</p>
<div id="fig-unnamed-chunk-87-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-87-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># sperm rates only
sperm_rates &lt;- one_ratio %&gt;% 
  filter(sperm_protein == &#39;Sperm&#39;) %&gt;% 
  mutate(sperm_bin = ntile(dN.dS, 10),
         Sfp = if_else(FBgn %in% remaining_sfp$FBgn, &#39;Sfp&#39;, &#39;other&#39;),
         OMIM = if_else(FBgn %in% intersect(DmSP3$FBgn, unique(OMIM$FBgn[is.na(OMIM$OMIM_Phenotype_IDs) == FALSE])), &#39;omim&#39;, &#39;other&#39;))

# write_csv(sperm_rates %&gt;% filter(sperm_bin == 10), file = &#39;output/GO_lists/fast_sperm.csv&#39;)
# write_csv(sperm_rates %&gt;% filter(sperm_bin == 1), file = &#39;output/GO_lists/slow_sperm.csv&#39;)

sperm_rates2 &lt;- bind_rows(sperm_rates %&gt;% mutate(gene_set = &#39;OMIM&#39;) %&gt;% filter(OMIM == &#39;omim&#39;),
                          sperm_rates %&gt;% mutate(gene_set = &#39;Sfp&#39;) %&gt;% filter(Sfp == &#39;Sfp&#39;),
                          sperm_rates %&gt;% mutate(gene_set = &#39;X-linked&#39;) %&gt;% filter(x_linked == &#39;X-linked&#39;),
                          sperm_rates %&gt;% mutate(gene_set = &#39;All&#39;))</code></pre>
</div>
<div id="sperm-rates" class="section level2">
<h2>Sperm rates</h2>
<pre class="r"><code>bind_rows(broom::tidy(wilcox.test(sperm_rates$dN.dS[sperm_rates$OMIM == &#39;omim&#39;], 
            sperm_rates$dN.dS[sperm_rates$OMIM != &#39;omim&#39;])),
          broom::tidy(wilcox.test(sperm_rates$dN.dS[sperm_rates$Sfp == &#39;Sfp&#39;], 
            sperm_rates$dN.dS[sperm_rates$Sfp != &#39;Sfp&#39;])),
          broom::tidy(wilcox.test(sperm_rates$dN.dS[sperm_rates$x_linked == &#39;X-linked&#39;], 
            sperm_rates$dN.dS[sperm_rates$x_linked != &#39;X-linked&#39;]))) %&gt;% 
  mutate(`Gene set` = c(&#39;OMIM&#39;, &#39;Sfp&#39;, &#39;X-linked&#39;),
         p.value = ifelse(p.value &lt; 0.001, &#39;&lt; 0.001&#39;, round(p.value, 3))) %&gt;% 
  dplyr::select(`Gene set`, Statistic = statistic, p.value) %&gt;% 
  kable(digits = 3, 
        caption = &#39;Wilcoxon rank sum test with continuity correction testing difference in evolutionary rates between each set of genes to the genome average&#39;) %&gt;% 
  kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Wilcoxon rank sum test with continuity correction testing difference in evolutionary rates between each set of genes to the genome average
</caption>
<thead>
<tr>
<th style="text-align:left;">
Gene set
</th>
<th style="text-align:right;">
Statistic
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
OMIM
</td>
<td style="text-align:right;">
623009.5
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Sfp
</td>
<td style="text-align:right;">
77064.5
</td>
<td style="text-align:left;">
&lt; 0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
X-linked
</td>
<td style="text-align:right;">
349603.5
</td>
<td style="text-align:left;">
0.955
</td>
</tr>
</tbody>
</table>
</div>
<div id="means-and-standard-errors" class="section level2">
<h2>Means and standard errors</h2>
<pre class="r"><code># means ± SE
rates_plot &lt;- sperm_rates2 %&gt;% 
  group_by(gene_set) %&gt;% 
  summarise(N = n(),
            mn = mean(dN.dS),
            se = sd(dN.dS)/sqrt(N)) %&gt;% 
  ggplot(aes(x = gene_set, y = mn, colour = gene_set)) +
  geom_point(size = 3) +
  geom_errorbar(aes(ymin = mn - se, ymax = mn + se), width = .35) +
  geom_hline(yintercept = mean(one_ratio$dN.dS), lty = 2) +
  scale_colour_manual(values = cbPalette) +
  labs(x = &#39;Gene set&#39;, y = &#39;dN/dS&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;,
        legend.title = element_blank(),
        legend.background = element_blank()) + 
  geom_text(aes(y = 0.05, label = N), size = 5, colour = &quot;black&quot;) +
  geom_signif(y_position = 0.205, xmin = 1, xmax = 2, annotation = &quot;***&quot;, tip_length = 0.01, colour = &#39;black&#39;) +
  geom_signif(y_position = 0.215, xmin = 1, xmax = 3, annotation = &quot;***&quot;, tip_length = 0.01, colour = &#39;black&#39;) +
  geom_signif(y_position = 0.225, xmin = 1, xmax = 4, annotation = &quot;n.s.&quot;, tip_length = 0.01, colour = &#39;black&#39;) +
  #ggsave(&#39;figures/sperm_rates_mn.pdf&#39;, height = 3, width = 3) +
  NULL

rates_plot</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-89-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-89-1">
Past versions of unnamed-chunk-89-1.png
</button>
</p>
<div id="fig-unnamed-chunk-89-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-89-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-experiments" class="section level1 tabset">
<h1 class="tabset">Compare experiments</h1>
<div id="exp-1-vs.-exp-2" class="section level2">
<h2>Exp 1 vs. Exp 2</h2>
<pre class="r"><code># plot exp 1 abundance vs. exp 2 no halt (PBS) abundance
one_v_two &lt;- DmSP3 %&gt;% 
  filter(mn.one &gt; 0, mn.two &gt; 0) %&gt;% 
  dplyr::select(mn.one, mn.two) %&gt;% 
  ggplot(aes(x = log2(mn.one), y = log2(mn.two))) + 
  geom_point() +
  geom_smooth(method = &#39;lm&#39;) +
  labs(x = &#39;Experiment 1&#39;, y = &#39;Experiment 2&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;) +
  stat_cor(aes(label = paste(..r.label.., ..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)), method = &#39;p&#39;) +
  NULL

one_v_two</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-90-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-90-1">
Past versions of unnamed-chunk-90-1.png
</button>
</p>
<div id="fig-unnamed-chunk-90-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-90-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-90-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="exp-1-vs.-exp-3" class="section level2">
<h2>Exp 1 vs. Exp 3</h2>
<pre class="r"><code># plot exp 1 abundance vs. exp 3 abundance
one_v_three &lt;- DmSP3 %&gt;% 
  filter(mn.one &gt; 0, mn.three &gt; 0) %&gt;% 
  dplyr::select(mn.one, mn.three) %&gt;% 
  ggplot(aes(x = log2(mn.one), y = log2(mn.three))) + 
  geom_point() +
  geom_smooth(method = &#39;lm&#39;) +
  labs(x = &#39;Experiment 1&#39;, y = &#39;Experiment 3&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;) +
  stat_cor(aes(label = paste(..r.label.., ..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)), method = &#39;p&#39;) +
  NULL

one_v_three</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-91-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-91-1">
Past versions of unnamed-chunk-91-1.png
</button>
</p>
<div id="fig-unnamed-chunk-91-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-91-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-91-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="exp-2-vs.-exp-3" class="section level2">
<h2>Exp 2 vs. Exp 3</h2>
<pre class="r"><code># plot exp 2 abundance vs. exp 3 abundance
two_v_three &lt;- DmSP3 %&gt;% 
  filter(mn.two &gt; 0, mn.three &gt; 0) %&gt;% 
  dplyr::select(mn.two, mn.three) %&gt;% 
  ggplot(aes(x = log2(mn.two), y = log2(mn.three))) + 
  geom_point() +
  geom_smooth(method = &#39;lm&#39;) +
  labs(x = &#39;Experiment 2&#39;, y = &#39;Experiment 3&#39;) +
  theme_bw() + 
  theme(legend.position = &#39;&#39;) +
  stat_cor(aes(label = paste(..r.label.., ..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)), method = &#39;p&#39;) +
  NULL

two_v_three</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-92-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-92-1">
Past versions of unnamed-chunk-92-1.png
</button>
</p>
<div id="fig-unnamed-chunk-92-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-92-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-92-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="make-combined-plots" class="section level1 tabset">
<h1 class="tabset">Make combined plots</h1>
<div id="representationnumbers" class="section level2">
<h2>Representation/numbers</h2>
<pre class="r"><code>#pdf(&#39;figures/ID_plot.pdf&#39;, height = 5, width = 9)
gridExtra::grid.arrange(#cum_plot, 
                        DmSP_overlap,
                        omim_plot, ncol = 2
                        # layout_matrix = rbind(c(1, 1, 1, 1, 2, 2),
                        #                       c(1, 1, 1, 1, 2, 2),
                        #                       c(1, 1, 1, 1, 3, 3),
                        #                       c(1, 1, 1, 1, 3, 3))
                        )</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-93-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-93-1">
Past versions of unnamed-chunk-93-1.png
</button>
</p>
<div id="fig-unnamed-chunk-93-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-93-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-93-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="correlation-between-experiemnts" class="section level2">
<h2>Correlation between experiemnts</h2>
<pre class="r"><code>#pdf(&#39;figures/corr_plot.pdf&#39;, height = 4, width = 12)
gridExtra::grid.arrange(one_v_two, one_v_three, two_v_three, ncol = 3)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-94-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-94-1">
Past versions of unnamed-chunk-94-1.png
</button>
</p>
<div id="fig-unnamed-chunk-94-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-94-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-94-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="ribosome-plot" class="section level2">
<h2>Ribosome plot</h2>
<pre class="r"><code>#pdf(&#39;figures/rib_combined.pdf&#39;, height = 6, width = 9)
gridExtra::grid.arrange(rib_plot,
                        rib_br_sp,
                        ribo_abun,
                        layout_matrix = rbind(c(3, 1, 1),
                                              c(2, 1, 1)))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-95-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-95-1">
Past versions of unnamed-chunk-95-1.png
</button>
</p>
<div id="fig-unnamed-chunk-95-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-95-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-95-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="chromosomal-comparisons" class="section level2">
<h2>Chromosomal comparisons</h2>
<pre class="r"><code>#pdf(&#39;figures/chm_combined.pdf&#39;, height = 5, width = 12)
gridExtra::grid.arrange(grobs = list(ggplotGrob(chm_plot),
                                     ggplotGrob(axy_plot)),
                        layout_matrix = rbind(c(1, 1, 1, 2, 2),
                                              c(1, 1, 1, 2, 2)))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-96-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-96-1">
Past versions of unnamed-chunk-96-1.png
</button>
</p>
<div id="fig-unnamed-chunk-96-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-96-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-96-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="evolutionary-dynamics-plot" class="section level2">
<h2>Evolutionary dynamics plot</h2>
<pre class="r"><code>#pdf(&#39;figures/evo_combined.pdf&#39;, height = 5, width = 12)
gridExtra::grid.arrange(grobs = list(ggplotGrob(age_plot),
                                     ggplotGrob(rates_plot)),
                        layout_matrix = rbind(c(1, 1, 1, 2, 2),
                                              c(1, 1, 1, 2, 2)))</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-97-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-97-1">
Past versions of unnamed-chunk-97-1.png
</button>
</p>
<div id="fig-unnamed-chunk-97-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-97-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-97-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="sfp-plot" class="section level2">
<h2>Sfp plot</h2>
<pre class="r"><code>#pdf(&#39;figures/sfp_combined.pdf&#39;, height = 4, width = 12)
gridExtra::grid.arrange(sfp_abun, 
                        #sfp_euler, 
                        pbst_volcano + theme(legend.position = &#39;&#39;), 
                        salt_ma + theme(legend.position = &#39;&#39;),
                        ncol = 3)</code></pre>
<p><img src="figure/statistical_analysis.Rmd/unnamed-chunk-98-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-98-1">
Past versions of unnamed-chunk-98-1.png
</button>
</p>
<div id="fig-unnamed-chunk-98-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/3ce1c47a48de012c7883d568131a50529ecd5316/docs/figure/statistical_analysis.Rmd/unnamed-chunk-98-1.png" target="_blank">3ce1c47</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MartinGarlovsky/DmSP3/blob/5b13fbc15787c20e47ccf83bc223aacfeb9e6db7/docs/figure/statistical_analysis.Rmd/unnamed-chunk-98-1.png" target="_blank">5b13fbc</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2022-02-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] knitrhooks_0.0.4     knitr_1.33           pals_1.7            
 [4] DT_0.18              boot_1.3-28          ComplexHeatmap_2.9.3
 [7] pheatmap_1.0.12      edgeR_3.32.1         limma_3.46.0        
[10] ggpubr_0.4.0         kableExtra_1.3.4     tidybayes_3.0.1     
[13] readxl_1.3.1         eulerr_6.1.0         UpSetR_1.4.0        
[16] forcats_0.5.1        stringr_1.4.0        dplyr_1.0.7         
[19] purrr_0.3.4          readr_2.0.1          tidyr_1.1.4         
[22] tibble_3.1.5         ggplot2_3.3.5        tidyverse_1.3.1     
[25] workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] backports_1.2.1      circlize_0.4.13      systemfonts_1.0.2   
  [4] plyr_1.8.6           splines_4.0.3        polylabelr_0.2.0    
  [7] svUnit_1.0.6         crosstalk_1.1.1      digest_0.6.28       
 [10] foreach_1.5.1        htmltools_0.5.1.1    viridis_0.6.1       
 [13] fansi_0.5.0          magrittr_2.0.1       checkmate_2.0.0     
 [16] cluster_2.1.2        doParallel_1.0.16    tzdb_0.1.2          
 [19] openxlsx_4.2.4       modelr_0.1.8         matrixStats_0.60.0  
 [22] vroom_1.5.4          svglite_2.0.0        colorspace_2.0-2    
 [25] ggrepel_0.9.1        rvest_1.0.1          ggdist_3.0.0        
 [28] haven_2.4.3          xfun_0.25            crayon_1.4.2        
 [31] jsonlite_1.7.2       iterators_1.0.13     glue_1.5.0          
 [34] polyclip_1.10-0      gtable_0.3.0         webshot_0.5.2       
 [37] GetoptLong_1.0.5     distributional_0.2.2 car_3.0-11          
 [40] shape_1.4.6          BiocGenerics_0.36.1  maps_3.4.0          
 [43] abind_1.4-5          scales_1.1.1         DBI_1.1.1           
 [46] rstatix_0.7.0        Rcpp_1.0.7           viridisLite_0.4.0   
 [49] clue_0.3-59          bit_4.0.4            foreign_0.8-81      
 [52] mapproj_1.2.7        stats4_4.0.3         htmlwidgets_1.5.3   
 [55] httr_1.4.2           arrayhelpers_1.1-0   RColorBrewer_1.1-2  
 [58] posterior_1.0.1      ellipsis_0.3.2       pkgconfig_2.0.3     
 [61] farver_2.1.0         sass_0.4.0           dbplyr_2.1.1        
 [64] locfit_1.5-9.4       utf8_1.2.2           reshape2_1.4.4      
 [67] labeling_0.4.2       tidyselect_1.1.1     rlang_0.4.12        
 [70] later_1.3.0          munsell_0.5.0        cellranger_1.1.0    
 [73] tools_4.0.3          cli_3.1.0            generics_0.1.1      
 [76] ggridges_0.5.3       broom_0.7.9          evaluate_0.14       
 [79] yaml_2.2.1           bit64_4.0.5          fs_1.5.0            
 [82] zip_2.2.0            nlme_3.1-152         whisker_0.4         
 [85] xml2_1.3.2           compiler_4.0.3       rstudioapi_0.13     
 [88] curl_4.3.2           png_0.1-7            ggsignif_0.6.2      
 [91] reprex_2.0.1         bslib_0.2.5.1        stringi_1.7.5       
 [94] highr_0.9            lattice_0.20-44      Matrix_1.3-4        
 [97] tensorA_0.36.2       vctrs_0.3.8          pillar_1.6.4        
[100] lifecycle_1.0.1      jquerylib_0.1.4      GlobalOptions_0.1.2 
[103] data.table_1.14.0    httpuv_1.6.2         R6_2.5.1            
[106] promises_1.2.0.1     gridExtra_2.3        rio_0.5.27          
[109] IRanges_2.24.1       codetools_0.2-18     dichromat_2.0-0     
[112] assertthat_0.2.1     rprojroot_2.0.2      rjson_0.2.20        
[115] withr_2.4.2          S4Vectors_0.28.1     mgcv_1.8-36         
[118] parallel_4.0.3       hms_1.1.0            gghalves_0.1.1      
[121] coda_0.19-4          rmarkdown_2.10       carData_3.0-4       
[124] Cairo_1.5-12.2       git2r_0.28.0         lubridate_1.7.10    
</pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
